// @arcgis/map-components-react/dist/components.js
/* eslint-disable */
/* tslint:disable */
/* auto-generated react proxies */
import { createReactComponent } from './react-component-lib';
export const ArcgisAreaMeasurement2d = /*@__PURE__*/ createReactComponent('arcgis-area-measurement-2d');
export const ArcgisAreaMeasurement3d = /*@__PURE__*/ createReactComponent('arcgis-area-measurement-3d');
export const ArcgisAttachments = /*@__PURE__*/ createReactComponent('arcgis-attachments');
export const ArcgisBasemapGallery = /*@__PURE__*/ createReactComponent('arcgis-basemap-gallery');
export const ArcgisBasemapLayerList = /*@__PURE__*/ createReactComponent('arcgis-basemap-layer-list');
export const ArcgisBasemapToggle = /*@__PURE__*/ createReactComponent('arcgis-basemap-toggle');
export const ArcgisBookmarks = /*@__PURE__*/ createReactComponent('arcgis-bookmarks');
export const ArcgisBuildingExplorer = /*@__PURE__*/ createReactComponent('arcgis-building-explorer');
export const ArcgisCatalogLayerList = /*@__PURE__*/ createReactComponent('arcgis-catalog-layer-list');
export const ArcgisCompass = /*@__PURE__*/ createReactComponent('arcgis-compass');
export const ArcgisCoordinateConversion = /*@__PURE__*/ createReactComponent('arcgis-coordinate-conversion');
export const ArcgisDaylight = /*@__PURE__*/ createReactComponent('arcgis-daylight');
export const ArcgisDirectionalPad = /*@__PURE__*/ createReactComponent('arcgis-directional-pad');
export const ArcgisDirections = /*@__PURE__*/ createReactComponent('arcgis-directions');
export const ArcgisDirectlineMeasurement3d = /*@__PURE__*/ createReactComponent('arcgis-directline-measurement-3d');
export const ArcgisDistanceMeasurement2d = /*@__PURE__*/ createReactComponent('arcgis-distance-measurement-2d');
export const ArcgisEditor = /*@__PURE__*/ createReactComponent('arcgis-editor');
export const ArcgisElevationProfile = /*@__PURE__*/ createReactComponent('arcgis-elevation-profile');
export const ArcgisExpand = /*@__PURE__*/ createReactComponent('arcgis-expand');
export const ArcgisFeature = /*@__PURE__*/ createReactComponent('arcgis-feature');
export const ArcgisFeatureForm = /*@__PURE__*/ createReactComponent('arcgis-feature-form');
export const ArcgisFeatureTable = /*@__PURE__*/ createReactComponent('arcgis-feature-table');
export const ArcgisFeatureTemplates = /*@__PURE__*/ createReactComponent('arcgis-feature-templates');
export const ArcgisFeatures = /*@__PURE__*/ createReactComponent('arcgis-features');
export const ArcgisFloorFilter = /*@__PURE__*/ createReactComponent('arcgis-floor-filter');
export const ArcgisFullscreen = /*@__PURE__*/ createReactComponent('arcgis-fullscreen');
export const ArcgisHistogram = /*@__PURE__*/ createReactComponent('arcgis-histogram');
export const ArcgisHistogramRangeSlider = /*@__PURE__*/ createReactComponent('arcgis-histogram-range-slider');
export const ArcgisHome = /*@__PURE__*/ createReactComponent('arcgis-home');
export const ArcgisLayerList = /*@__PURE__*/ createReactComponent('arcgis-layer-list');
export const ArcgisLegend = /*@__PURE__*/ createReactComponent('arcgis-legend');
export const ArcgisLineOfSight = /*@__PURE__*/ createReactComponent('arcgis-line-of-sight');
export const ArcgisLocate = /*@__PURE__*/ createReactComponent('arcgis-locate');
export const ArcgisMap = /*@__PURE__*/ createReactComponent('arcgis-map');
export const ArcgisMeasurement = /*@__PURE__*/ createReactComponent('arcgis-measurement');
export const ArcgisNavigationToggle = /*@__PURE__*/ createReactComponent('arcgis-navigation-toggle');
export const ArcgisOrientedImageryViewer = /*@__PURE__*/ createReactComponent('arcgis-oriented-imagery-viewer');
export const ArcgisPlacement = /*@__PURE__*/ createReactComponent('arcgis-placement');
export const ArcgisPrint = /*@__PURE__*/ createReactComponent('arcgis-print');
export const ArcgisScaleBar = /*@__PURE__*/ createReactComponent('arcgis-scale-bar');
export const ArcgisScaleRangeSlider = /*@__PURE__*/ createReactComponent('arcgis-scale-range-slider');
export const ArcgisScene = /*@__PURE__*/ createReactComponent('arcgis-scene');
export const ArcgisSearch = /*@__PURE__*/ createReactComponent('arcgis-search');
export const ArcgisShadowCast = /*@__PURE__*/ createReactComponent('arcgis-shadow-cast');
export const ArcgisSketch = /*@__PURE__*/ createReactComponent('arcgis-sketch');
export const ArcgisSlice = /*@__PURE__*/ createReactComponent('arcgis-slice');
export const ArcgisSlider = /*@__PURE__*/ createReactComponent('arcgis-slider');
export const ArcgisSwipe = /*@__PURE__*/ createReactComponent('arcgis-swipe');
export const ArcgisTableList = /*@__PURE__*/ createReactComponent('arcgis-table-list');
export const ArcgisTimeSlider = /*@__PURE__*/ createReactComponent('arcgis-time-slider');
export const ArcgisTrack = /*@__PURE__*/ createReactComponent('arcgis-track');
export const ArcgisUtilityNetworkAssociations = /*@__PURE__*/ createReactComponent('arcgis-utility-network-associations');
export const ArcgisUtilityNetworkTrace = /*@__PURE__*/ createReactComponent('arcgis-utility-network-trace');
export const ArcgisUtilityNetworkValidateTopology = /*@__PURE__*/ createReactComponent('arcgis-utility-network-validate-topology');
export const ArcgisValuePicker = /*@__PURE__*/ createReactComponent('arcgis-value-picker');
export const ArcgisVersionManagement = /*@__PURE__*/ createReactComponent('arcgis-version-management');
export const ArcgisWeather = /*@__PURE__*/ createReactComponent('arcgis-weather');
export const ArcgisZoom = /*@__PURE__*/ createReactComponent('arcgis-zoom');
