// @arcgis/map-components/dist/components/arcgis-slice.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { ab as newWidgetsSlice } from './index2.js';
import { u as usePropertyChange } from './index3.js';
import { m as makeWidgetController } from './useWidget.js';
import { u as useControllerManager } from './index4.js';

const sliceCss = "arcgis-slice{}";
const ArcgisSliceStyle0 = sliceCss;

const useSliceWidget = makeWidgetController(newWidgetsSlice);
const __cmpMeta = [0,"arcgis-slice",{"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"referenceElement":[1025,"reference-element"],"analysis":[1040],"excludedLayers":[1040],"excludeGroundSurface":[1028,"exclude-ground-surface"],"icon":[1025],"label":[1025],"headingLevel":[1026,"heading-level"],"shape":[1040],"tiltEnabled":[1028,"tilt-enabled"],"position":[1025],"state":[1537],"clear":[64],"destroy":[64],"start":[64]}];
const ArcgisSlice$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisSlice extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.widget = useSliceWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("analysis", "state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.analysis = this.widget.analysis;
        this.excludedLayers = this.viewModel.excludedLayers;
        this.excludeGroundSurface = this.viewModel.excludeGroundSurface;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.headingLevel = this.widget.headingLevel;
        this.shape = this.viewModel.shape;
        this.tiltEnabled = this.viewModel.tiltEnabled;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    async clear() {
        this.widget?.viewModel.clear();
    }
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async start() {
        this.widget?.viewModel.start();
    }
    get el() { return this; }
    static get style() { return ArcgisSliceStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-slice"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-slice":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisSlice$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisSlice = ArcgisSlice$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisSlice, defineCustomElement };
