// @arcgis/map-components/dist/components/arcgis-navigation-toggle.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { Z as newWidgetsNavigationToggle } from './index2.js';
import { u as usePropertyChange } from './index3.js';
import { m as makeWidgetController } from './useWidget.js';
import { u as useControllerManager } from './index4.js';

const navigationToggleCss = "arcgis-navigation-toggle{}";
const ArcgisNavigationToggleStyle0 = navigationToggleCss;

const useNavigationToggleWidget = makeWidgetController(newWidgetsNavigationToggle);
const __cmpMeta = [0,"arcgis-navigation-toggle",{"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"icon":[1025],"label":[1025],"referenceElement":[1025,"reference-element"],"layout":[1537],"position":[1025],"state":[1537],"destroy":[64],"toggle":[64]}];
const ArcgisNavigationToggle$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisNavigationToggle extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.widget = useNavigationToggleWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("layout", "state");
        this.autoDestroyDisabled = false;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.referenceElement = undefined;
        this.layout = this.widget.layout;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Watchers
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async toggle() {
        this.widget?.toggle();
    }
    get el() { return this; }
    static get style() { return ArcgisNavigationToggleStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-navigation-toggle"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-navigation-toggle":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisNavigationToggle$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisNavigationToggle = ArcgisNavigationToggle$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisNavigationToggle, defineCustomElement };
