// @arcgis/map-components/dist/components/arcgis-print.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { a0 as newWidgetsPrint } from './index2.js';
import { u as usePropertyChange } from './index3.js';
import { m as makeWidgetController } from './useWidget.js';
import { u as useControllerManager, g as getSet } from './index4.js';

const printCss = "";
const ArcgisPrintStyle0 = printCss;

const usePrintWidget = makeWidgetController(newWidgetsPrint);
const __cmpMeta = [0,"arcgis-print",{"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"referenceElement":[1025,"reference-element"],"allowedFormats":[1025,"allowed-formats"],"allowedLayouts":[1025,"allowed-layouts"],"exportedLinks":[1040],"extraParameters":[1032,"extra-parameters"],"icon":[1025],"label":[1025],"headingLevel":[1026,"heading-level"],"includeDefaultTemplatesDisabled":[1028,"include-default-templates-disabled"],"portal":[1040],"printServiceUrl":[1025,"print-service-url"],"showPrintAreaEnabled":[1028,"show-print-area-enabled"],"templateCustomTextElements":[1040],"templateOptions":[1040],"position":[1025],"state":[1537],"destroy":[64]}];
const ArcgisPrint$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisPrint extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.widgetComplete = createEvent(this, "arcgisComplete", 7);
        this.widgetSubmit = createEvent(this, "arcgisSubmit", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.widget = usePrintWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.allowedFormats = getSet(this.widget.allowedFormats, {
            set: (value) => (typeof value === "string" && value !== "all" ? value.split(",") : value),
        });
        this.allowedLayouts = getSet(this.widget.allowedLayouts, {
            set: (value) => (typeof value === "string" && value !== "all" ? value.split(",") : value),
        });
        this.exportedLinks = this.widget.exportedLinks;
        this.extraParameters = this.widget.extraParameters;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.headingLevel = this.widget.headingLevel;
        this.includeDefaultTemplatesDisabled = this.widget.includeDefaultTemplates;
        this.portal = this.widget.portal;
        this.printServiceUrl = this.widget.printServiceUrl;
        this.showPrintAreaEnabled = this.widget.showPrintAreaEnabled;
        this.templateCustomTextElements = this.widget.templateCustomTextElements;
        this.templateOptions = this.widget.templateOptions;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return this; }
    static get style() { return ArcgisPrintStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-print"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-print":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisPrint$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisPrint = ArcgisPrint$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisPrint, defineCustomElement };
