// @arcgis/map-components/dist/components/arcgis-placement.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { proxyCustomElement, HTMLElement, createEvent, h } from '@stencil/core/internal/client';
import { r as referenceElementHelper, d as defineViewProperty, a as attachToParent } from './component-utils.js';

const placementCss = "arcgis-placement{display:block;pointer-events:auto;box-shadow:none}.arcgis-placement__container{display:flex}";
const ArcgisPlacementStyle0 = placementCss;

const ArcgisPlacement$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisPlacement extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this._setContainer = (element) => {
            this.el.childElem = element;
        };
        this.referenceElement = undefined;
        this.position = "bottom-left";
    }
    referenceElementWatcher(newValue) {
        void referenceElementHelper(this, newValue);
    }
    positionWatcher(newValue) {
        if (this.el.childElem && newValue !== undefined) {
            this.el.view?.ui.move(this.el.childElem, newValue);
        }
    }
    //--------------------------------------------------------------------------
    //
    //  Lifecycle
    //
    //--------------------------------------------------------------------------
    async componentDidLoad() {
        defineViewProperty(this);
        attachToParent(this);
    }
    disconnectedCallback() {
        if (this.el.childElem) {
            this.el.view?.ui?.remove(this.el.childElem);
        }
    }
    render() {
        return (h("div", { key: '71b98c7f94af271e6881f46e4036085576e1fbd4', ref: this._setContainer, class: "arcgis-placement__container" }, h("slot", { key: 'ac18bc6a5a4ae5a8642d73b7b692c6180a3c13ec' })));
    }
    get el() { return this; }
    static get watchers() { return {
        "referenceElement": ["referenceElementWatcher"],
        "position": ["positionWatcher"]
    }; }
    static get style() { return ArcgisPlacementStyle0; }
}, [4, "arcgis-placement", {
        "referenceElement": [1, "reference-element"],
        "position": [1]
    }, undefined, {
        "referenceElement": ["referenceElementWatcher"],
        "position": ["positionWatcher"]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-placement"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-placement":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisPlacement$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisPlacement = ArcgisPlacement$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisPlacement, defineCustomElement };
