// @arcgis/map-components/dist/components/arcgis-layer-list.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { S as newWidgetsLayerList } from './index2.js';
import { u as usePropertyChange } from './index3.js';
import { m as makeWidgetController } from './useWidget.js';
import { u as useControllerManager, g as getSet } from './index4.js';

const layerListCss = "";
const ArcgisLayerListStyle0 = layerListCss;

const useLayerListWidget = makeWidgetController(newWidgetsLayerList);
const __cmpMeta = [0,"arcgis-layer-list",{"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"referenceElement":[1025,"reference-element"],"collapsed":[1540],"headingLevel":[1026,"heading-level"],"catalogLayerList":[1040],"catalogOptions":[1040],"dragEnabled":[1028,"drag-enabled"],"filterPlaceholder":[1025,"filter-placeholder"],"filterText":[1025,"filter-text"],"icon":[1025],"knowledgeGraphOptions":[1040],"tableList":[1040],"label":[1025],"listItemCreatedFunction":[1040],"minDragEnabledItems":[1026,"min-drag-enabled-items"],"minFilterItems":[1026,"min-filter-items"],"operationalItems":[1040],"selectedItems":[1040],"selectionMode":[1025,"selection-mode"],"visibilityAppearance":[1025,"visibility-appearance"],"hideCatalogLayerList":[1028,"hide-catalog-layer-list"],"showCloseButton":[1028,"show-close-button"],"showCollapseButton":[1028,"show-collapse-button"],"showErrors":[1028,"show-errors"],"showFilter":[1028,"show-filter"],"hideFlow":[1028,"hide-flow"],"showHeading":[1028,"show-heading"],"hideStatusIndicators":[1028,"hide-status-indicators"],"showTemporaryLayerIndicators":[1028,"show-temporary-layer-indicators"],"position":[1025],"state":[1537],"destroy":[64],"triggerAction":[64]}];
const ArcgisLayerList$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisLayerList extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.widgetTriggerAction = createEvent(this, "arcgisTriggerAction", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.widget = useLayerListWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.collapsed = this.widget.collapsed;
        this.headingLevel = this.widget.headingLevel;
        this.catalogLayerList = getSet(undefined, {
            get: () => this.widget.catalogLayerList,
            set: "ignore",
        });
        this.catalogOptions = this.widget.catalogOptions;
        this.dragEnabled = this.widget.dragEnabled;
        this.filterPlaceholder = this.widget.filterPlaceholder;
        this.filterText = this.widget.filterText;
        this.icon = this.widget.icon;
        this.knowledgeGraphOptions = this.widget.knowledgeGraphOptions;
        this.tableList = getSet(undefined, {
            get: () => this.widget.tableList,
            set: "ignore",
        });
        this.label = this.widget.label;
        this.listItemCreatedFunction = this.widget.listItemCreatedFunction;
        this.minDragEnabledItems = this.widget.minDragEnabledItems;
        this.minFilterItems = this.widget.minFilterItems;
        this.operationalItems = getSet(undefined, {
            get: () => this.widget.operationalItems,
            set: "ignore",
        });
        this.selectedItems = this.widget.selectedItems;
        this.selectionMode = this.widget.selectionMode;
        this.visibilityAppearance = this.widget.visibilityAppearance;
        this.hideCatalogLayerList = this.widget.visibleElements.catalogLayerList;
        this.showCloseButton = this.widget.visibleElements.closeButton;
        this.showCollapseButton = this.widget.visibleElements.collapseButton;
        this.showErrors = this.widget.visibleElements.errors;
        this.showFilter = this.widget.visibleElements.filter;
        this.hideFlow = this.widget.visibleElements.flow;
        this.showHeading = this.widget.visibleElements.heading;
        this.hideStatusIndicators = this.widget.visibleElements.statusIndicators;
        this.showTemporaryLayerIndicators = this.widget.visibleElements.temporaryLayerIndicators;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async triggerAction(action, item) {
        this.widget?.triggerAction(action, item);
    }
    get el() { return this; }
    static get style() { return ArcgisLayerListStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-layer-list"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-layer-list":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisLayerList$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisLayerList = ArcgisLayerList$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisLayerList, defineCustomElement };
