// @arcgis/map-components/dist/components/arcgis-basemap-gallery.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { l as newWidgetsBasemapGallery } from './index2.js';
import { m as makeWidgetController } from './useWidget.js';
import { u as usePropertyChange } from './index3.js';
import { u as useControllerManager } from './index4.js';

const basemapGalleryCss = "";
const ArcgisBasemapGalleryStyle0 = basemapGalleryCss;

const useBasemapGalleryWidget = makeWidgetController(newWidgetsBasemapGallery);
const __cmpMeta = [0,"arcgis-basemap-gallery",{"activeBasemap":[1040],"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"disabled":[1540],"icon":[1025],"label":[1025],"headingLevel":[1026,"heading-level"],"source":[1040],"position":[1025],"referenceElement":[1025,"reference-element"],"state":[1537],"destroy":[64]}];
const ArcgisBasemapGallery$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisBasemapGallery extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.widget = useBasemapGalleryWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("activeBasemap", "state");
        this.activeBasemap = this.widget.activeBasemap;
        this.autoDestroyDisabled = false;
        this.disabled = this.widget.disabled;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.headingLevel = this.widget.headingLevel;
        this.source = this.widget.source;
        this.position = "bottom-left";
        this.referenceElement = undefined;
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return this; }
    static get style() { return ArcgisBasemapGalleryStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-basemap-gallery"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-basemap-gallery":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisBasemapGallery$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisBasemapGallery = ArcgisBasemapGallery$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisBasemapGallery, defineCustomElement };
