// @arcgis/map-components/dist/components/arcgis-floor-filter.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { N as newWidgetsFloorFilter } from './index2.js';
import { m as makeWidgetController } from './useWidget.js';
import { u as useControllerManager } from './index4.js';

const floorFilterCss = "";
const ArcgisFloorFilterStyle0 = floorFilterCss;

const useFloorFilterWidget = makeWidgetController(newWidgetsFloorFilter);
const __cmpMeta = [0,"arcgis-floor-filter",{"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"icon":[1025],"label":[1025],"referenceElement":[1025,"reference-element"],"facility":[1537],"headingLevel":[1026,"heading-level"],"level":[1537],"longNames":[1028,"long-names"],"site":[1537],"position":[1025],"destroy":[64],"updateWebDocument":[64]}];
const ArcgisFloorFilter$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisFloorFilter extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.widget = useFloorFilterWidget(this);
        this.viewModel = this.widget.viewModel;
        this.autoDestroyDisabled = false;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.referenceElement = undefined;
        this.facility = this.widget.facility;
        this.headingLevel = this.widget.headingLevel;
        this.level = this.widget.level;
        this.longNames = this.widget.longNames;
        this.site = this.widget.site;
        this.position = "bottom-left";
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async updateWebDocument(webMap) {
        this.widget?.updateWebDocument(webMap);
    }
    get el() { return this; }
    static get style() { return ArcgisFloorFilterStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-floor-filter"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-floor-filter":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisFloorFilter$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisFloorFilter = ArcgisFloorFilter$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisFloorFilter, defineCustomElement };
