// @arcgis/map-components/dist/components/arcgis-distance-measurement-2d.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { C as newWidgetsDistanceMeasurement2D } from './index2.js';
import { u as usePropertyChange } from './index3.js';
import { m as makeWidgetController } from './useWidget.js';
import { u as useControllerManager } from './index4.js';

const distanceMeasurement2dCss = "";
const ArcgisDistanceMeasurement2dStyle0 = distanceMeasurement2dCss;

const useDistanceMeasurement2dWidget = makeWidgetController(newWidgetsDistanceMeasurement2D);
const __cmpMeta = [0,"arcgis-distance-measurement-2d",{"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"referenceElement":[1025,"reference-element"],"icon":[1025],"label":[1025],"unit":[1025],"unitOptions":[1040],"position":[1025],"state":[1537],"destroy":[64]}];
const ArcgisDistanceMeasurement2d$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisDistanceMeasurement2d extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.widget = useDistanceMeasurement2dWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.unit = this.widget.unit;
        this.unitOptions = this.widget.unitOptions;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return this; }
    static get style() { return ArcgisDistanceMeasurement2dStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-distance-measurement-2d"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-distance-measurement-2d":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisDistanceMeasurement2d$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisDistanceMeasurement2d = ArcgisDistanceMeasurement2d$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisDistanceMeasurement2d, defineCustomElement };
