// @arcgis/map-components/dist/components/arcgis-catalog-layer-list.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { u as usePropertyChange } from './index3.js';
import { u as newWidgetsCatalogLayerList } from './index2.js';
import { m as makeWidgetController } from './useWidget.js';
import { u as useControllerManager, g as getSet } from './index4.js';

const catalogLayerListCss = "";
const ArcgisCatalogLayerListStyle0 = catalogLayerListCss;

const useCatalogLayerListWidget = makeWidgetController(newWidgetsCatalogLayerList);
const __cmpMeta = [0,"arcgis-catalog-layer-list",{"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"headingLevel":[1026,"heading-level"],"label":[1025],"referenceElement":[1025,"reference-element"],"catalogItems":[1040],"catalogLayer":[1040],"collapsed":[1540],"filterPlaceholder":[1025,"filter-placeholder"],"filterText":[1025,"filter-text"],"icon":[1025],"listItemCreatedFunction":[1040],"minFilterItems":[1026,"min-filter-items"],"selectedItems":[1040],"selectionMode":[1025,"selection-mode"],"visibilityAppearance":[1025,"visibility-appearance"],"showCloseButton":[1028,"show-close-button"],"showCollapseButton":[1028,"show-collapse-button"],"showErrors":[1028,"show-errors"],"showFilter":[1028,"show-filter"],"hideFlow":[1028,"hide-flow"],"showHeading":[1028,"show-heading"],"hideStatusIndicators":[1028,"hide-status-indicators"],"showTemporaryLayerIndicators":[1028,"show-temporary-layer-indicators"],"position":[1025],"state":[1537],"destroy":[64]}];
const ArcgisCatalogLayerList$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisCatalogLayerList extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.widgetTriggerAction = createEvent(this, "arcgisTriggerAction", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.widget = useCatalogLayerListWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.headingLevel = this.widget.headingLevel;
        this.label = this.widget.label;
        this.referenceElement = undefined;
        this.catalogItems = getSet(undefined, {
            get: () => this.widget.catalogItems,
            set: "ignore",
        });
        this.catalogLayer = this.widget.catalogLayer;
        this.collapsed = this.widget.collapsed;
        this.filterPlaceholder = this.widget.filterPlaceholder;
        this.filterText = this.widget.filterText;
        this.icon = this.widget.icon;
        this.listItemCreatedFunction = this.widget.listItemCreatedFunction;
        this.minFilterItems = this.widget.minFilterItems;
        this.selectedItems = this.widget.selectedItems;
        this.selectionMode = this.widget.selectionMode;
        this.visibilityAppearance = this.widget.visibilityAppearance;
        this.showCloseButton = this.widget.visibleElements.closeButton;
        this.showCollapseButton = this.widget.visibleElements.collapseButton;
        this.showErrors = this.widget.visibleElements.errors;
        this.showFilter = this.widget.visibleElements.filter;
        this.hideFlow = this.widget.visibleElements.flow;
        this.showHeading = this.widget.visibleElements.heading;
        this.hideStatusIndicators = this.widget.visibleElements.statusIndicators;
        this.showTemporaryLayerIndicators = this.widget.visibleElements.temporaryLayerIndicators;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return this; }
    static get style() { return ArcgisCatalogLayerListStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-catalog-layer-list"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-catalog-layer-list":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisCatalogLayerList$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisCatalogLayerList = ArcgisCatalogLayerList$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisCatalogLayerList, defineCustomElement };
