// @arcgis/map-components/dist/components/arcgis-daylight.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { x as newWidgetsDaylight } from './index2.js';
import { u as useControllerManager } from './index4.js';
import { m as makeWidgetController } from './useWidget.js';

const daylightCss = "";
const ArcgisDaylightStyle0 = daylightCss;

const useDaylightWidget = makeWidgetController(newWidgetsDaylight);
const __cmpMeta = [0,"arcgis-daylight",{"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"referenceElement":[1025,"reference-element"],"currentSeason":[1025,"current-season"],"dayPlaying":[1028,"day-playing"],"yearPlaying":[1028,"year-playing"],"dateOrSeason":[1025,"date-or-season"],"icon":[1025],"label":[1025],"headingLevel":[1026,"heading-level"],"playSpeedMultiplier":[1026,"play-speed-multiplier"],"timeSliderSteps":[1026,"time-slider-steps"],"hideHeader":[1028,"hide-header"],"hidePlayButtons":[1028,"hide-play-buttons"],"hideDatePicker":[1028,"hide-date-picker"],"hideTimezone":[1028,"hide-timezone"],"hideSunLightingToggle":[1028,"hide-sun-lighting-toggle"],"hideShadowsToggle":[1028,"hide-shadows-toggle"],"position":[1025],"destroy":[64]}];
const ArcgisDaylight$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisDaylight extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.widget = useDaylightWidget(this);
        this.viewModel = this.widget.viewModel;
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.currentSeason = this.viewModel.currentSeason;
        this.dayPlaying = this.viewModel.dayPlaying;
        this.yearPlaying = this.viewModel.yearPlaying;
        this.dateOrSeason = this.widget.dateOrSeason;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.headingLevel = this.widget.headingLevel;
        this.playSpeedMultiplier = this.widget.playSpeedMultiplier;
        this.timeSliderSteps = this.widget.timeSliderSteps;
        this.hideHeader = this.widget.visibleElements.header;
        this.hidePlayButtons = this.widget.visibleElements.playButtons;
        this.hideDatePicker = this.widget.visibleElements.datePicker;
        this.hideTimezone = this.widget.visibleElements.timezone;
        this.hideSunLightingToggle = this.widget.visibleElements.sunLightingToggle;
        this.hideShadowsToggle = this.widget.visibleElements.shadowsToggle;
        this.position = "bottom-left";
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return this; }
    static get style() { return ArcgisDaylightStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-daylight"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-daylight":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisDaylight$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisDaylight = ArcgisDaylight$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisDaylight, defineCustomElement };
