// @arcgis/map-components/dist/components/arcgis-scale-range-slider.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { a4 as newWidgetsScaleRangeSlider } from './index2.js';
import { u as usePropertyChange } from './index3.js';
import { m as makeWidgetController } from './useWidget.js';
import { u as useControllerManager } from './index4.js';

const scaleRangeSliderCss = "arcgis-scale-range-slider{}";
const ArcgisScaleRangeSliderStyle0 = scaleRangeSliderCss;

const useScaleRangeSliderWidget = makeWidgetController(newWidgetsScaleRangeSlider);
const __cmpMeta = [0,"arcgis-scale-range-slider",{"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"referenceElement":[1025,"reference-element"],"disabled":[1540],"icon":[1025],"label":[1025],"layer":[1040],"maxScale":[1026,"max-scale"],"maxScaleLimit":[1026,"max-scale-limit"],"minScale":[1026,"min-scale"],"minScaleLimit":[1026,"min-scale-limit"],"mode":[1025],"region":[1025],"hidePreview":[1028,"hide-preview"],"hideScaleMenusMinScaleMenu":[1028,"hide-scale-menus-min-scale-menu"],"hideScaleMenusMaxScaleMenu":[1028,"hide-scale-menus-max-scale-menu"],"position":[1025],"state":[1537],"destroy":[64]}];
const ArcgisScaleRangeSlider$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisScaleRangeSlider extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.widget = useScaleRangeSliderWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.disabled = this.widget.disabled;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.layer = this.widget.layer;
        this.maxScale = this.widget.maxScale;
        this.maxScaleLimit = this.widget.maxScaleLimit;
        this.minScale = this.widget.minScale;
        this.minScaleLimit = this.widget.minScaleLimit;
        this.mode = this.widget.mode;
        this.region = this.widget.region;
        this.hidePreview = this.widget.visibleElements.preview;
        this.hideScaleMenusMinScaleMenu = this.widget.visibleElements.scaleMenus
            .minScaleMenu;
        this.hideScaleMenusMaxScaleMenu = this.widget.visibleElements.scaleMenus
            .maxScaleMenu;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return this; }
    static get style() { return ArcgisScaleRangeSliderStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-scale-range-slider"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-scale-range-slider":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisScaleRangeSlider$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisScaleRangeSlider = ArcgisScaleRangeSlider$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisScaleRangeSlider, defineCustomElement };
