// @arcgis/map-components/dist/components/arcgis-feature-templates.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { L as newWidgetsFeatureTemplates } from './index2.js';
import { u as usePropertyChange } from './index3.js';
import { m as makeWidgetController } from './useWidget.js';
import { u as useControllerManager } from './index4.js';

const featureTemplatesCss = "";
const ArcgisFeatureTemplatesStyle0 = featureTemplatesCss;

const useFeatureTemplatesWidget = makeWidgetController(newWidgetsFeatureTemplates);
const __cmpMeta = [0,"arcgis-feature-templates",{"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"referenceElement":[1025,"reference-element"],"position":[1025],"icon":[1025],"label":[1025],"headingLevel":[1026,"heading-level"],"enableListScroll":[1028,"enable-list-scroll"],"filterFunction":[1040],"filterText":[1025,"filter-text"],"groupBy":[1025,"group-by"],"layers":[1040],"selectionMode":[1025,"selection-mode"],"hideFilter":[1028,"hide-filter"],"state":[1537],"destroy":[64]}];
const ArcgisFeatureTemplates$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisFeatureTemplates extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.widgetSelect = createEvent(this, "arcgisSelect", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.widget = useFeatureTemplatesWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.position = "bottom-left";
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.headingLevel = this.widget.headingLevel;
        this.enableListScroll = this.widget.enableListScroll;
        this.filterFunction = this.widget.filterFunction;
        this.filterText = this.widget.filterText;
        this.groupBy = this.widget.groupBy;
        this.layers = this.widget.layers;
        this.selectionMode = this.widget.selectionMode;
        this.hideFilter = this.widget.visibleElements.filter;
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return this; }
    static get style() { return ArcgisFeatureTemplatesStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-feature-templates"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-feature-templates":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisFeatureTemplates$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisFeatureTemplates = ArcgisFeatureTemplates$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisFeatureTemplates, defineCustomElement };
