// @arcgis/map-components/dist/components/arcgis-measurement.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { Y as newWidgetsMeasurement } from './index2.js';
import { u as usePropertyChange } from './index3.js';
import { m as makeWidgetController } from './useWidget.js';
import { u as useControllerManager } from './index4.js';

const measurementCss = "arcgis-measurement{}";
const ArcgisMeasurementStyle0 = measurementCss;

const useMeasurementWidget = makeWidgetController(newWidgetsMeasurement);
const __cmpMeta = [0,"arcgis-measurement",{"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"referenceElement":[1025,"reference-element"],"activeTool":[1025,"active-tool"],"activeWidget":[1040],"areaUnit":[1025,"area-unit"],"icon":[1025],"label":[1025],"linearUnit":[1025,"linear-unit"],"position":[1025],"state":[1537],"clear":[64],"destroy":[64],"startMeasurement":[64]}];
const ArcgisMeasurement$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisMeasurement extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.widget = useMeasurementWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("activeTool", "state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.activeTool = this.widget.activeTool;
        this.activeWidget = this.widget.activeWidget;
        this.areaUnit = this.widget.areaUnit;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.linearUnit = this.widget.linearUnit;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    async clear() {
        this.widget?.clear();
    }
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async startMeasurement() {
        this.widget?.startMeasurement();
    }
    get el() { return this; }
    static get style() { return ArcgisMeasurementStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-measurement"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-measurement":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisMeasurement$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisMeasurement = ArcgisMeasurement$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisMeasurement, defineCustomElement };
