// @arcgis/map-components/dist/components/arcgis-attachments.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { k as newWidgetsAttachments } from './index2.js';
import { m as makeWidgetController } from './useWidget.js';
import { u as usePropertyChange } from './index3.js';
import { u as useControllerManager } from './index4.js';

const attachmentsCss = "";
const ArcgisAttachmentsStyle0 = attachmentsCss;

const useAttachmentsWidget = makeWidgetController(newWidgetsAttachments);
const __cmpMeta = [0,"arcgis-attachments",{"referenceElement":[1025,"reference-element"],"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"icon":[1025],"label":[1025],"position":[1025],"capabilities":[1040],"displayType":[1025,"display-type"],"graphic":[1040],"submitting":[1028],"hideAddButton":[1028,"hide-add-button"],"hideAddSubmitButton":[1028,"hide-add-submit-button"],"hideCancelAddButton":[1028,"hide-cancel-add-button"],"hideCancelUpdateButton":[1028,"hide-cancel-update-button"],"hideDeleteButton":[1028,"hide-delete-button"],"hideErrorMessage":[1028,"hide-error-message"],"hideProgressBar":[1028,"hide-progress-bar"],"hideUpdateButton":[1028,"hide-update-button"],"state":[1537],"destroy":[64]}];
const ArcgisAttachments$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisAttachments extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.widget = useAttachmentsWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("submitting", "state");
        this.referenceElement = undefined;
        this.autoDestroyDisabled = false;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.position = "bottom-left";
        this.capabilities = this.widget.capabilities;
        this.displayType = this.widget.displayType;
        this.graphic = this.widget.graphic;
        this.submitting = this.widget.submitting;
        this.hideAddButton = this.widget.visibleElements.addButton;
        this.hideAddSubmitButton = this.widget.visibleElements.addSubmitButton;
        this.hideCancelAddButton = this.widget.visibleElements.cancelAddButton;
        this.hideCancelUpdateButton = this.widget.visibleElements.cancelUpdateButton;
        this.hideDeleteButton = this.widget.visibleElements.deleteButton;
        this.hideErrorMessage = this.widget.visibleElements.errorMessage;
        this.hideProgressBar = this.widget.visibleElements.progressBar;
        this.hideUpdateButton = this.widget.visibleElements.updateButton;
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return this; }
    static get style() { return ArcgisAttachmentsStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-attachments"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-attachments":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisAttachments$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisAttachments = ArcgisAttachments$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisAttachments, defineCustomElement };
