// @arcgis/map-components/dist/components/arcgis-swipe.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { ad as newWidgetsSwipe } from './index2.js';
import { u as usePropertyChange } from './index3.js';
import { m as makeWidgetController } from './useWidget.js';
import { u as useControllerManager } from './index4.js';

const swipeCss = "arcgis-swipe{}";
const ArcgisSwipeStyle0 = swipeCss;

const useSwipeWidget = makeWidgetController(newWidgetsSwipe);
const __cmpMeta = [0,"arcgis-swipe",{"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"referenceElement":[1025,"reference-element"],"icon":[1025],"label":[1025],"direction":[1025],"dragLabel":[1025,"drag-label"],"leadingLayers":[1040],"swipePosition":[1026,"swipe-position"],"trailingLayers":[1040],"position":[1025],"state":[1537],"destroy":[64]}];
const ArcgisSwipe$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisSwipe extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.widget = useSwipeWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.direction = this.widget.direction;
        this.dragLabel = this.widget.dragLabel;
        this.leadingLayers = this.widget.leadingLayers;
        this.swipePosition = this.widget.position;
        this.trailingLayers = this.widget.trailingLayers;
        this.position = "manual";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return this; }
    static get style() { return ArcgisSwipeStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-swipe"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-swipe":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisSwipe$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisSwipe = ArcgisSwipe$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisSwipe, defineCustomElement };
