// @arcgis/map-components/dist/components/arcgis-building-explorer.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { t as newWidgetsBuildingExplorer } from './index2.js';
import { m as makeWidgetController } from './useWidget.js';
import { u as usePropertyChange } from './index3.js';
import { u as useControllerManager } from './index4.js';

const buildingExplorerCss = "";
const ArcgisBuildingExplorerStyle0 = buildingExplorerCss;

const useBuildingExplorerWidget = makeWidgetController(newWidgetsBuildingExplorer);
const __cmpMeta = [0,"arcgis-building-explorer",{"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"icon":[1025],"label":[1025],"layers":[1040],"headingLevel":[1026,"heading-level"],"level":[1040],"phase":[1040],"state":[1537],"hideLevels":[1028,"hide-levels"],"hidePhases":[1028,"hide-phases"],"hideDisciplines":[1028,"hide-disciplines"],"position":[1025],"referenceElement":[1025,"reference-element"],"destroy":[64]}];
const ArcgisBuildingExplorer$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisBuildingExplorer extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.widget = useBuildingExplorerWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("level", "phase", "state");
        this.autoDestroyDisabled = false;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.layers = this.widget.layers;
        this.headingLevel = this.widget.headingLevel;
        this.level = this.viewModel.level;
        this.phase = this.viewModel.phase;
        this.state = this.viewModel.state;
        this.hideLevels = this.widget.visibleElements.levels;
        this.hidePhases = this.widget.visibleElements.phases;
        this.hideDisciplines = this.widget.visibleElements.disciplines;
        this.position = "bottom-left";
        this.referenceElement = undefined;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return this; }
    static get style() { return ArcgisBuildingExplorerStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-building-explorer"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-building-explorer":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisBuildingExplorer$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisBuildingExplorer = ArcgisBuildingExplorer$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisBuildingExplorer, defineCustomElement };
