// @arcgis/map-components/dist/components/arcgis-line-of-sight.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { U as newWidgetsLineOfSight } from './index2.js';
import { u as usePropertyChange } from './index3.js';
import { m as makeWidgetController } from './useWidget.js';
import { u as useControllerManager } from './index4.js';

const lineOfSightCss = "arcgis-line-of-sight{}";
const ArcgisLineOfSightStyle0 = lineOfSightCss;

const useLineOfSightWidget = makeWidgetController(newWidgetsLineOfSight);
const __cmpMeta = [0,"arcgis-line-of-sight",{"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"referenceElement":[1025,"reference-element"],"analysis":[1040],"icon":[1025],"label":[1025],"position":[1025],"state":[1537],"destroy":[64],"clear":[64],"continue":[64],"start":[64],"stop":[64]}];
const ArcgisLineOfSight$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisLineOfSight extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.widget = useLineOfSightWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("analysis", "state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.analysis = this.widget.analysis;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async clear() {
        this.widget?.viewModel.clear();
    }
    async continue() {
        this.widget?.viewModel.continue();
    }
    async start() {
        this.widget?.viewModel.start();
    }
    async stop() {
        this.widget?.viewModel.stop();
    }
    get el() { return this; }
    static get style() { return ArcgisLineOfSightStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-line-of-sight"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-line-of-sight":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisLineOfSight$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisLineOfSight = ArcgisLineOfSight$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisLineOfSight, defineCustomElement };
