// @arcgis/map-components/dist/components/arcgis-track.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate, h } from '@stencil/core/internal/client';
import { g as globalCss } from './globalCss.js';
import { u as useT9n, m as makeViewModelController } from './useViewModel.js';
import { p as classes, ag as newWidgetsTrackTrackViewModel } from './index2.js';
import { u as usePropertyChange } from './index3.js';
import { u as useControllerManager } from './index4.js';
import "@esri/calcite-components/dist/components/calcite-button.js";
import "@esri/calcite-components/dist/components/calcite-tooltip.js";

const css = {
    base: "esri-track",
};
const useTrackViewModel = makeViewModelController(newWidgetsTrackTrackViewModel);
const __cmpMeta = [0,"arcgis-track",{"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"error":[1032],"geolocationOptions":[1032,"geolocation-options"],"goToLocationDisabled":[1540,"go-to-location-disabled"],"graphic":[1040],"icon":[1537],"label":[1025],"positionFilterFunction":[1040],"position":[1537],"referenceElement":[1025,"reference-element"],"scale":[1026],"state":[1025],"tracking":[1028],"destroy":[64],"start":[64],"stop":[64]}];
const ArcgisTrack$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisTrack extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.viewModelTrack = createEvent(this, "arcgisComplete", 7);
        this.viewModelTrackError = createEvent(this, "arcgisError", 7);
        //--------------------------------------------------------------------------
        //
        //  Controllers
        //
        //--------------------------------------------------------------------------
        this.manager = useControllerManager(this,forceUpdate);
        this.messages = useT9n();
        this.viewModel = useTrackViewModel(this);
        //--------------------------------------------------------------------------
        //
        //  Events
        //
        //--------------------------------------------------------------------------
        this.arcgisPropertyChange = usePropertyChange()("state");
        //--------------------------------------------------------------------------
        //
        //  Private Methods
        //
        //--------------------------------------------------------------------------
        this._toggleTracking = () => {
            if (!this.viewModel) {
                return;
            }
            const { state } = this.viewModel;
            if (state === "feature-unsupported" ||
                state === "disabled") {
                return;
            }
            if (state === "tracking" || state === "waiting") {
                this.viewModel.stop();
                return;
            }
            this.viewModel.start();
        };
        this.autoDestroyDisabled = false;
        this.error = this.viewModel.error;
        this.geolocationOptions = this.viewModel.geolocationOptions;
        this.goToLocationDisabled = this.viewModel.goToLocationEnabled;
        this.graphic = this.viewModel.graphic;
        this.icon = "compass-north-circle";
        this.label = undefined;
        this.positionFilterFunction = this.viewModel.positionFilterFunction;
        this.position = "top-left";
        this.referenceElement = undefined;
        this.scale = this.viewModel.scale;
        this.state = this.viewModel.state;
        this.tracking = this.viewModel.tracking;
    }
    //--------------------------------------------------------------------------
    //
    //  Public Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    render() {
        const { viewModel, messages, icon } = this;
        const state = viewModel?.state;
        const isTracking = state === "tracking";
        const errorText = this._localizeError(this.viewModel?.error);
        const label = (isTracking ? messages?.stopTracking : messages?.startTracking) ?? "";
        const hidden = state === "feature-unsupported";
        const iconStart = (state === "tracking" ? "pause" : undefined) ?? icon ?? undefined;
        return (h("div", { key: '61dd5a2bc84057cb74d3edf796b820e3d4de9d57', class: classes(css.base, globalCss.widget) }, !hidden && (h("calcite-button", { key: '4e8b016417d3bb7609d570026e9bdb76cf8e9793', class: globalCss.widgetButton, disabled: state === "disabled", "icon-start": iconStart, kind: "neutral", label: label, loading: state === "waiting", onClick: this._toggleTracking, title: label })), errorText ? (h("calcite-tooltip", { open: true, "overlay-positioning": "fixed", "reference-element": this.el }, errorText)) : null));
    }
    async start() {
        this.viewModel.start();
    }
    async stop() {
        this.viewModel.stop();
    }
    _localizeError(error) {
        if (error instanceof GeolocationPositionError) {
            if (error.code === GeolocationPositionError.PERMISSION_DENIED) {
                return this.messages?.permissionError;
            }
            else if (error.code === GeolocationPositionError.TIMEOUT) {
                return this.messages?.timeoutError;
            }
            return this.messages?.positionUnavailable;
        }
        if (error instanceof Error) {
            return error.message;
        }
        return undefined;
    }
    static get assetsDirs() { return ["assets"]; }
    get el() { return this; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-track"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-track":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisTrack$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisTrack = ArcgisTrack$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisTrack, defineCustomElement };
