// @arcgis/map-components/dist/components/arcgis-directline-measurement-3d.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { B as newWidgetsDirectLineMeasurement3D } from './index2.js';
import { u as usePropertyChange } from './index3.js';
import { m as makeWidgetController } from './useWidget.js';
import { u as useControllerManager } from './index4.js';

const directlineMeasurement3dCss = "";
const ArcgisDirectlineMeasurement3dStyle0 = directlineMeasurement3dCss;

const useDirectLineMeasurement3dWidget = makeWidgetController(newWidgetsDirectLineMeasurement3D);
const __cmpMeta = [0,"arcgis-directline-measurement-3d",{"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"referenceElement":[1025,"reference-element"],"analysis":[1040],"icon":[1025],"label":[1025],"unit":[1025],"unitOptions":[1040],"position":[1025],"state":[1537],"clear":[64],"destroy":[64],"start":[64]}];
const ArcgisDirectLineMeasurement3d = /*@__PURE__*/ proxyCustomElement(class ArcgisDirectLineMeasurement3d extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.widget = useDirectLineMeasurement3dWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("analysis", "state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.analysis = this.widget.analysis;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.unit = this.widget.unit;
        this.unitOptions = this.widget.unitOptions;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    async clear() {
        this.widget?.viewModel.clear();
    }
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async start() {
        this.widget?.viewModel.start();
    }
    get el() { return this; }
    static get style() { return ArcgisDirectlineMeasurement3dStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-directline-measurement-3d"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-directline-measurement-3d":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisDirectLineMeasurement3d);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisDirectlineMeasurement3d = ArcgisDirectLineMeasurement3d;
const defineCustomElement = defineCustomElement$1;

export { ArcgisDirectlineMeasurement3d, defineCustomElement };
