// @arcgis/map-components/dist/components/arcgis-directional-pad.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { proxyCustomElement, HTMLElement, createEvent, forceUpdate } from '@stencil/core/internal/client';
import { y as newWidgetsDirectionalPad } from './index2.js';
import { u as usePropertyChange } from './index3.js';
import { m as makeWidgetController } from './useWidget.js';
import { u as useControllerManager } from './index4.js';

const directionalPadCss = "arcgis-directional-pad{}";
const ArcgisDirectionalPadStyle0 = directionalPadCss;

const useDirectionalPadWidget = makeWidgetController(newWidgetsDirectionalPad);
const __cmpMeta = [0,"arcgis-directional-pad",{"autoDestroyDisabled":[1028,"auto-destroy-disabled"],"disabled":[1028],"referenceElement":[1025,"reference-element"],"icon":[1025],"label":[1025],"visualScale":[1025,"visual-scale"],"position":[1025],"state":[1537],"destroy":[64]}];
const ArcgisDirectionalPad$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisDirectionalPad extends HTMLElement {
    connectedCallback() { this.manager._connectedCallback(); }
    disconnectedCallback() { this.manager._disconnectedCallback(); }
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this,forceUpdate);
        this.widget = useDirectionalPadWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.disabled = this.widget.disabled;
        this.referenceElement = undefined;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.visualScale = this.widget.visualScale;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return this; }
    static get style() { return ArcgisDirectionalPadStyle0; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-directional-pad"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-directional-pad":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisDirectionalPad$1);
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisDirectionalPad = ArcgisDirectionalPad$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisDirectionalPad, defineCustomElement };
