// @arcgis/map-components/dist/components/version-management-version-item.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { proxyCustomElement, HTMLElement, createEvent, h } from '@stencil/core/internal/client';
import "@esri/calcite-components/dist/components/calcite-action.js";
import "@esri/calcite-components/dist/components/calcite-tooltip.js";
import "@esri/calcite-components/dist/components/calcite-list-item.js";
import "@esri/calcite-components/dist/components/calcite-action-menu.js";

const ArcgisVersionManagementVersionItem = /*@__PURE__*/ proxyCustomElement(class ArcgisVersionManagementVersionItem extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        this.arcgisManageVersion = createEvent(this, "arcgisManageVersion", 7);
        this.versionItemElementProps = undefined;
    }
    render() {
        const { versionItemElementProps: { currentUser, currentVersionIdentifier, hasAdvEditingUte, isVersioningApiAvailable, serviceUrl, strings: { actions }, versionInfo, }, } = this;
        const fullVersionName = versionInfo.versionIdentifier.name;
        const versionName = fullVersionName.substring(fullVersionName.lastIndexOf(".") + 1); // label
        const versionOwner = fullVersionName.substring(0, fullVersionName.lastIndexOf(".")).toUpperCase(); // description
        const guid = versionInfo.versionIdentifier.guid;
        const actionElementId = `actionSwitchToVersion_${guid}`;
        const canEditVersion = versionOwner.toUpperCase() === currentUser.toUpperCase() && isVersioningApiAvailable && hasAdvEditingUte;
        const isCurrentVersion = currentVersionIdentifier.guid === versionInfo.versionIdentifier.guid;
        const startActionChangeVersion = (h("calcite-action", { key: '14f7c8b59b3beb9a160b10a1f7252a9c490dd96b', disabled: isCurrentVersion, id: actionElementId, icon: "code-branch", text: "", onClick: () => {
                this.arcgisManageVersion.emit({ actionType: "changeVersion", serviceUrl, versionInfo });
            }, slot: "actions-start" }));
        const startActionChangeVersionTooltip = !isCurrentVersion ? (h("calcite-tooltip", { "overlay-positioning": "fixed", referenceElement: actionElementId }, h("span", null, actions.switchToVersion))) : undefined;
        return (h("calcite-list-item", { key: 'a031cfd07866803ae8d06f53593288d3f17fa288', description: versionOwner, label: versionName, selected: false, value: guid }, startActionChangeVersion, startActionChangeVersionTooltip, h("calcite-action-menu", { key: 'fb64c8bdc470942f2348073d40c894be4a88fd93', "overlay-positioning": "fixed", slot: "actions-end", label: "" }, h("calcite-action", { key: '1009f3d41a1b36d1d835f4b7fc36159baa4c1320', disabled: isCurrentVersion, icon: "code-branch", onClick: () => {
                this.arcgisManageVersion.emit({ actionType: "changeVersion", serviceUrl, versionInfo });
            }, scale: "s", text: actions.switchToVersion, textEnabled: true }), h("calcite-action", { key: 'a7e7400c15dc6151a8b68075c509c30b2a75fcb6', disabled: !canEditVersion || isCurrentVersion, icon: "code-branch-edit", onClick: () => {
                this.arcgisManageVersion.emit({ actionType: "editVersion", serviceUrl, versionInfo });
            }, scale: "s", text: actions.editVersion, textEnabled: true }), h("calcite-action", { key: 'afafe0b02b45bc233a758ac5ab1f18a7aef77d8d', disabled: !canEditVersion || (canEditVersion && isCurrentVersion), icon: "trash", onClick: () => {
                this.arcgisManageVersion.emit({ actionType: "deleteVersion", serviceUrl, versionInfo });
            }, scale: "s", text: actions.deleteVersion, textEnabled: true }))));
    }
    get el() { return this; }
}, [0, "arcgis-version-management-version-item", {
        "versionItemElementProps": [16]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-version-management-version-item"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-version-management-version-item":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisVersionManagementVersionItem);
            }
            break;
    } });
}
defineCustomElement();

export { ArcgisVersionManagementVersionItem as A, defineCustomElement as d };
