// @arcgis/map-components/dist/cjs/arcgis-scale-range-slider.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-80ecaa0e.js');
const index$3 = require('./index-87ff723d.js');
const index$2 = require('./index-fbb18e12.js');
const useWidget = require('./useWidget-3971aff7.js');
const index$1 = require('./index-bb8f4b21.js');
require('./component-utils-b8b66725.js');

const scaleRangeSliderCss = "arcgis-scale-range-slider{}";
const ArcgisScaleRangeSliderStyle0 = scaleRangeSliderCss;

const useScaleRangeSliderWidget = useWidget.makeWidgetController(index$3.newWidgetsScaleRangeSlider);
const ArcgisScaleRangeSlider = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = useScaleRangeSliderWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = index$2.usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.disabled = this.widget.disabled;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.layer = this.widget.layer;
        this.maxScale = this.widget.maxScale;
        this.maxScaleLimit = this.widget.maxScaleLimit;
        this.minScale = this.widget.minScale;
        this.minScaleLimit = this.widget.minScaleLimit;
        this.mode = this.widget.mode;
        this.region = this.widget.region;
        this.hidePreview = this.widget.visibleElements.preview;
        this.hideScaleMenusMinScaleMenu = this.widget.visibleElements.scaleMenus
            .minScaleMenu;
        this.hideScaleMenusMaxScaleMenu = this.widget.visibleElements.scaleMenus
            .maxScaleMenu;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return index.getElement(this); }
};
ArcgisScaleRangeSlider.style = ArcgisScaleRangeSliderStyle0;

exports.arcgis_scale_range_slider = ArcgisScaleRangeSlider;
