// @arcgis/map-components/dist/cjs/arcgis-directline-measurement-3d.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-80ecaa0e.js');
const index$3 = require('./index-87ff723d.js');
const index$2 = require('./index-fbb18e12.js');
const useWidget = require('./useWidget-3971aff7.js');
const index$1 = require('./index-bb8f4b21.js');
require('./component-utils-b8b66725.js');

const directlineMeasurement3dCss = "";
const ArcgisDirectlineMeasurement3dStyle0 = directlineMeasurement3dCss;

const useDirectLineMeasurement3dWidget = useWidget.makeWidgetController(index$3.newWidgetsDirectLineMeasurement3D);
const ArcgisDirectLineMeasurement3d = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = useDirectLineMeasurement3dWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = index$2.usePropertyChange()("analysis", "state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.analysis = this.widget.analysis;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.unit = this.widget.unit;
        this.unitOptions = this.widget.unitOptions;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    async clear() {
        this.widget?.viewModel.clear();
    }
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async start() {
        this.widget?.viewModel.start();
    }
    get el() { return index.getElement(this); }
};
ArcgisDirectLineMeasurement3d.style = ArcgisDirectlineMeasurement3dStyle0;

exports.arcgis_directline_measurement_3d = ArcgisDirectLineMeasurement3d;
