// @arcgis/map-components/dist/cjs/arcgis-histogram.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-80ecaa0e.js');
const index$3 = require('./index-87ff723d.js');
const useWidget = require('./useWidget-3971aff7.js');
const index$2 = require('./index-fbb18e12.js');
const index$1 = require('./index-bb8f4b21.js');
require('./component-utils-b8b66725.js');

const histogramCss = "arcgis-histogram{}";
const ArcgisHistogramStyle0 = histogramCss;

const useHistogramWidget = useWidget.makeWidgetController(index$3.newWidgetsHistogram);
const ArcgisHistogram = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = useHistogramWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = index$2.usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.average = this.widget.average;
        this.barCreatedFunction = this.widget.barCreatedFunction;
        this.bins = this.widget.bins;
        this.dataLineCreatedFunction = this.widget.dataLineCreatedFunction;
        this.dataLines = this.widget.dataLines;
        this.labelFormatFunction = this.widget.labelFormatFunction;
        this.layout = this.widget.layout;
        this.max = this.widget.max;
        this.min = this.widget.min;
        this.position = "bottom-left";
        this.state = this.widget.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return index.getElement(this); }
};
ArcgisHistogram.style = ArcgisHistogramStyle0;

exports.arcgis_histogram = ArcgisHistogram;
