// @arcgis/map-components/dist/cjs/arcgis-building-explorer.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-80ecaa0e.js');
const index$3 = require('./index-87ff723d.js');
const useWidget = require('./useWidget-3971aff7.js');
const index$2 = require('./index-fbb18e12.js');
const index$1 = require('./index-bb8f4b21.js');
require('./component-utils-b8b66725.js');

const buildingExplorerCss = "";
const ArcgisBuildingExplorerStyle0 = buildingExplorerCss;

const useBuildingExplorerWidget = useWidget.makeWidgetController(index$3.newWidgetsBuildingExplorer);
const ArcgisBuildingExplorer = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = useBuildingExplorerWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = index$2.usePropertyChange()("level", "phase", "state");
        this.autoDestroyDisabled = false;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.layers = this.widget.layers;
        this.headingLevel = this.widget.headingLevel;
        this.level = this.viewModel.level;
        this.phase = this.viewModel.phase;
        this.state = this.viewModel.state;
        this.hideLevels = this.widget.visibleElements.levels;
        this.hidePhases = this.widget.visibleElements.phases;
        this.hideDisciplines = this.widget.visibleElements.disciplines;
        this.position = "bottom-left";
        this.referenceElement = undefined;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return index.getElement(this); }
};
ArcgisBuildingExplorer.style = ArcgisBuildingExplorerStyle0;

exports.arcgis_building_explorer = ArcgisBuildingExplorer;
