// @arcgis/map-components/dist/cjs/arcgis-area-measurement-2d.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-80ecaa0e.js');
const index$3 = require('./index-87ff723d.js');
const index$2 = require('./index-fbb18e12.js');
const useWidget = require('./useWidget-3971aff7.js');
const index$1 = require('./index-bb8f4b21.js');
require('./component-utils-b8b66725.js');

const areaMeasurement2dCss = "";
const ArcgisAreaMeasurement2dStyle0 = areaMeasurement2dCss;

const useAreaMeasurement2dWidget = useWidget.makeWidgetController(
// Workaround for https://devtopia.esri.com/WebGIS/arcgis-js-api/issues/61559
index$3.newWidgetsAreaMeasurement2D);
const ArcgisAreaMeasurement2d = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = useAreaMeasurement2dWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = index$2.usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.label = this.widget.label;
        this.icon = this.widget.icon;
        this.unit = this.widget.unit;
        this.unitOptions = this.widget.unitOptions;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return index.getElement(this); }
};
ArcgisAreaMeasurement2d.style = ArcgisAreaMeasurement2dStyle0;

exports.arcgis_area_measurement_2d = ArcgisAreaMeasurement2d;
