// @arcgis/map-components/dist/cjs/arcgis-daylight.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-80ecaa0e.js');
const index$2 = require('./index-87ff723d.js');
const index$1 = require('./index-bb8f4b21.js');
const useWidget = require('./useWidget-3971aff7.js');
require('./component-utils-b8b66725.js');

const daylightCss = "";
const ArcgisDaylightStyle0 = daylightCss;

const useDaylightWidget = useWidget.makeWidgetController(index$2.newWidgetsDaylight);
const ArcgisDaylight = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = useDaylightWidget(this);
        this.viewModel = this.widget.viewModel;
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.currentSeason = this.viewModel.currentSeason;
        this.dayPlaying = this.viewModel.dayPlaying;
        this.yearPlaying = this.viewModel.yearPlaying;
        this.dateOrSeason = this.widget.dateOrSeason;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.headingLevel = this.widget.headingLevel;
        this.playSpeedMultiplier = this.widget.playSpeedMultiplier;
        this.timeSliderSteps = this.widget.timeSliderSteps;
        this.hideHeader = this.widget.visibleElements.header;
        this.hidePlayButtons = this.widget.visibleElements.playButtons;
        this.hideDatePicker = this.widget.visibleElements.datePicker;
        this.hideTimezone = this.widget.visibleElements.timezone;
        this.hideSunLightingToggle = this.widget.visibleElements.sunLightingToggle;
        this.hideShadowsToggle = this.widget.visibleElements.shadowsToggle;
        this.position = "bottom-left";
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return index.getElement(this); }
};
ArcgisDaylight.style = ArcgisDaylightStyle0;

exports.arcgis_daylight = ArcgisDaylight;
