// @arcgis/map-components/dist/cjs/arcgis-zoom.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-80ecaa0e.js');
const index$2 = require('./index-fbb18e12.js');
const index$3 = require('./index-87ff723d.js');
const useViewModel = require('./useViewModel-408f0a4f.js');
const globalCss = require('./globalCss-8e3b5c64.js');
const index$1 = require('./index-bb8f4b21.js');
require('./component-utils-b8b66725.js');

const css = {
    base: "esri-zoom",
    horizontalLayout: "esri-zoom--horizontal",
};

const zoomCss = ".esri-zoom{display:flex;flex-flow:column nowrap}.esri-zoom .esri-widget--button{box-shadow:none}.esri-zoom .esri-widget--button:first-child{margin-bottom:0}.esri-zoom .esri-widget--button:last-child{border-top:solid 1px var(--calcite-color-border-1)}.esri-zoom.esri-zoom--horizontal{flex-flow:row-reverse nowrap}.esri-zoom.esri-zoom--horizontal .esri-widget--button:last-child{border-top:none}.esri-zoom.esri-zoom--horizontal .esri-widget--button:first-child{border-left:solid 1px var(--calcite-color-border-1)}[dir=\"rtl\"] .esri-zoom.esri-zoom--horizontal .esri-widget--button:first-child{border-top:none}[dir=\"rtl\"] .esri-zoom.esri-zoom--horizontal .esri-widget--button:last-child{border-left:solid 1px var(--calcite-color-border-1)}";
const ArcgisZoomStyle0 = zoomCss;

const useZoomViewModel = useViewModel.makeViewModelController(index$3.newWidgetsZoomZoomViewModel);
const ArcgisZoom = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        //--------------------------------------------------------------------------
        //
        //  Controllers
        //
        //--------------------------------------------------------------------------
        this.manager = index$1.useControllerManager(this);
        this.messages = useViewModel.useT9n({});
        this.viewModel = useZoomViewModel(this);
        //--------------------------------------------------------------------------
        //
        //  Events
        //
        //--------------------------------------------------------------------------
        this.arcgisPropertyChange = index$2.usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.icon = "magnifying-glass-plus";
        this.label = undefined;
        this.layout = "vertical";
        this.messageOverrides = undefined;
        this.position = "top-left";
        this.referenceElement = undefined;
        this.state = this.viewModel.state;
        this.canZoomIn = this.viewModel.canZoomIn;
        this.canZoomOut = this.viewModel.canZoomOut;
    }
    //--------------------------------------------------------------------------
    //
    //  Public Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    /**
     * Zooms the view in by an LOD factor of 0.5.
     */
    async zoomIn() {
        this.viewModel.zoomIn();
    }
    /**
     * Zooms the view out by an LOD factor of 2.
     */
    async zoomOut() {
        this.viewModel.zoomOut();
    }
    render() {
        const rootClasses = {
            [css.horizontalLayout]: this.layout === "horizontal",
        };
        const { canZoomIn, canZoomOut } = this.viewModel;
        const { zoomIn, zoomOut } = this.messages;
        return (index.h("div", { key: '1669be1a2b0018a4efa0009cf4a465339801af84', class: index$3.classes(css.base, globalCss.globalCss.widget, rootClasses) }, index.h("calcite-button", { key: 'e283c90ad1ab58d6fc6d87b7eb79f87b8642e418', class: globalCss.globalCss.widgetButton, disabled: !canZoomIn, iconStart: "plus", kind: "neutral", label: zoomIn, onClick: this.zoomIn.bind(this), title: zoomIn }), index.h("calcite-button", { key: '1a47a30cb4cf74f4ffc3c8e7d159f32c08c410f6', class: globalCss.globalCss.widgetButton, disabled: !canZoomOut, iconStart: "minus", kind: "neutral", label: zoomOut, onClick: this.zoomOut.bind(this), title: zoomOut })));
    }
    static get assetsDirs() { return ["assets"]; }
    get el() { return index.getElement(this); }
};
ArcgisZoom.style = ArcgisZoomStyle0;

exports.arcgis_zoom = ArcgisZoom;
