// @arcgis/map-components/dist/cjs/arcgis-directions.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-80ecaa0e.js');
const index$3 = require('./index-87ff723d.js');
const componentUtils = require('./component-utils-b8b66725.js');
const index$2 = require('./index-fbb18e12.js');
const useWidget = require('./useWidget-3971aff7.js');
const index$1 = require('./index-bb8f4b21.js');

const directionsCss = "";
const ArcgisDirectionsStyle0 = directionsCss;

const useDirectionsWidget = useWidget.makeWidgetController(index$3.newWidgetsDirections);
const loggerName = "map-components:arcgis-directions";
const ArcgisDirections = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = useDirectionsWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = index$2.usePropertyChange()("lastRoute", "state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.apiKey = this.widget.apiKey;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.routeLayerItemId = index$1.getSet(undefined, {
            set: (value) => {
                if (componentUtils.isNullOrEmptyString(value) && !this.useDefaultRouteLayer) {
                    this.widget.view?.map.remove(this.layer);
                    this._setDefaultLayer();
                }
                else if (value && this.widget) {
                    index$3.newLayersRouteLayer({
                        portalItem: {
                            id: value,
                        },
                    })
                        .then((layer) => {
                        this.layer = layer;
                        this.widget.view?.map.add(this.layer);
                    })
                        .catch(console.error);
                }
                return value;
            },
        });
        this.lastRoute = this.widget.lastRoute;
        this.layer = this.widget.layer;
        this.headingLevel = this.widget.headingLevel;
        this.maxStops = this.widget.maxStops;
        this.searchProperties = this.widget.searchProperties;
        this.unit = this.widget.unit;
        this.useDefaultRouteLayer = index$1.getSet(false, {
            set: (newValue) => {
                this._setDefaultLayer();
                return newValue;
            },
        });
        this.hideLayerDetails = this.widget.visibleElements.layerDetails;
        this.hideSaveAsButton = this.widget.visibleElements.saveAsButton;
        this.hideSaveButton = this.widget.visibleElements.saveButton;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Lifecycle
    //
    //--------------------------------------------------------------------------
    componentDidLoad() {
        this._setDefaultLayer();
    }
    //--------------------------------------------------------------------------
    //
    //  Private Methods
    //
    //--------------------------------------------------------------------------
    /** Maintain a default route layer */
    _setDefaultLayer() {
        if (this.useDefaultRouteLayer && !this.routeLayerItemId) {
            index$3.newLayersRouteLayer({})
                .then((layer) => {
                this.layer = layer;
                this.widget.view?.map.add(this.layer);
            })
                .catch(console.error);
        }
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async getDirections() {
        if (this.widget?.viewModel?.state !== "ready") {
            throw new Error(`${loggerName} - component not ready to get directions`);
        }
        return await this.widget?.getDirections();
    }
    async save() {
        if (this.widget?.viewModel?.state !== "ready") {
            throw new Error(`${loggerName} - component not ready to save directions`);
        }
        return await this.widget?.save();
    }
    async saveAs(portalItem, options) {
        if (this.widget?.viewModel?.state !== "ready") {
            throw new Error(`${loggerName} - component not ready to save directions as a Portal Item`);
        }
        return await this.widget?.saveAs(portalItem, options);
    }
    async zoomToRoute() {
        if (this.widget?.viewModel?.state !== "ready") {
            throw new Error(`${loggerName} - component not ready to zoom to route`);
        }
        this.widget?.zoomToRoute();
    }
    get el() { return index.getElement(this); }
};
ArcgisDirections.style = ArcgisDirectionsStyle0;

exports.arcgis_directions = ArcgisDirections;
