// @arcgis/map-components/dist/cjs/arcgis-floor-filter.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-80ecaa0e.js');
const index$2 = require('./index-87ff723d.js');
const useWidget = require('./useWidget-3971aff7.js');
const index$1 = require('./index-bb8f4b21.js');
require('./component-utils-b8b66725.js');

const floorFilterCss = "";
const ArcgisFloorFilterStyle0 = floorFilterCss;

const useFloorFilterWidget = useWidget.makeWidgetController(index$2.newWidgetsFloorFilter);
const ArcgisFloorFilter = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = useFloorFilterWidget(this);
        this.viewModel = this.widget.viewModel;
        this.autoDestroyDisabled = false;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.referenceElement = undefined;
        this.facility = this.widget.facility;
        this.headingLevel = this.widget.headingLevel;
        this.level = this.widget.level;
        this.longNames = this.widget.longNames;
        this.site = this.widget.site;
        this.position = "bottom-left";
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async updateWebDocument(webMap) {
        this.widget?.updateWebDocument(webMap);
    }
    get el() { return index.getElement(this); }
};
ArcgisFloorFilter.style = ArcgisFloorFilterStyle0;

exports.arcgis_floor_filter = ArcgisFloorFilter;
