// @arcgis/map-components/dist/cjs/arcgis-feature-form.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-80ecaa0e.js');
const index$3 = require('./index-87ff723d.js');
const index$2 = require('./index-fbb18e12.js');
const useWidget = require('./useWidget-3971aff7.js');
const index$1 = require('./index-bb8f4b21.js');
require('./component-utils-b8b66725.js');

const featureFormCss = "";
const ArcgisFeatureFormStyle0 = featureFormCss;

const useFeatureFormWidget = useWidget.makeWidgetController(index$3.newWidgetsFeatureForm);
const ArcgisFeatureForm = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.widgetSubmit = index.createEvent(this, "arcgisSubmit", 7);
        this.widgetValueChange = index.createEvent(this, "arcgisValueChange", 7);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = useFeatureFormWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = index$2.usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.headingLevel = this.widget.headingLevel;
        this.referenceElement = undefined;
        this.position = "bottom-left";
        this.feature = this.widget.feature;
        this.formTemplate = this.widget.formTemplate;
        this.groupDisplay = this.widget.groupDisplay;
        this.layer = this.widget.layer;
        this.timeZone = this.widget.timeZone;
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async getValues() {
        return this.widget?.getValues();
    }
    async submit() {
        return this.widget?.submit();
    }
    get el() { return index.getElement(this); }
};
ArcgisFeatureForm.style = ArcgisFeatureFormStyle0;

exports.arcgis_feature_form = ArcgisFeatureForm;
