// @arcgis/map-components/dist/cjs/arcgis-fullscreen.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-80ecaa0e.js');
const index$2 = require('./index-fbb18e12.js');
const index$3 = require('./index-87ff723d.js');
const useViewModel = require('./useViewModel-408f0a4f.js');
const globalCss = require('./globalCss-8e3b5c64.js');
const index$1 = require('./index-bb8f4b21.js');
require('./component-utils-b8b66725.js');

const css = {
    base: "esri-fullscreen",
};
const useFullscreenViewModel = useViewModel.makeViewModelController(index$3.newWidgetsFullscreenFullscreenViewModel);
const ArcgisFullscreen = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        //--------------------------------------------------------------------------
        //
        //  Controllers
        //
        //--------------------------------------------------------------------------
        this.manager = index$1.useControllerManager(this);
        this.messages = useViewModel.useT9n();
        this.viewModel = useFullscreenViewModel(this);
        //--------------------------------------------------------------------------
        //
        //  Events
        //
        //--------------------------------------------------------------------------
        this.arcgisPropertyChange = index$2.usePropertyChange()("state");
        //--------------------------------------------------------------------------
        //
        //  Private Methods
        //
        //--------------------------------------------------------------------------
        this._toggle = () => {
            this.viewModel.toggle();
        };
        this.element = this.viewModel.element;
        this.icon = "zoom-out-fixed";
        this.autoDestroyDisabled = false;
        this.label = undefined;
        this.messageOverrides = undefined;
        this.position = "top-left";
        this.referenceElement = undefined;
        this.state = this.viewModel.state;
    }
    get fullscreenTitle() {
        const state = this.viewModel?.state;
        return state === "active" ? this.messages.exit : state === "ready" ? this.messages.enter : "";
    }
    //--------------------------------------------------------------------------
    //
    //  Public Methods
    //
    //--------------------------------------------------------------------------
    async destroy() {
        await this.manager.destroy();
    }
    render() {
        const { fullscreenTitle, icon } = this;
        const state = this.viewModel?.state;
        const hidden = state === "feature-unsupported";
        const defaultIcon = state === "ready" || state === "disabled" || state === "feature-unsupported" ? "zoom-out-fixed" : "zoom-in-fixed";
        return (index.h("div", { key: 'e8420490515fe56879137c622b729965e456851b', class: index$3.classes(css.base, globalCss.globalCss.widget) }, !hidden && (index.h("calcite-button", { key: '1e2f85600fb51debcc884b89590d09c38f93ff6c', class: globalCss.globalCss.widgetButton, disabled: state === "disabled", "icon-start": icon ?? defaultIcon, kind: "neutral", label: fullscreenTitle, onClick: this._toggle, title: fullscreenTitle }))));
    }
    static get assetsDirs() { return ["assets"]; }
    get el() { return index.getElement(this); }
};

exports.arcgis_fullscreen = ArcgisFullscreen;
