// @arcgis/charts-js-common/dist/commonjs/errors.t9n.ar-0213ec3f-4eb5a896.js
'use strict';

const a="حدث خطأ أثناء تحميل المخطط.",e="يتعذر تطبيق تحويل السِّجل إلى قيم سلبية أو إلى صفر.",t="يتعذر تطبيق تحويل الجذر التربيعي إلى قيم سلبية.",r="حدث خطأ أثناء تحميل الطبقة. URL = ${ url }. معرف عنصر البوابة = ${ portalItemId }.",i="يجب أن يكون ${ dataPath } مميزًا. السلسلة التي اسمها ${ seriesName } تتضمن معرفًا (${ seriesID }) مستخدمًا بالفعل من قبل سلسلة أخرى.",d="يتعذر على ${ dataPath } إجراء تجميع من دون حساب في حقل غير رقمي.",o="يتفقد ${ dataPath } خاصية باسم ${ missingProperty }.",n="يجب ألا يكون ${ dataPath } أقصر من ${ limit } من الحروف.",$="يجب ألا يحتوي ${ dataPath } على أقل من ${ limit } من العناصر.",l="يجب ألا يحتوي ${ dataPath } على أكثر من ${ limit } من العناصر.",s="يجب أن يتضمن ${ dataPath } حرفًا واحدًا على الأقل ليس مسافة بيضاء.",m="يجب أن يحتوي ${ dataPath } على ${ additionalProperty }.",u="يجب أن يكون ${ dataPath } مساويًا لإحدى القيم هذه المسموح بها: ${ allowedValues }",P="يجب أن يتطابق ${ dataPath } مع مخطط أحد ما يلي: ${ schemaOptions }.",h="Scatterplots ذات الرموز المتناسبة غير مدعومة. تم تطبيق حجم الرمز الافتراضي.",p="فشلت قراءة البيانات المدخلة.",y="نوع السلسلة المتوقع في الفهرس ${ seriesIndex } هو '${ expectedType }' ولكن تم استلام '${ receivedType }' بدلاً من ذلك",g="أو";var I={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:t,layerLoadFailure:r,duplicateSeriesID:i,nonNumericAggregation:d,requiredProperty:o,minLength:n,minItems:$,maxItems:l,whiteSpacePattern:s,additionalProperty:m,enumValues:u,anyOfValues:P,bubbleChartValidateMsg:h,queryError:p,invalidSeriesType:y,or:"أو"};

exports.additionalProperty = m;
exports.anyOfValues = P;
exports.bubbleChartValidateMsg = h;
exports["default"] = I;
exports.defaultError = a;
exports.duplicateSeriesID = i;
exports.enumValues = u;
exports.invalidSeriesType = y;
exports.layerLoadFailure = r;
exports.maxItems = l;
exports.minItems = $;
exports.minLength = n;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = t;
exports.nonNumericAggregation = d;
exports.or = g;
exports.queryError = p;
exports.requiredProperty = o;
exports.whiteSpacePattern = s;
