// @arcgis/charts-js-common/dist/commonjs/errors.t9n.vi-1a6ce09c-daa0dac6.js
'use strict';

const t="Đã xảy ra lỗi khi tải biểu đồ.",n="Không thể áp dụng phép biến đổi log lên giá trị âm hoặc bằng không.",a="Không thể áp dụng phép biến đổi căn bậc hai lên giá trị âm.",h="Đã xảy ra lỗi khi tải lớp. URL = ${ url }. ID mục cổng thông tin = ${ portalItemId }.",i="${ dataPath } phải là duy nhất. Chuỗi có tên ${ seriesName } có một ID (${ seriesID }) đã được chuỗi khác sử dụng.",e="${ dataPath } không thể thực hiện tổng hợp không đếm trên trường không phải là số.",r="${ dataPath } đang thiếu một tên phù hợp ${ missingProperty }.",g="${ dataPath } không được ngắn hơn ${ limit } ký tự.",c="${ dataPath } không được chứa dưới ${ limit } mục.",l="${ dataPath } không được chứa nhiều hơn ${ limit } mục.",o="${ dataPath } phải chứa ít nhất một ký tự không phải khoảng trắng.",p="${ dataPath } không được chứa ${ additionalProperty }.",d="${ dataPath } phải bằng một trong những giá trị được phép này: ${ allowedValues }.",m="${ dataPath } phải khớp với lược đồ của một trong những thứ sau: ${ schemaOptions }.",u="Biểu đồ phân tán có ký hiệu tỷ lệ không được hỗ trợ. Đã áp dụng kích cỡ ký hiệu theo mặc định.",s="Không thể đọc dữ liệu đầu vào.",$="Loại chuỗi mong đợi tại chỉ mục ${ seriesIndex } là '${ expectedType }' nhưng thay vào đó lại nhận được '${ receivedType }'",k="hoặc";var y={defaultError:t,negativeValueInDataForLogTransformation:n,negativeValueInDataForSqrtTransformation:a,layerLoadFailure:h,duplicateSeriesID:i,nonNumericAggregation:e,requiredProperty:r,minLength:g,minItems:c,maxItems:l,whiteSpacePattern:o,additionalProperty:p,enumValues:d,anyOfValues:m,bubbleChartValidateMsg:u,queryError:s,invalidSeriesType:$,or:k};

exports.additionalProperty = p;
exports.anyOfValues = m;
exports.bubbleChartValidateMsg = u;
exports["default"] = y;
exports.defaultError = t;
exports.duplicateSeriesID = i;
exports.enumValues = d;
exports.invalidSeriesType = $;
exports.layerLoadFailure = h;
exports.maxItems = l;
exports.minItems = c;
exports.minLength = g;
exports.negativeValueInDataForLogTransformation = n;
exports.negativeValueInDataForSqrtTransformation = a;
exports.nonNumericAggregation = e;
exports.or = k;
exports.queryError = s;
exports.requiredProperty = r;
exports.whiteSpacePattern = o;
