// @arcgis/charts-js-common/dist/commonjs/errors.t9n.ko-ece64dad-f290b9a5.js
'use strict';

const a="차트를 불러오는 동안 오류가 발생했습니다.",e="로그 변환을 음수 값 또는 0 값에 적용할 수 없습니다.",t="제곱근 변환을 음수 값에 적용할 수 없습니다.",r="레이어를 불러오는 동안 오류가 발생했습니다. URL = ${ url }. 포털 항목 ID = ${ portalItemId }.",i="${ dataPath }은(는) 고유해야 합니다. ${ seriesName }(이)라는 계열에 다른 계열에서 이미 사용된 ID(${ seriesID })가 있습니다.",d="${ dataPath }은(는) 숫자가 아닌 필드에서 불가산 집계를 수행할 수 없습니다.",n="${ dataPath }에 ${ missingProperty } 등록정보가 누락되었습니다.",o="${ dataPath }은(는) ${ limit }자보다 짧으면 안 됩니다.",$="${ dataPath }에는 ${ limit }개 이상의 항목이 있어야 합니다.",l="${ dataPath }에는 ${ limit }개 이하의 항목이 있어야 합니다.",s="${ dataPath }은(는) 공백 이외의 문자를 1개 이상 포함해야 합니다.",m="${ dataPath }에는 ${ additionalProperty }이(가) 없어야 합니다.",u="${ dataPath }은(는) 허용된 값(${ allowedValues }) 중 하나와 같아야 합니다.",P="${ dataPath }은(는) ${ schemaOptions } 중 하나의 스키마와 일치해야 합니다.",h="비례 심볼이 있는 산점도는 지원되지 않습니다. 기본 심볼 크기가 적용되었습니다.",p="입력 데이터를 읽지 못했습니다.",I="인덱스 ${ seriesIndex }에 예상되는 시리즈 유형은 '${ expectedType }'인데 '${ receivedType }'이(가) 대신 수신되었습니다.",y="또는";var g={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:t,layerLoadFailure:r,duplicateSeriesID:i,nonNumericAggregation:d,requiredProperty:n,minLength:o,minItems:$,maxItems:l,whiteSpacePattern:s,additionalProperty:m,enumValues:u,anyOfValues:P,bubbleChartValidateMsg:h,queryError:p,invalidSeriesType:I,or:"또는"};

exports.additionalProperty = m;
exports.anyOfValues = P;
exports.bubbleChartValidateMsg = h;
exports["default"] = g;
exports.defaultError = a;
exports.duplicateSeriesID = i;
exports.enumValues = u;
exports.invalidSeriesType = I;
exports.layerLoadFailure = r;
exports.maxItems = l;
exports.minItems = $;
exports.minLength = o;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = t;
exports.nonNumericAggregation = d;
exports.or = y;
exports.queryError = p;
exports.requiredProperty = n;
exports.whiteSpacePattern = s;
