// @arcgis/charts-js-common/dist/commonjs/errors.t9n.he-7950cba0-7a48e5dd.js
'use strict';

const a="אירעה שגיאה במהלך טעינת התרשים.",e="לא ניתן לבצע טרנספורמציה לוגריתמית על ערכים שליליים או על אפס.",t="לא ניתן לבצע טרנספורמציה של שורש ריבועי על ערכים שליליים.",r="אירעה שגיאה במהלך טעינת השכבה. URL = ${ url }. מזהה פריט פורטל =${ portalItemId }.",i="${ dataPath } חייב/ת להיות ייחודי/ת. לסדרה בשם ${ seriesName } יש מזהה (${ seriesID }) שבו משתמשת כבר סדרה אחרת.",d="ל-${ dataPath } אסור לבצע צבירה שאינה ספירה על שדה לא מספרי.",n="ב-${ dataPath } חסר מאפיין בשם ${ missingProperty }.",o="${ dataPath } לא יכול להיות קצר יות מ-${ limit } תווים.",$="ל-${ dataPath } לא יכולים להיות פחות מ-${ limit } פריטים.",l="ל-${ dataPath } לא יכולים להיות יותר מ-${ limit } פריטים.",s="ל-${ dataPath } חייב להיות לפחות תו אחד שאינו רווח בלתי נראה.",m="ל-${ dataPath } אסור לכלול ${ additionalProperty }.",u="${ dataPath } חייב להיות שווה לאחד מהערכים המותרים הבאים: ${ allowedValues }.",P="${ dataPath } חייב להתאים לסכמה של אחד מאלה: ${ schemaOptions }.",h="תרשימי פיזור עם סמלים פרופורציונליים לא נתמכים. הוחל גודל סמל שמוגדר כברירת מחדל.",p="קריאת נתוני הקלט נכשלה.",y="סוג הסדרה הצפוי במדד ${ seriesIndex } הוא '${ expectedType }', אבל במקום זאת התקבל '${ receivedType }'.",g="או";var I={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:t,layerLoadFailure:r,duplicateSeriesID:i,nonNumericAggregation:d,requiredProperty:n,minLength:o,minItems:$,maxItems:l,whiteSpacePattern:s,additionalProperty:m,enumValues:u,anyOfValues:P,bubbleChartValidateMsg:h,queryError:p,invalidSeriesType:y,or:"או"};

exports.additionalProperty = m;
exports.anyOfValues = P;
exports.bubbleChartValidateMsg = h;
exports["default"] = I;
exports.defaultError = a;
exports.duplicateSeriesID = i;
exports.enumValues = u;
exports.invalidSeriesType = y;
exports.layerLoadFailure = r;
exports.maxItems = l;
exports.minItems = $;
exports.minLength = o;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = t;
exports.nonNumericAggregation = d;
exports.or = g;
exports.queryError = p;
exports.requiredProperty = n;
exports.whiteSpacePattern = s;
