// @arcgis/charts-js-common/dist/commonjs/errors.t9n.zh-CN-e8b0b9ca-9e897179.js
'use strict';

const a="加载图表时出错。",e="无法向负值或零值应用对数变换。",t="无法向负值应用平方根变换。",r="加载图层时出错。 URL = ${ url }。 门户项目 ID = ${ portalItemId }。",i="${ dataPath } 必须唯一。 名为 ${ seriesName } 的系列具有一个 ID (${ seriesID })，此 ID 已被另一个系列使用。",d="${ dataPath } 无法在非数值字段上执行非计数聚合。",n="${ dataPath } 缺少名为 ${ missingProperty } 的属性。",o="${ dataPath } 不应短于 ${ limit } 个字符。",$="${ dataPath } 包含的项目数量不应少于 ${ limit } 个。",l="${ dataPath } 包含的项目数量不应多于 ${ limit } 个。",s="${ dataPath } 必须至少包含一个空白字符。",m="${ dataPath } 不能有 ${ additionalProperty }。",u="${ dataPath } 必须等于以下允许值之一：${ allowedValues }。",P="${ dataPath } 必须与下列各项之一的方案匹配：${ schemaOptions }。",h="包含比例符号的散点图不受支持。 已应用默认符号大小。",p="无法读取输入数据。",I="索引 ${ seriesIndex } 处的预期系列类型是 '${ expectedType }'，但接受的是 '${ receivedType }'",y="或";var g={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:t,layerLoadFailure:r,duplicateSeriesID:i,nonNumericAggregation:d,requiredProperty:n,minLength:o,minItems:$,maxItems:l,whiteSpacePattern:s,additionalProperty:m,enumValues:u,anyOfValues:P,bubbleChartValidateMsg:h,queryError:p,invalidSeriesType:I,or:"或"};

exports.additionalProperty = m;
exports.anyOfValues = P;
exports.bubbleChartValidateMsg = h;
exports["default"] = g;
exports.defaultError = a;
exports.duplicateSeriesID = i;
exports.enumValues = u;
exports.invalidSeriesType = I;
exports.layerLoadFailure = r;
exports.maxItems = l;
exports.minItems = $;
exports.minLength = o;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = t;
exports.nonNumericAggregation = d;
exports.or = y;
exports.queryError = p;
exports.requiredProperty = n;
exports.whiteSpacePattern = s;
