// @esri/calcite-components/dist/collection/components/input-time-zone/input-time-zone.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/dev/LICENSE.md for details.
 * v2.13.2
 */
import { h, Host, } from "@stencil/core";
import { connectLabel, disconnectLabel } from "../../utils/label";
import { InteractiveContainer, updateHostInteraction, } from "../../utils/interactive";
import { connectLocalized, disconnectLocalized, } from "../../utils/locale";
import { connectMessages, disconnectMessages, setUpMessages, updateMessages, } from "../../utils/t9n";
import { componentFocusable, setComponentLoaded, setUpLoadableComponent, } from "../../utils/loadable";
import { afterConnectDefaultValueSet, connectForm, disconnectForm, HiddenFormInputSlot, } from "../../utils/form";
import { CSS } from "./resources";
import { createTimeZoneItems, findTimeZoneItemByProp, getNormalizer, getSelectedRegionTimeZoneLabel, getUserTimeZoneName, getUserTimeZoneOffset, } from "./utils";
export class InputTimeZone {
    constructor() {
        this.setComboboxRef = (el) => {
            this.comboboxEl = el;
        };
        this.onComboboxBeforeClose = (event) => {
            event.stopPropagation();
            this.overrideSelectedLabelForRegion(false);
            this.calciteInputTimeZoneBeforeClose.emit();
        };
        this.onComboboxBeforeOpen = (event) => {
            event.stopPropagation();
            this.overrideSelectedLabelForRegion(true);
            this.calciteInputTimeZoneBeforeOpen.emit();
        };
        this.onComboboxChange = (event) => {
            event.stopPropagation();
            const combobox = event.target;
            const selectedItem = combobox.selectedItems[0];
            if (!selectedItem) {
                this.value = null;
                this.selectedTimeZoneItem = null;
                this.calciteInputTimeZoneChange.emit();
                return;
            }
            const selected = this.findTimeZoneItemByLabel(selectedItem.getAttribute("data-label"));
            const selectedValue = `${selected.value}`;
            if (this.value === selectedValue && selected.label === this.selectedTimeZoneItem.label) {
                return;
            }
            this.value = selectedValue;
            this.selectedTimeZoneItem = selected;
            this.calciteInputTimeZoneChange.emit();
        };
        this.onComboboxClose = (event) => {
            event.stopPropagation();
            this.open = false;
            this.calciteInputTimeZoneClose.emit();
        };
        this.onComboboxOpen = (event) => {
            this.open = true;
            event.stopPropagation();
            this.calciteInputTimeZoneOpen.emit();
        };
        this.clearable = false;
        this.disabled = false;
        this.form = undefined;
        this.maxItems = 0;
        this.messages = undefined;
        this.messageOverrides = undefined;
        this.mode = "offset";
        this.offsetStyle = "user";
        this.validationMessage = undefined;
        this.validationIcon = undefined;
        this.validity = {
            valid: false,
            badInput: false,
            customError: false,
            patternMismatch: false,
            rangeOverflow: false,
            rangeUnderflow: false,
            stepMismatch: false,
            tooLong: false,
            tooShort: false,
            typeMismatch: false,
            valueMissing: false,
        };
        this.name = undefined;
        this.open = false;
        this.overlayPositioning = "absolute";
        this.referenceDate = undefined;
        this.required = false;
        this.scale = "m";
        this.status = "idle";
        this.value = undefined;
        this.readOnly = false;
        this.defaultMessages = undefined;
        this.effectiveLocale = "";
    }
    onMessagesChange() {
        /* wired up by t9n util */
    }
    handleTimeZoneItemPropsChange() {
        if (!this.timeZoneItems) {
            return;
        }
        this.updateTimeZoneItems();
        this.updateTimeZoneSelection();
    }
    openChanged() {
        // we set the property instead of the attribute to ensure open/close events are emitted properly
        this.comboboxEl.open = this.open;
    }
    handleValueChange(value, oldValue) {
        value = this.normalizeValue(value);
        if (!value) {
            if (this.clearable) {
                this.value = value;
                this.selectedTimeZoneItem = null;
                return;
            }
            this.value = oldValue;
            this.selectedTimeZoneItem = this.findTimeZoneItem(oldValue);
            return;
        }
        const timeZoneItem = this.findTimeZoneItem(value);
        if (!timeZoneItem) {
            this.value = oldValue;
            return;
        }
        this.selectedTimeZoneItem = timeZoneItem;
    }
    //--------------------------------------------------------------------------
    //
    //  Public Methods
    //
    //--------------------------------------------------------------------------
    async setFocus() {
        await componentFocusable(this);
        await this.comboboxEl.setFocus();
    }
    effectiveLocaleWatcher() {
        updateMessages(this, this.effectiveLocale);
    }
    //--------------------------------------------------------------------------
    //
    //  Private Methods
    //
    //--------------------------------------------------------------------------
    onLabelClick() {
        this.setFocus();
    }
    /**
     * Helps override the selected item's label for region mode outside of item rendering logic to avoid flickering text change
     *
     * @param open
     * @private
     */
    overrideSelectedLabelForRegion(open) {
        if (this.mode !== "region" || !this.selectedTimeZoneItem) {
            return;
        }
        const { label, metadata } = this.selectedTimeZoneItem;
        this.comboboxEl.selectedItems[0].textLabel =
            !metadata.country || open
                ? label
                : getSelectedRegionTimeZoneLabel(label, metadata.country, this.messages);
    }
    findTimeZoneItem(value) {
        return findTimeZoneItemByProp(this.timeZoneItems, "value", value);
    }
    findTimeZoneItemByLabel(label) {
        return findTimeZoneItemByProp(this.timeZoneItems, "label", label);
    }
    async updateTimeZoneItems() {
        this.timeZoneItems = await this.createTimeZoneItems();
    }
    async updateTimeZoneSelection() {
        if (this.value === "" && this.clearable) {
            this.selectedTimeZoneItem = null;
            return;
        }
        const fallbackValue = this.mode === "offset" ? getUserTimeZoneOffset() : getUserTimeZoneName();
        const valueToMatch = this.value ?? fallbackValue;
        this.selectedTimeZoneItem =
            this.findTimeZoneItem(valueToMatch) || this.findTimeZoneItem(fallbackValue);
    }
    async createTimeZoneItems() {
        if (!this.effectiveLocale || !this.messages) {
            return [];
        }
        return createTimeZoneItems(this.effectiveLocale, this.messages, this.mode, this.referenceDate instanceof Date
            ? this.referenceDate
            : new Date(this.referenceDate ?? Date.now()), this.offsetStyle);
    }
    // --------------------------------------------------------------------------
    //
    //  Lifecycle
    //
    //--------------------------------------------------------------------------
    connectedCallback() {
        connectForm(this);
        connectLabel(this);
        connectLocalized(this);
        connectMessages(this);
    }
    disconnectedCallback() {
        disconnectForm(this);
        disconnectLabel(this);
        disconnectLocalized(this);
        disconnectMessages(this);
    }
    normalizeValue(value) {
        value = value === null ? "" : value;
        return value ? this.normalizer(value) : value;
    }
    async componentWillLoad() {
        setUpLoadableComponent(this);
        const [, normalizer] = await Promise.all([setUpMessages(this), getNormalizer(this.mode)]);
        this.normalizer = normalizer;
        await this.updateTimeZoneItems();
        this.value = this.normalizeValue(this.value);
        await this.updateTimeZoneSelection();
        const selectedValue = this.selectedTimeZoneItem ? `${this.selectedTimeZoneItem.value}` : null;
        afterConnectDefaultValueSet(this, selectedValue);
        this.value = selectedValue;
    }
    componentDidLoad() {
        setComponentLoaded(this);
        this.openChanged();
    }
    componentDidRender() {
        updateHostInteraction(this);
        this.overrideSelectedLabelForRegion(this.open);
    }
    render() {
        return (h(Host, { key: 'b94fb2d93cdcaf0c44bbd2a0c7deaf59701078bf' }, h(InteractiveContainer, { key: '45246bbab441e9daf0e372832e74d7660039c770', disabled: this.disabled }, h("calcite-combobox", { key: '294e44d8ab01079651c417f7808348e584f135b7', clearDisabled: !this.clearable, disabled: this.disabled, label: this.messages.chooseTimeZone, lang: this.effectiveLocale, maxItems: this.maxItems, onCalciteComboboxBeforeClose: this.onComboboxBeforeClose, onCalciteComboboxBeforeOpen: this.onComboboxBeforeOpen, onCalciteComboboxChange: this.onComboboxChange, onCalciteComboboxClose: this.onComboboxClose, onCalciteComboboxOpen: this.onComboboxOpen, overlayPositioning: this.overlayPositioning, placeholder: this.mode === "name"
                ? this.messages.namePlaceholder
                : this.mode === "offset"
                    ? this.messages.offsetPlaceholder
                    : this.messages.regionPlaceholder, placeholderIcon: "search", readOnly: this.readOnly, ref: this.setComboboxRef, scale: this.scale, selectionMode: this.clearable ? "single" : "single-persist", status: this.status, "validation-icon": this.validationIcon, "validation-message": this.validationMessage }, this.renderItems()), h(HiddenFormInputSlot, { key: '0b7a0694166f1d5b36b780d3436dad15d466bc3c', component: this }))));
    }
    renderItems() {
        if (this.mode === "region") {
            return this.renderRegionItems();
        }
        return this.timeZoneItems.map((group) => {
            const selected = this.selectedTimeZoneItem === group;
            const { label, metadata, value } = group;
            return (h("calcite-combobox-item", { "data-label": label, key: label, metadata: metadata, selected: selected, textLabel: label, value: value }));
        });
    }
    renderRegionItems() {
        return this.timeZoneItems.flatMap(({ label, items }) => (h("calcite-combobox-item-group", { key: label, label: label }, items.map((item) => {
            const selected = this.selectedTimeZoneItem === item;
            const { label, metadata, value } = item;
            return (h("calcite-combobox-item", { "data-label": label, description: metadata.country, key: label, metadata: metadata, selected: selected, textLabel: label, value: value }, h("span", { class: CSS.offset, slot: "content-end" }, metadata.offset)));
        }))));
    }
    static get is() { return "calcite-input-time-zone"; }
    static get encapsulation() { return "shadow"; }
    static get delegatesFocus() { return true; }
    static get originalStyleUrls() {
        return {
            "$": ["input-time-zone.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["input-time-zone.css"]
        };
    }
    static get assetsDirs() { return ["assets"]; }
    static get properties() {
        return {
            "clearable": {
                "type": "boolean",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "When `true`, an empty value (`null`) will be allowed as a `value`.\n\nWhen `false`, an offset or name value is enforced, and clearing the input or blurring will restore the last valid `value`."
                },
                "attribute": "clearable",
                "reflect": true,
                "defaultValue": "false"
            },
            "disabled": {
                "type": "boolean",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "When `true`, interaction is prevented and the component is displayed with lower opacity."
                },
                "attribute": "disabled",
                "reflect": true,
                "defaultValue": "false"
            },
            "form": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "The `id` of the form that will be associated with the component.\n\nWhen not set, the component will be associated with its ancestor form element, if any."
                },
                "attribute": "form",
                "reflect": true
            },
            "maxItems": {
                "type": "number",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Specifies the component's maximum number of options to display before displaying a scrollbar."
                },
                "attribute": "max-items",
                "reflect": true,
                "defaultValue": "0"
            },
            "messages": {
                "type": "unknown",
                "mutable": true,
                "complexType": {
                    "original": "InputTimeZoneMessages",
                    "resolved": "{ chooseTimeZone: string; offsetPlaceholder: string; regionPlaceholder: string; namePlaceholder: string; timeZoneLabel: string; timeZoneRegionLabel: string; \"Africa/Abidjan\": string; \"Africa/Accra\": string; \"Africa/Addis_Ababa\": string; \"Africa/Algiers\": string; \"Africa/Asmera\": string; \"Africa/Bamako\": string; \"Africa/Bangui\": string; \"Africa/Banjul\": string; \"Africa/Bissau\": string; \"Africa/Blantyre\": string; \"Africa/Brazzaville\": string; \"Africa/Bujumbura\": string; \"Africa/Cairo\": string; \"Africa/Casablanca\": string; \"Africa/Ceuta\": string; \"Africa/Conakry\": string; \"Africa/Dakar\": string; \"Africa/Dar_es_Salaam\": string; \"Africa/Djibouti\": string; \"Africa/Douala\": string; \"Africa/El_Aaiun\": string; \"Africa/Freetown\": string; \"Africa/Gaborone\": string; \"Africa/Harare\": string; \"Africa/Johannesburg\": string; \"Africa/Juba\": string; \"Africa/Kampala\": string; \"Africa/Khartoum\": string; \"Africa/Kigali\": string; \"Africa/Kinshasa\": string; \"Africa/Lagos\": string; \"Africa/Libreville\": string; \"Africa/Lome\": string; \"Africa/Luanda\": string; \"Africa/Lubumbashi\": string; \"Africa/Lusaka\": string; \"Africa/Malabo\": string; \"Africa/Maputo\": string; \"Africa/Maseru\": string; \"Africa/Mbabane\": string; \"Africa/Mogadishu\": string; \"Africa/Monrovia\": string; \"Africa/Nairobi\": string; \"Africa/Ndjamena\": string; \"Africa/Niamey\": string; \"Africa/Nouakchott\": string; \"Africa/Ouagadougou\": string; \"Africa/Porto-Novo\": string; \"Africa/Sao_Tome\": string; \"Africa/Tripoli\": string; \"Africa/Tunis\": string; \"Africa/Windhoek\": string; \"America/Adak\": string; \"America/Anchorage\": string; \"America/Anguilla\": string; \"America/Antigua\": string; \"America/Araguaina\": string; \"America/Argentina/Buenos_Aires\": string; \"America/Argentina/Catamarca\": string; \"America/Argentina/Cordoba\": string; \"America/Argentina/Jujuy\": string; \"America/Argentina/La_Rioja\": string; \"America/Argentina/Mendoza\": string; \"America/Argentina/Rio_Gallegos\": string; \"America/Argentina/Salta\": string; \"America/Argentina/San_Juan\": string; \"America/Argentina/San_Luis\": string; \"America/Argentina/Tucuman\": string; \"America/Argentina/Ushuaia\": string; \"America/Aruba\": string; \"America/Asuncion\": string; \"America/Bahia\": string; \"America/Bahia_Banderas\": string; \"America/Barbados\": string; \"America/Belem\": string; \"America/Belize\": string; \"America/Blanc-Sablon\": string; \"America/Boa_Vista\": string; \"America/Bogota\": string; \"America/Boise\": string; \"America/Buenos_Aires\": string; \"America/Cambridge_Bay\": string; \"America/Campo_Grande\": string; \"America/Cancun\": string; \"America/Caracas\": string; \"America/Catamarca\": string; \"America/Cayenne\": string; \"America/Cayman\": string; \"America/Chicago\": string; \"America/Chihuahua\": string; \"America/Ciudad_Juarez\": string; \"America/Coral_Harbour\": string; \"America/Cordoba\": string; \"America/Costa_Rica\": string; \"America/Creston\": string; \"America/Cuiaba\": string; \"America/Curacao\": string; \"America/Danmarkshavn\": string; \"America/Dawson\": string; \"America/Dawson_Creek\": string; \"America/Denver\": string; \"America/Detroit\": string; \"America/Dominica\": string; \"America/Edmonton\": string; \"America/Eirunepe\": string; \"America/El_Salvador\": string; \"America/Fort_Nelson\": string; \"America/Fortaleza\": string; \"America/Glace_Bay\": string; \"America/Godthab\": string; \"America/Goose_Bay\": string; \"America/Grand_Turk\": string; \"America/Grenada\": string; \"America/Guadeloupe\": string; \"America/Guatemala\": string; \"America/Guayaquil\": string; \"America/Guyana\": string; \"America/Halifax\": string; \"America/Havana\": string; \"America/Hermosillo\": string; \"America/Indiana/Indianapolis\": string; \"America/Indiana/Knox\": string; \"America/Indiana/Marengo\": string; \"America/Indiana/Petersburg\": string; \"America/Indiana/Tell_City\": string; \"America/Indiana/Vevay\": string; \"America/Indiana/Vincennes\": string; \"America/Indiana/Winamac\": string; \"America/Indianapolis\": string; \"America/Inuvik\": string; \"America/Iqaluit\": string; \"America/Jamaica\": string; \"America/Jujuy\": string; \"America/Juneau\": string; \"America/Kentucky/Louisville\": string; \"America/Kentucky/Monticello\": string; \"America/Kralendijk\": string; \"America/La_Paz\": string; \"America/Lima\": string; \"America/Los_Angeles\": string; \"America/Louisville\": string; \"America/Lower_Princes\": string; \"America/Maceio\": string; \"America/Managua\": string; \"America/Manaus\": string; \"America/Marigot\": string; \"America/Martinique\": string; \"America/Matamoros\": string; \"America/Mazatlan\": string; \"America/Mendoza\": string; \"America/Menominee\": string; \"America/Merida\": string; \"America/Metlakatla\": string; \"America/Mexico_City\": string; \"America/Miquelon\": string; \"America/Moncton\": string; \"America/Monterrey\": string; \"America/Montevideo\": string; \"America/Montreal\": string; \"America/Montserrat\": string; \"America/Nassau\": string; \"America/New_York\": string; \"America/Nipigon\": string; \"America/Nome\": string; \"America/Noronha\": string; \"America/North_Dakota/Beulah\": string; \"America/North_Dakota/Center\": string; \"America/North_Dakota/New_Salem\": string; \"America/Nuuk\": string; \"America/Ojinaga\": string; \"America/Panama\": string; \"America/Pangnirtung\": string; \"America/Paramaribo\": string; \"America/Phoenix\": string; \"America/Port-au-Prince\": string; \"America/Port_of_Spain\": string; \"America/Porto_Velho\": string; \"America/Puerto_Rico\": string; \"America/Punta_Arenas\": string; \"America/Rainy_River\": string; \"America/Rankin_Inlet\": string; \"America/Recife\": string; \"America/Regina\": string; \"America/Resolute\": string; \"America/Rio_Branco\": string; \"America/Santa_Isabel\": string; \"America/Santarem\": string; \"America/Santiago\": string; \"America/Santo_Domingo\": string; \"America/Sao_Paulo\": string; \"America/Scoresbysund\": string; \"America/Sitka\": string; \"America/St_Barthelemy\": string; \"America/St_Johns\": string; \"America/St_Kitts\": string; \"America/St_Lucia\": string; \"America/St_Thomas\": string; \"America/St_Vincent\": string; \"America/Swift_Current\": string; \"America/Tegucigalpa\": string; \"America/Thule\": string; \"America/Thunder_Bay\": string; \"America/Tijuana\": string; \"America/Toronto\": string; \"America/Tortola\": string; \"America/Vancouver\": string; \"America/Whitehorse\": string; \"America/Winnipeg\": string; \"America/Yakutat\": string; \"America/Yellowknife\": string; \"Antarctica/Casey\": string; \"Antarctica/Davis\": string; \"Antarctica/DumontDUrville\": string; \"Antarctica/Macquarie\": string; \"Antarctica/Mawson\": string; \"Antarctica/McMurdo\": string; \"Antarctica/Palmer\": string; \"Antarctica/Rothera\": string; \"Antarctica/Syowa\": string; \"Antarctica/Troll\": string; \"Antarctica/Vostok\": string; \"Arctic/Longyearbyen\": string; \"Asia/Aden\": string; \"Asia/Almaty\": string; \"Asia/Amman\": string; \"Asia/Anadyr\": string; \"Asia/Aqtau\": string; \"Asia/Aqtobe\": string; \"Asia/Ashgabat\": string; \"Asia/Atyrau\": string; \"Asia/Baghdad\": string; \"Asia/Bahrain\": string; \"Asia/Baku\": string; \"Asia/Bangkok\": string; \"Asia/Barnaul\": string; \"Asia/Beirut\": string; \"Asia/Bishkek\": string; \"Asia/Brunei\": string; \"Asia/Calcutta\": string; \"Asia/Chita\": string; \"Asia/Choibalsan\": string; \"Asia/Colombo\": string; \"Asia/Damascus\": string; \"Asia/Dhaka\": string; \"Asia/Dili\": string; \"Asia/Dubai\": string; \"Asia/Dushanbe\": string; \"Asia/Famagusta\": string; \"Asia/Gaza\": string; \"Asia/Hebron\": string; \"Asia/Ho_Chi_Minh\": string; \"Asia/Hong_Kong\": string; \"Asia/Hovd\": string; \"Asia/Irkutsk\": string; \"Asia/Jakarta\": string; \"Asia/Jayapura\": string; \"Asia/Jerusalem\": string; \"Asia/Kabul\": string; \"Asia/Kamchatka\": string; \"Asia/Karachi\": string; \"Asia/Katmandu\": string; \"Asia/Kathmandu\": string; \"Asia/Khandyga\": string; \"Asia/Kolkata\": string; \"Asia/Krasnoyarsk\": string; \"Asia/Kuala_Lumpur\": string; \"Asia/Kuching\": string; \"Asia/Kuwait\": string; \"Asia/Macau\": string; \"Asia/Magadan\": string; \"Asia/Makassar\": string; \"Asia/Manila\": string; \"Asia/Muscat\": string; \"Asia/Nicosia\": string; \"Asia/Novokuznetsk\": string; \"Asia/Novosibirsk\": string; \"Asia/Omsk\": string; \"Asia/Oral\": string; \"Asia/Phnom_Penh\": string; \"Asia/Pontianak\": string; \"Asia/Pyongyang\": string; \"Asia/Qatar\": string; \"Asia/Qostanay\": string; \"Asia/Qyzylorda\": string; \"Asia/Rangoon\": string; \"Asia/Riyadh\": string; \"Asia/Saigon\": string; \"Asia/Sakhalin\": string; \"Asia/Samarkand\": string; \"Asia/Seoul\": string; \"Asia/Shanghai\": string; \"Asia/Singapore\": string; \"Asia/Srednekolymsk\": string; \"Asia/Taipei\": string; \"Asia/Tashkent\": string; \"Asia/Tbilisi\": string; \"Asia/Tehran\": string; \"Asia/Thimphu\": string; \"Asia/Tokyo\": string; \"Asia/Tomsk\": string; \"Asia/Ulaanbaatar\": string; \"Asia/Urumqi\": string; \"Asia/Ust-Nera\": string; \"Asia/Vientiane\": string; \"Asia/Vladivostok\": string; \"Asia/Yakutsk\": string; \"Asia/Yangon\": string; \"Asia/Yekaterinburg\": string; \"Asia/Yerevan\": string; \"Atlantic/Azores\": string; \"Atlantic/Bermuda\": string; \"Atlantic/Canary\": string; \"Atlantic/Cape_Verde\": string; \"Atlantic/Faeroe\": string; \"Atlantic/Faroe\": string; \"Atlantic/Madeira\": string; \"Atlantic/Reykjavik\": string; \"Atlantic/South_Georgia\": string; \"Atlantic/St_Helena\": string; \"Atlantic/Stanley\": string; \"Australia/Adelaide\": string; \"Australia/Brisbane\": string; \"Australia/Broken_Hill\": string; \"Australia/Currie\": string; \"Australia/Darwin\": string; \"Australia/Eucla\": string; \"Australia/Hobart\": string; \"Australia/Lindeman\": string; \"Australia/Lord_Howe\": string; \"Australia/Melbourne\": string; \"Australia/Perth\": string; \"Australia/Sydney\": string; \"Europe/Amsterdam\": string; \"Etc/GMT\": string; \"Etc/GMT+1\": string; \"Etc/GMT+10\": string; \"Etc/GMT+11\": string; \"Etc/GMT+12\": string; \"Etc/GMT+2\": string; \"Etc/GMT+3\": string; \"Etc/GMT+4\": string; \"Etc/GMT+5\": string; \"Etc/GMT+6\": string; \"Etc/GMT+7\": string; \"Etc/GMT+8\": string; \"Etc/GMT+9\": string; \"Etc/GMT-1\": string; \"Etc/GMT-10\": string; \"Etc/GMT-11\": string; \"Etc/GMT-12\": string; \"Etc/GMT-13\": string; \"Etc/GMT-14\": string; \"Etc/GMT-2\": string; \"Etc/GMT-3\": string; \"Etc/GMT-4\": string; \"Etc/GMT-5\": string; \"Etc/GMT-6\": string; \"Etc/GMT-7\": string; \"Etc/GMT-8\": string; \"Etc/GMT-9\": string; \"Etc/UTC\": string; \"Europe/Andorra\": string; \"Europe/Astrakhan\": string; \"Europe/Athens\": string; \"Europe/Belgrade\": string; \"Europe/Berlin\": string; \"Europe/Bratislava\": string; \"Europe/Brussels\": string; \"Europe/Bucharest\": string; \"Europe/Budapest\": string; \"Europe/Busingen\": string; \"Europe/Chisinau\": string; \"Europe/Copenhagen\": string; \"Europe/Dublin\": string; \"Europe/Gibraltar\": string; \"Europe/Guernsey\": string; \"Europe/Helsinki\": string; \"Europe/Isle_of_Man\": string; \"Europe/Istanbul\": string; \"Europe/Jersey\": string; \"Europe/Kaliningrad\": string; \"Europe/Kiev\": string; \"Europe/Kirov\": string; \"Europe/Kyiv\": string; \"Europe/Lisbon\": string; \"Europe/Ljubljana\": string; \"Europe/London\": string; \"Europe/Luxembourg\": string; \"Europe/Madrid\": string; \"Europe/Malta\": string; \"Europe/Mariehamn\": string; \"Europe/Minsk\": string; \"Europe/Monaco\": string; \"Europe/Moscow\": string; \"Europe/Oslo\": string; \"Europe/Paris\": string; \"Europe/Podgorica\": string; \"Europe/Prague\": string; \"Europe/Riga\": string; \"Europe/Rome\": string; \"Europe/Samara\": string; \"Europe/San_Marino\": string; \"Europe/Sarajevo\": string; \"Europe/Saratov\": string; \"Europe/Simferopol\": string; \"Europe/Skopje\": string; \"Europe/Sofia\": string; \"Europe/Stockholm\": string; \"Europe/Tallinn\": string; \"Europe/Tirane\": string; \"Europe/Ulyanovsk\": string; \"Europe/Uzhgorod\": string; \"Europe/Vaduz\": string; \"Europe/Vatican\": string; \"Europe/Vienna\": string; \"Europe/Vilnius\": string; \"Europe/Volgograd\": string; \"Europe/Warsaw\": string; \"Europe/Zagreb\": string; \"Europe/Zaporozhye\": string; \"Europe/Zurich\": string; \"Indian/Antananarivo\": string; Factory: string; \"Indian/Chagos\": string; \"Indian/Christmas\": string; \"Indian/Cocos\": string; \"Indian/Comoro\": string; \"Indian/Kerguelen\": string; \"Indian/Mahe\": string; \"Indian/Maldives\": string; \"Indian/Mauritius\": string; \"Indian/Mayotte\": string; \"Indian/Reunion\": string; \"Pacific/Apia\": string; \"Pacific/Auckland\": string; \"Pacific/Bougainville\": string; \"Pacific/Chatham\": string; \"Pacific/Easter\": string; \"Pacific/Efate\": string; \"Pacific/Enderbury\": string; \"Pacific/Fakaofo\": string; \"Pacific/Fiji\": string; \"Pacific/Funafuti\": string; \"Pacific/Galapagos\": string; \"Pacific/Gambier\": string; \"Pacific/Guadalcanal\": string; \"Pacific/Guam\": string; \"Pacific/Honolulu\": string; \"Pacific/Johnston\": string; \"Pacific/Kanton\": string; \"Pacific/Kiritimati\": string; \"Pacific/Kosrae\": string; \"Pacific/Kwajalein\": string; \"Pacific/Majuro\": string; \"Pacific/Marquesas\": string; \"Pacific/Midway\": string; \"Pacific/Nauru\": string; \"Pacific/Niue\": string; \"Pacific/Norfolk\": string; \"Pacific/Noumea\": string; \"Pacific/Pago_Pago\": string; \"Pacific/Palau\": string; \"Pacific/Pitcairn\": string; \"Pacific/Ponape\": string; \"Pacific/Port_Moresby\": string; \"Pacific/Rarotonga\": string; \"Pacific/Saipan\": string; \"Pacific/Tahiti\": string; \"Pacific/Tarawa\": string; \"Pacific/Tongatapu\": string; \"Pacific/Truk\": string; \"Pacific/Wake\": string; \"Pacific/Wallis\": string; Africa: string; America: string; Antarctica: string; Arctic: string; Asia: string; Atlantic: string; Australia: string; Europe: string; Global: string; Indian: string; Pacific: string; AD: string; AE: string; AF: string; AG: string; AI: string; AL: string; AM: string; AO: string; AQ: string; AR: string; AS: string; AT: string; AU: string; AW: string; AX: string; AZ: string; BA: string; BB: string; BD: string; BE: string; BF: string; BG: string; BH: string; BI: string; BJ: string; BL: string; BM: string; BN: string; BO: string; BQ: string; BR: string; BS: string; BT: string; BV: string; BW: string; BY: string; BZ: string; CA: string; CC: string; CD: string; CF: string; CG: string; CH: string; CI: string; CK: string; CL: string; CM: string; CN: string; CO: string; CR: string; CU: string; CV: string; CW: string; CX: string; CY: string; CZ: string; DE: string; DJ: string; DK: string; DM: string; DO: string; DZ: string; EC: string; EE: string; EG: string; EH: string; ER: string; ES: string; ET: string; FI: string; FJ: string; FK: string; FM: string; FO: string; FR: string; GA: string; GB: string; GD: string; GE: string; GF: string; GG: string; GH: string; GI: string; GL: string; GM: string; GN: string; GP: string; GQ: string; GR: string; GS: string; GT: string; GU: string; GW: string; GY: string; HK: string; HM: string; HN: string; HR: string; HT: string; HU: string; ID: string; IE: string; IL: string; IM: string; IN: string; IO: string; IQ: string; IR: string; IS: string; IT: string; JE: string; JM: string; JO: string; JP: string; KE: string; KG: string; KH: string; KI: string; KM: string; KN: string; KP: string; KR: string; KW: string; KY: string; KZ: string; LA: string; LB: string; LC: string; LI: string; LK: string; LR: string; LS: string; LT: string; LU: string; LV: string; LY: string; MA: string; MC: string; MD: string; ME: string; MF: string; MG: string; MH: string; MK: string; ML: string; MM: string; MN: string; MO: string; MP: string; MQ: string; MR: string; MS: string; MT: string; MU: string; MV: string; MW: string; MX: string; MY: string; MZ: string; NA: string; NC: string; NE: string; NF: string; NG: string; NI: string; NL: string; NO: string; NP: string; NR: string; NU: string; NZ: string; OM: string; PA: string; PE: string; PF: string; PG: string; PH: string; PK: string; PL: string; PM: string; PN: string; PR: string; PS: string; PT: string; PW: string; PY: string; QA: string; RE: string; RO: string; RS: string; RU: string; RW: string; SA: string; SB: string; SC: string; SD: string; SE: string; SG: string; SH: string; SI: string; SJ: string; SK: string; SL: string; SM: string; SN: string; SO: string; SR: string; SS: string; ST: string; SV: string; SX: string; SY: string; SZ: string; TC: string; TD: string; TF: string; TG: string; TH: string; TJ: string; TK: string; TL: string; TM: string; TN: string; TO: string; TR: string; TT: string; TV: string; TW: string; TZ: string; UA: string; UG: string; UM: string; US: string; UY: string; UZ: string; VA: string; VC: string; VE: string; VG: string; VI: string; VN: string; VU: string; WF: string; WS: string; YE: string; YT: string; ZA: string; ZM: string; ZW: string; }",
                    "references": {
                        "InputTimeZoneMessages": {
                            "location": "import",
                            "path": "./assets/input-time-zone/t9n",
                            "id": "src/components/input-time-zone/assets/input-time-zone/t9n/index.d.ts::InputTimeZoneMessages"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [{
                            "name": "internal",
                            "text": undefined
                        }],
                    "text": "Made into a prop for testing purposes only"
                }
            },
            "messageOverrides": {
                "type": "unknown",
                "mutable": true,
                "complexType": {
                    "original": "Partial<InputTimeZoneMessages>",
                    "resolved": "{ chooseTimeZone?: string; offsetPlaceholder?: string; regionPlaceholder?: string; namePlaceholder?: string; timeZoneLabel?: string; timeZoneRegionLabel?: string; \"Africa/Abidjan\"?: string; \"Africa/Accra\"?: string; \"Africa/Addis_Ababa\"?: string; \"Africa/Algiers\"?: string; \"Africa/Asmera\"?: string; \"Africa/Bamako\"?: string; \"Africa/Bangui\"?: string; \"Africa/Banjul\"?: string; \"Africa/Bissau\"?: string; \"Africa/Blantyre\"?: string; \"Africa/Brazzaville\"?: string; \"Africa/Bujumbura\"?: string; \"Africa/Cairo\"?: string; \"Africa/Casablanca\"?: string; \"Africa/Ceuta\"?: string; \"Africa/Conakry\"?: string; \"Africa/Dakar\"?: string; \"Africa/Dar_es_Salaam\"?: string; \"Africa/Djibouti\"?: string; \"Africa/Douala\"?: string; \"Africa/El_Aaiun\"?: string; \"Africa/Freetown\"?: string; \"Africa/Gaborone\"?: string; \"Africa/Harare\"?: string; \"Africa/Johannesburg\"?: string; \"Africa/Juba\"?: string; \"Africa/Kampala\"?: string; \"Africa/Khartoum\"?: string; \"Africa/Kigali\"?: string; \"Africa/Kinshasa\"?: string; \"Africa/Lagos\"?: string; \"Africa/Libreville\"?: string; \"Africa/Lome\"?: string; \"Africa/Luanda\"?: string; \"Africa/Lubumbashi\"?: string; \"Africa/Lusaka\"?: string; \"Africa/Malabo\"?: string; \"Africa/Maputo\"?: string; \"Africa/Maseru\"?: string; \"Africa/Mbabane\"?: string; \"Africa/Mogadishu\"?: string; \"Africa/Monrovia\"?: string; \"Africa/Nairobi\"?: string; \"Africa/Ndjamena\"?: string; \"Africa/Niamey\"?: string; \"Africa/Nouakchott\"?: string; \"Africa/Ouagadougou\"?: string; \"Africa/Porto-Novo\"?: string; \"Africa/Sao_Tome\"?: string; \"Africa/Tripoli\"?: string; \"Africa/Tunis\"?: string; \"Africa/Windhoek\"?: string; \"America/Adak\"?: string; \"America/Anchorage\"?: string; \"America/Anguilla\"?: string; \"America/Antigua\"?: string; \"America/Araguaina\"?: string; \"America/Argentina/Buenos_Aires\"?: string; \"America/Argentina/Catamarca\"?: string; \"America/Argentina/Cordoba\"?: string; \"America/Argentina/Jujuy\"?: string; \"America/Argentina/La_Rioja\"?: string; \"America/Argentina/Mendoza\"?: string; \"America/Argentina/Rio_Gallegos\"?: string; \"America/Argentina/Salta\"?: string; \"America/Argentina/San_Juan\"?: string; \"America/Argentina/San_Luis\"?: string; \"America/Argentina/Tucuman\"?: string; \"America/Argentina/Ushuaia\"?: string; \"America/Aruba\"?: string; \"America/Asuncion\"?: string; \"America/Bahia\"?: string; \"America/Bahia_Banderas\"?: string; \"America/Barbados\"?: string; \"America/Belem\"?: string; \"America/Belize\"?: string; \"America/Blanc-Sablon\"?: string; \"America/Boa_Vista\"?: string; \"America/Bogota\"?: string; \"America/Boise\"?: string; \"America/Buenos_Aires\"?: string; \"America/Cambridge_Bay\"?: string; \"America/Campo_Grande\"?: string; \"America/Cancun\"?: string; \"America/Caracas\"?: string; \"America/Catamarca\"?: string; \"America/Cayenne\"?: string; \"America/Cayman\"?: string; \"America/Chicago\"?: string; \"America/Chihuahua\"?: string; \"America/Ciudad_Juarez\"?: string; \"America/Coral_Harbour\"?: string; \"America/Cordoba\"?: string; \"America/Costa_Rica\"?: string; \"America/Creston\"?: string; \"America/Cuiaba\"?: string; \"America/Curacao\"?: string; \"America/Danmarkshavn\"?: string; \"America/Dawson\"?: string; \"America/Dawson_Creek\"?: string; \"America/Denver\"?: string; \"America/Detroit\"?: string; \"America/Dominica\"?: string; \"America/Edmonton\"?: string; \"America/Eirunepe\"?: string; \"America/El_Salvador\"?: string; \"America/Fort_Nelson\"?: string; \"America/Fortaleza\"?: string; \"America/Glace_Bay\"?: string; \"America/Godthab\"?: string; \"America/Goose_Bay\"?: string; \"America/Grand_Turk\"?: string; \"America/Grenada\"?: string; \"America/Guadeloupe\"?: string; \"America/Guatemala\"?: string; \"America/Guayaquil\"?: string; \"America/Guyana\"?: string; \"America/Halifax\"?: string; \"America/Havana\"?: string; \"America/Hermosillo\"?: string; \"America/Indiana/Indianapolis\"?: string; \"America/Indiana/Knox\"?: string; \"America/Indiana/Marengo\"?: string; \"America/Indiana/Petersburg\"?: string; \"America/Indiana/Tell_City\"?: string; \"America/Indiana/Vevay\"?: string; \"America/Indiana/Vincennes\"?: string; \"America/Indiana/Winamac\"?: string; \"America/Indianapolis\"?: string; \"America/Inuvik\"?: string; \"America/Iqaluit\"?: string; \"America/Jamaica\"?: string; \"America/Jujuy\"?: string; \"America/Juneau\"?: string; \"America/Kentucky/Louisville\"?: string; \"America/Kentucky/Monticello\"?: string; \"America/Kralendijk\"?: string; \"America/La_Paz\"?: string; \"America/Lima\"?: string; \"America/Los_Angeles\"?: string; \"America/Louisville\"?: string; \"America/Lower_Princes\"?: string; \"America/Maceio\"?: string; \"America/Managua\"?: string; \"America/Manaus\"?: string; \"America/Marigot\"?: string; \"America/Martinique\"?: string; \"America/Matamoros\"?: string; \"America/Mazatlan\"?: string; \"America/Mendoza\"?: string; \"America/Menominee\"?: string; \"America/Merida\"?: string; \"America/Metlakatla\"?: string; \"America/Mexico_City\"?: string; \"America/Miquelon\"?: string; \"America/Moncton\"?: string; \"America/Monterrey\"?: string; \"America/Montevideo\"?: string; \"America/Montreal\"?: string; \"America/Montserrat\"?: string; \"America/Nassau\"?: string; \"America/New_York\"?: string; \"America/Nipigon\"?: string; \"America/Nome\"?: string; \"America/Noronha\"?: string; \"America/North_Dakota/Beulah\"?: string; \"America/North_Dakota/Center\"?: string; \"America/North_Dakota/New_Salem\"?: string; \"America/Nuuk\"?: string; \"America/Ojinaga\"?: string; \"America/Panama\"?: string; \"America/Pangnirtung\"?: string; \"America/Paramaribo\"?: string; \"America/Phoenix\"?: string; \"America/Port-au-Prince\"?: string; \"America/Port_of_Spain\"?: string; \"America/Porto_Velho\"?: string; \"America/Puerto_Rico\"?: string; \"America/Punta_Arenas\"?: string; \"America/Rainy_River\"?: string; \"America/Rankin_Inlet\"?: string; \"America/Recife\"?: string; \"America/Regina\"?: string; \"America/Resolute\"?: string; \"America/Rio_Branco\"?: string; \"America/Santa_Isabel\"?: string; \"America/Santarem\"?: string; \"America/Santiago\"?: string; \"America/Santo_Domingo\"?: string; \"America/Sao_Paulo\"?: string; \"America/Scoresbysund\"?: string; \"America/Sitka\"?: string; \"America/St_Barthelemy\"?: string; \"America/St_Johns\"?: string; \"America/St_Kitts\"?: string; \"America/St_Lucia\"?: string; \"America/St_Thomas\"?: string; \"America/St_Vincent\"?: string; \"America/Swift_Current\"?: string; \"America/Tegucigalpa\"?: string; \"America/Thule\"?: string; \"America/Thunder_Bay\"?: string; \"America/Tijuana\"?: string; \"America/Toronto\"?: string; \"America/Tortola\"?: string; \"America/Vancouver\"?: string; \"America/Whitehorse\"?: string; \"America/Winnipeg\"?: string; \"America/Yakutat\"?: string; \"America/Yellowknife\"?: string; \"Antarctica/Casey\"?: string; \"Antarctica/Davis\"?: string; \"Antarctica/DumontDUrville\"?: string; \"Antarctica/Macquarie\"?: string; \"Antarctica/Mawson\"?: string; \"Antarctica/McMurdo\"?: string; \"Antarctica/Palmer\"?: string; \"Antarctica/Rothera\"?: string; \"Antarctica/Syowa\"?: string; \"Antarctica/Troll\"?: string; \"Antarctica/Vostok\"?: string; \"Arctic/Longyearbyen\"?: string; \"Asia/Aden\"?: string; \"Asia/Almaty\"?: string; \"Asia/Amman\"?: string; \"Asia/Anadyr\"?: string; \"Asia/Aqtau\"?: string; \"Asia/Aqtobe\"?: string; \"Asia/Ashgabat\"?: string; \"Asia/Atyrau\"?: string; \"Asia/Baghdad\"?: string; \"Asia/Bahrain\"?: string; \"Asia/Baku\"?: string; \"Asia/Bangkok\"?: string; \"Asia/Barnaul\"?: string; \"Asia/Beirut\"?: string; \"Asia/Bishkek\"?: string; \"Asia/Brunei\"?: string; \"Asia/Calcutta\"?: string; \"Asia/Chita\"?: string; \"Asia/Choibalsan\"?: string; \"Asia/Colombo\"?: string; \"Asia/Damascus\"?: string; \"Asia/Dhaka\"?: string; \"Asia/Dili\"?: string; \"Asia/Dubai\"?: string; \"Asia/Dushanbe\"?: string; \"Asia/Famagusta\"?: string; \"Asia/Gaza\"?: string; \"Asia/Hebron\"?: string; \"Asia/Ho_Chi_Minh\"?: string; \"Asia/Hong_Kong\"?: string; \"Asia/Hovd\"?: string; \"Asia/Irkutsk\"?: string; \"Asia/Jakarta\"?: string; \"Asia/Jayapura\"?: string; \"Asia/Jerusalem\"?: string; \"Asia/Kabul\"?: string; \"Asia/Kamchatka\"?: string; \"Asia/Karachi\"?: string; \"Asia/Katmandu\"?: string; \"Asia/Kathmandu\"?: string; \"Asia/Khandyga\"?: string; \"Asia/Kolkata\"?: string; \"Asia/Krasnoyarsk\"?: string; \"Asia/Kuala_Lumpur\"?: string; \"Asia/Kuching\"?: string; \"Asia/Kuwait\"?: string; \"Asia/Macau\"?: string; \"Asia/Magadan\"?: string; \"Asia/Makassar\"?: string; \"Asia/Manila\"?: string; \"Asia/Muscat\"?: string; \"Asia/Nicosia\"?: string; \"Asia/Novokuznetsk\"?: string; \"Asia/Novosibirsk\"?: string; \"Asia/Omsk\"?: string; \"Asia/Oral\"?: string; \"Asia/Phnom_Penh\"?: string; \"Asia/Pontianak\"?: string; \"Asia/Pyongyang\"?: string; \"Asia/Qatar\"?: string; \"Asia/Qostanay\"?: string; \"Asia/Qyzylorda\"?: string; \"Asia/Rangoon\"?: string; \"Asia/Riyadh\"?: string; \"Asia/Saigon\"?: string; \"Asia/Sakhalin\"?: string; \"Asia/Samarkand\"?: string; \"Asia/Seoul\"?: string; \"Asia/Shanghai\"?: string; \"Asia/Singapore\"?: string; \"Asia/Srednekolymsk\"?: string; \"Asia/Taipei\"?: string; \"Asia/Tashkent\"?: string; \"Asia/Tbilisi\"?: string; \"Asia/Tehran\"?: string; \"Asia/Thimphu\"?: string; \"Asia/Tokyo\"?: string; \"Asia/Tomsk\"?: string; \"Asia/Ulaanbaatar\"?: string; \"Asia/Urumqi\"?: string; \"Asia/Ust-Nera\"?: string; \"Asia/Vientiane\"?: string; \"Asia/Vladivostok\"?: string; \"Asia/Yakutsk\"?: string; \"Asia/Yangon\"?: string; \"Asia/Yekaterinburg\"?: string; \"Asia/Yerevan\"?: string; \"Atlantic/Azores\"?: string; \"Atlantic/Bermuda\"?: string; \"Atlantic/Canary\"?: string; \"Atlantic/Cape_Verde\"?: string; \"Atlantic/Faeroe\"?: string; \"Atlantic/Faroe\"?: string; \"Atlantic/Madeira\"?: string; \"Atlantic/Reykjavik\"?: string; \"Atlantic/South_Georgia\"?: string; \"Atlantic/St_Helena\"?: string; \"Atlantic/Stanley\"?: string; \"Australia/Adelaide\"?: string; \"Australia/Brisbane\"?: string; \"Australia/Broken_Hill\"?: string; \"Australia/Currie\"?: string; \"Australia/Darwin\"?: string; \"Australia/Eucla\"?: string; \"Australia/Hobart\"?: string; \"Australia/Lindeman\"?: string; \"Australia/Lord_Howe\"?: string; \"Australia/Melbourne\"?: string; \"Australia/Perth\"?: string; \"Australia/Sydney\"?: string; \"Europe/Amsterdam\"?: string; \"Etc/GMT\"?: string; \"Etc/GMT+1\"?: string; \"Etc/GMT+10\"?: string; \"Etc/GMT+11\"?: string; \"Etc/GMT+12\"?: string; \"Etc/GMT+2\"?: string; \"Etc/GMT+3\"?: string; \"Etc/GMT+4\"?: string; \"Etc/GMT+5\"?: string; \"Etc/GMT+6\"?: string; \"Etc/GMT+7\"?: string; \"Etc/GMT+8\"?: string; \"Etc/GMT+9\"?: string; \"Etc/GMT-1\"?: string; \"Etc/GMT-10\"?: string; \"Etc/GMT-11\"?: string; \"Etc/GMT-12\"?: string; \"Etc/GMT-13\"?: string; \"Etc/GMT-14\"?: string; \"Etc/GMT-2\"?: string; \"Etc/GMT-3\"?: string; \"Etc/GMT-4\"?: string; \"Etc/GMT-5\"?: string; \"Etc/GMT-6\"?: string; \"Etc/GMT-7\"?: string; \"Etc/GMT-8\"?: string; \"Etc/GMT-9\"?: string; \"Etc/UTC\"?: string; \"Europe/Andorra\"?: string; \"Europe/Astrakhan\"?: string; \"Europe/Athens\"?: string; \"Europe/Belgrade\"?: string; \"Europe/Berlin\"?: string; \"Europe/Bratislava\"?: string; \"Europe/Brussels\"?: string; \"Europe/Bucharest\"?: string; \"Europe/Budapest\"?: string; \"Europe/Busingen\"?: string; \"Europe/Chisinau\"?: string; \"Europe/Copenhagen\"?: string; \"Europe/Dublin\"?: string; \"Europe/Gibraltar\"?: string; \"Europe/Guernsey\"?: string; \"Europe/Helsinki\"?: string; \"Europe/Isle_of_Man\"?: string; \"Europe/Istanbul\"?: string; \"Europe/Jersey\"?: string; \"Europe/Kaliningrad\"?: string; \"Europe/Kiev\"?: string; \"Europe/Kirov\"?: string; \"Europe/Kyiv\"?: string; \"Europe/Lisbon\"?: string; \"Europe/Ljubljana\"?: string; \"Europe/London\"?: string; \"Europe/Luxembourg\"?: string; \"Europe/Madrid\"?: string; \"Europe/Malta\"?: string; \"Europe/Mariehamn\"?: string; \"Europe/Minsk\"?: string; \"Europe/Monaco\"?: string; \"Europe/Moscow\"?: string; \"Europe/Oslo\"?: string; \"Europe/Paris\"?: string; \"Europe/Podgorica\"?: string; \"Europe/Prague\"?: string; \"Europe/Riga\"?: string; \"Europe/Rome\"?: string; \"Europe/Samara\"?: string; \"Europe/San_Marino\"?: string; \"Europe/Sarajevo\"?: string; \"Europe/Saratov\"?: string; \"Europe/Simferopol\"?: string; \"Europe/Skopje\"?: string; \"Europe/Sofia\"?: string; \"Europe/Stockholm\"?: string; \"Europe/Tallinn\"?: string; \"Europe/Tirane\"?: string; \"Europe/Ulyanovsk\"?: string; \"Europe/Uzhgorod\"?: string; \"Europe/Vaduz\"?: string; \"Europe/Vatican\"?: string; \"Europe/Vienna\"?: string; \"Europe/Vilnius\"?: string; \"Europe/Volgograd\"?: string; \"Europe/Warsaw\"?: string; \"Europe/Zagreb\"?: string; \"Europe/Zaporozhye\"?: string; \"Europe/Zurich\"?: string; \"Indian/Antananarivo\"?: string; Factory?: string; \"Indian/Chagos\"?: string; \"Indian/Christmas\"?: string; \"Indian/Cocos\"?: string; \"Indian/Comoro\"?: string; \"Indian/Kerguelen\"?: string; \"Indian/Mahe\"?: string; \"Indian/Maldives\"?: string; \"Indian/Mauritius\"?: string; \"Indian/Mayotte\"?: string; \"Indian/Reunion\"?: string; \"Pacific/Apia\"?: string; \"Pacific/Auckland\"?: string; \"Pacific/Bougainville\"?: string; \"Pacific/Chatham\"?: string; \"Pacific/Easter\"?: string; \"Pacific/Efate\"?: string; \"Pacific/Enderbury\"?: string; \"Pacific/Fakaofo\"?: string; \"Pacific/Fiji\"?: string; \"Pacific/Funafuti\"?: string; \"Pacific/Galapagos\"?: string; \"Pacific/Gambier\"?: string; \"Pacific/Guadalcanal\"?: string; \"Pacific/Guam\"?: string; \"Pacific/Honolulu\"?: string; \"Pacific/Johnston\"?: string; \"Pacific/Kanton\"?: string; \"Pacific/Kiritimati\"?: string; \"Pacific/Kosrae\"?: string; \"Pacific/Kwajalein\"?: string; \"Pacific/Majuro\"?: string; \"Pacific/Marquesas\"?: string; \"Pacific/Midway\"?: string; \"Pacific/Nauru\"?: string; \"Pacific/Niue\"?: string; \"Pacific/Norfolk\"?: string; \"Pacific/Noumea\"?: string; \"Pacific/Pago_Pago\"?: string; \"Pacific/Palau\"?: string; \"Pacific/Pitcairn\"?: string; \"Pacific/Ponape\"?: string; \"Pacific/Port_Moresby\"?: string; \"Pacific/Rarotonga\"?: string; \"Pacific/Saipan\"?: string; \"Pacific/Tahiti\"?: string; \"Pacific/Tarawa\"?: string; \"Pacific/Tongatapu\"?: string; \"Pacific/Truk\"?: string; \"Pacific/Wake\"?: string; \"Pacific/Wallis\"?: string; Africa?: string; America?: string; Antarctica?: string; Arctic?: string; Asia?: string; Atlantic?: string; Australia?: string; Europe?: string; Global?: string; Indian?: string; Pacific?: string; AD?: string; AE?: string; AF?: string; AG?: string; AI?: string; AL?: string; AM?: string; AO?: string; AQ?: string; AR?: string; AS?: string; AT?: string; AU?: string; AW?: string; AX?: string; AZ?: string; BA?: string; BB?: string; BD?: string; BE?: string; BF?: string; BG?: string; BH?: string; BI?: string; BJ?: string; BL?: string; BM?: string; BN?: string; BO?: string; BQ?: string; BR?: string; BS?: string; BT?: string; BV?: string; BW?: string; BY?: string; BZ?: string; CA?: string; CC?: string; CD?: string; CF?: string; CG?: string; CH?: string; CI?: string; CK?: string; CL?: string; CM?: string; CN?: string; CO?: string; CR?: string; CU?: string; CV?: string; CW?: string; CX?: string; CY?: string; CZ?: string; DE?: string; DJ?: string; DK?: string; DM?: string; DO?: string; DZ?: string; EC?: string; EE?: string; EG?: string; EH?: string; ER?: string; ES?: string; ET?: string; FI?: string; FJ?: string; FK?: string; FM?: string; FO?: string; FR?: string; GA?: string; GB?: string; GD?: string; GE?: string; GF?: string; GG?: string; GH?: string; GI?: string; GL?: string; GM?: string; GN?: string; GP?: string; GQ?: string; GR?: string; GS?: string; GT?: string; GU?: string; GW?: string; GY?: string; HK?: string; HM?: string; HN?: string; HR?: string; HT?: string; HU?: string; ID?: string; IE?: string; IL?: string; IM?: string; IN?: string; IO?: string; IQ?: string; IR?: string; IS?: string; IT?: string; JE?: string; JM?: string; JO?: string; JP?: string; KE?: string; KG?: string; KH?: string; KI?: string; KM?: string; KN?: string; KP?: string; KR?: string; KW?: string; KY?: string; KZ?: string; LA?: string; LB?: string; LC?: string; LI?: string; LK?: string; LR?: string; LS?: string; LT?: string; LU?: string; LV?: string; LY?: string; MA?: string; MC?: string; MD?: string; ME?: string; MF?: string; MG?: string; MH?: string; MK?: string; ML?: string; MM?: string; MN?: string; MO?: string; MP?: string; MQ?: string; MR?: string; MS?: string; MT?: string; MU?: string; MV?: string; MW?: string; MX?: string; MY?: string; MZ?: string; NA?: string; NC?: string; NE?: string; NF?: string; NG?: string; NI?: string; NL?: string; NO?: string; NP?: string; NR?: string; NU?: string; NZ?: string; OM?: string; PA?: string; PE?: string; PF?: string; PG?: string; PH?: string; PK?: string; PL?: string; PM?: string; PN?: string; PR?: string; PS?: string; PT?: string; PW?: string; PY?: string; QA?: string; RE?: string; RO?: string; RS?: string; RU?: string; RW?: string; SA?: string; SB?: string; SC?: string; SD?: string; SE?: string; SG?: string; SH?: string; SI?: string; SJ?: string; SK?: string; SL?: string; SM?: string; SN?: string; SO?: string; SR?: string; SS?: string; ST?: string; SV?: string; SX?: string; SY?: string; SZ?: string; TC?: string; TD?: string; TF?: string; TG?: string; TH?: string; TJ?: string; TK?: string; TL?: string; TM?: string; TN?: string; TO?: string; TR?: string; TT?: string; TV?: string; TW?: string; TZ?: string; UA?: string; UG?: string; UM?: string; US?: string; UY?: string; UZ?: string; VA?: string; VC?: string; VE?: string; VG?: string; VI?: string; VN?: string; VU?: string; WF?: string; WS?: string; YE?: string; YT?: string; ZA?: string; ZM?: string; ZW?: string; }",
                    "references": {
                        "Partial": {
                            "location": "global",
                            "id": "global::Partial"
                        },
                        "InputTimeZoneMessages": {
                            "location": "import",
                            "path": "./assets/input-time-zone/t9n",
                            "id": "src/components/input-time-zone/assets/input-time-zone/t9n/index.d.ts::InputTimeZoneMessages"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Use this property to override individual strings used by the component."
                }
            },
            "mode": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "TimeZoneMode",
                    "resolved": "\"name\" | \"offset\" | \"region\"",
                    "references": {
                        "TimeZoneMode": {
                            "location": "import",
                            "path": "./interfaces",
                            "id": "src/components/input-time-zone/interfaces.d.ts::TimeZoneMode"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [{
                            "name": "default",
                            "text": "\"offset\""
                        }],
                    "text": "This specifies the type of `value` and the associated options presented to the user:\n\nUsing `\"offset\"` will provide options that show timezone offsets.\n\nUsing `\"name\"` will provide options that show the IANA time zone names."
                },
                "attribute": "mode",
                "reflect": true,
                "defaultValue": "\"offset\""
            },
            "offsetStyle": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "OffsetStyle",
                    "resolved": "\"gmt\" | \"user\" | \"utc\"",
                    "references": {
                        "OffsetStyle": {
                            "location": "import",
                            "path": "./interfaces",
                            "id": "src/components/input-time-zone/interfaces.d.ts::OffsetStyle"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [{
                            "name": "default",
                            "text": "\"user\""
                        }],
                    "text": "Specifies how the offset will be displayed, where\n\n`\"user\"` uses `UTC` or `GMT` depending on the user's locale,\n`\"gmt\"` always uses `GMT`, and\n`\"utc\"` always uses `UTC`.\n\nThis only applies to the `offset` mode."
                },
                "attribute": "offset-style",
                "reflect": true,
                "defaultValue": "\"user\""
            },
            "validationMessage": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Specifies the validation message to display under the component."
                },
                "attribute": "validation-message",
                "reflect": false
            },
            "validationIcon": {
                "type": "any",
                "mutable": false,
                "complexType": {
                    "original": "IconNameOrString | boolean",
                    "resolved": "boolean | string",
                    "references": {
                        "IconNameOrString": {
                            "location": "import",
                            "path": "../icon/interfaces",
                            "id": "src/components/icon/interfaces.ts::IconNameOrString"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Specifies the validation icon to display under the component."
                },
                "attribute": "validation-icon",
                "reflect": true
            },
            "validity": {
                "type": "unknown",
                "mutable": true,
                "complexType": {
                    "original": "MutableValidityState",
                    "resolved": "{ valid: boolean; badInput: boolean; customError: boolean; patternMismatch: boolean; rangeOverflow: boolean; rangeUnderflow: boolean; stepMismatch: boolean; tooLong: boolean; tooShort: boolean; typeMismatch: boolean; valueMissing: boolean; }",
                    "references": {
                        "MutableValidityState": {
                            "location": "import",
                            "path": "../../utils/form",
                            "id": "src/utils/form.tsx::MutableValidityState"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [{
                            "name": "readonly",
                            "text": undefined
                        }, {
                            "name": "mdn",
                            "text": "[ValidityState](https://developer.mozilla.org/en-US/docs/Web/API/ValidityState)"
                        }],
                    "text": "The current validation state of the component."
                },
                "defaultValue": "{\n    valid: false,\n    badInput: false,\n    customError: false,\n    patternMismatch: false,\n    rangeOverflow: false,\n    rangeUnderflow: false,\n    stepMismatch: false,\n    tooLong: false,\n    tooShort: false,\n    typeMismatch: false,\n    valueMissing: false,\n  }"
            },
            "name": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Specifies the name of the component.\n\nRequired to pass the component's `value` on form submission."
                },
                "attribute": "name",
                "reflect": true
            },
            "open": {
                "type": "boolean",
                "mutable": true,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "When `true`, displays and positions the component."
                },
                "attribute": "open",
                "reflect": true,
                "defaultValue": "false"
            },
            "overlayPositioning": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "OverlayPositioning",
                    "resolved": "\"absolute\" | \"fixed\"",
                    "references": {
                        "OverlayPositioning": {
                            "location": "import",
                            "path": "../../utils/floating-ui",
                            "id": "src/utils/floating-ui.ts::OverlayPositioning"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Determines the type of positioning to use for the overlaid content.\n\nUsing `\"absolute\"` will work for most cases. The component will be positioned inside of overflowing parent containers and will affect the container's layout.\n\n`\"fixed\"` should be used to escape an overflowing parent container, or when the reference element's `position` CSS property is `\"fixed\"`."
                },
                "attribute": "overlay-positioning",
                "reflect": true,
                "defaultValue": "\"absolute\""
            },
            "referenceDate": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "Date | string",
                    "resolved": "Date | string",
                    "references": {
                        "Date": {
                            "location": "global",
                            "id": "global::Date"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [{
                            "name": "see",
                            "text": "[Date.prototype.toISOString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString)"
                        }],
                    "text": "This `date` will be used as a reference to Daylight Savings Time when creating time zone item groups.\n\nIt can be either a Date instance or a string in ISO format (`\"YYYY-MM-DD\"`, `\"YYYY-MM-DDTHH:MM:SS.SSSZ\"`)."
                },
                "attribute": "reference-date",
                "reflect": false
            },
            "required": {
                "type": "boolean",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [{
                            "name": "internal",
                            "text": undefined
                        }],
                    "text": "When `true`, the component must have a value in order for the form to submit."
                },
                "attribute": "required",
                "reflect": true,
                "defaultValue": "false"
            },
            "scale": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "Scale",
                    "resolved": "\"l\" | \"m\" | \"s\"",
                    "references": {
                        "Scale": {
                            "location": "import",
                            "path": "../interfaces",
                            "id": "src/components/interfaces.ts::Scale"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Specifies the size of the component."
                },
                "attribute": "scale",
                "reflect": true,
                "defaultValue": "\"m\""
            },
            "status": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "Status",
                    "resolved": "\"idle\" | \"invalid\" | \"valid\"",
                    "references": {
                        "Status": {
                            "location": "import",
                            "path": "../interfaces",
                            "id": "src/components/interfaces.ts::Status"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Specifies the status of the input field, which determines message and icons."
                },
                "attribute": "status",
                "reflect": true,
                "defaultValue": "\"idle\""
            },
            "value": {
                "type": "string",
                "mutable": true,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [{
                            "name": "see",
                            "text": "https://www.w3.org/International/core/2005/09/timezone.html#:~:text=What%20is%20a%20%22zone%20offset,or%20%22%2D%22%20from%20UTC."
                        }],
                    "text": "The component's value, where the value is the time zone offset or the difference, in minutes, between the selected time zone and UTC.\n\nIf no value is provided, the user's time zone offset will be selected by default."
                },
                "attribute": "value",
                "reflect": false
            },
            "readOnly": {
                "type": "boolean",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "When `true`, the component's value can be read, but controls are not accessible and the value cannot be modified."
                },
                "attribute": "read-only",
                "reflect": true,
                "defaultValue": "false"
            }
        };
    }
    static get states() {
        return {
            "defaultMessages": {},
            "effectiveLocale": {}
        };
    }
    static get events() {
        return [{
                "method": "calciteInputTimeZoneBeforeClose",
                "name": "calciteInputTimeZoneBeforeClose",
                "bubbles": true,
                "cancelable": false,
                "composed": true,
                "docs": {
                    "tags": [],
                    "text": "Fires when the component is requested to be closed and before the closing transition begins."
                },
                "complexType": {
                    "original": "void",
                    "resolved": "void",
                    "references": {}
                }
            }, {
                "method": "calciteInputTimeZoneBeforeOpen",
                "name": "calciteInputTimeZoneBeforeOpen",
                "bubbles": true,
                "cancelable": false,
                "composed": true,
                "docs": {
                    "tags": [],
                    "text": "Fires when the component is added to the DOM but not rendered, and before the opening transition begins."
                },
                "complexType": {
                    "original": "void",
                    "resolved": "void",
                    "references": {}
                }
            }, {
                "method": "calciteInputTimeZoneChange",
                "name": "calciteInputTimeZoneChange",
                "bubbles": true,
                "cancelable": false,
                "composed": true,
                "docs": {
                    "tags": [],
                    "text": "Fires when the component's `value` changes."
                },
                "complexType": {
                    "original": "void",
                    "resolved": "void",
                    "references": {}
                }
            }, {
                "method": "calciteInputTimeZoneClose",
                "name": "calciteInputTimeZoneClose",
                "bubbles": true,
                "cancelable": false,
                "composed": true,
                "docs": {
                    "tags": [],
                    "text": "Fires after the component is closed and animation is complete."
                },
                "complexType": {
                    "original": "void",
                    "resolved": "void",
                    "references": {}
                }
            }, {
                "method": "calciteInputTimeZoneOpen",
                "name": "calciteInputTimeZoneOpen",
                "bubbles": true,
                "cancelable": false,
                "composed": true,
                "docs": {
                    "tags": [],
                    "text": "Fires after the component is opened and animation is complete."
                },
                "complexType": {
                    "original": "void",
                    "resolved": "void",
                    "references": {}
                }
            }];
    }
    static get methods() {
        return {
            "setFocus": {
                "complexType": {
                    "signature": "() => Promise<void>",
                    "parameters": [],
                    "references": {
                        "Promise": {
                            "location": "global",
                            "id": "global::Promise"
                        }
                    },
                    "return": "Promise<void>"
                },
                "docs": {
                    "text": "",
                    "tags": []
                }
            }
        };
    }
    static get elementRef() { return "el"; }
    static get watchers() {
        return [{
                "propName": "messageOverrides",
                "methodName": "onMessagesChange"
            }, {
                "propName": "messages",
                "methodName": "handleTimeZoneItemPropsChange"
            }, {
                "propName": "mode",
                "methodName": "handleTimeZoneItemPropsChange"
            }, {
                "propName": "referenceDate",
                "methodName": "handleTimeZoneItemPropsChange"
            }, {
                "propName": "open",
                "methodName": "openChanged"
            }, {
                "propName": "value",
                "methodName": "handleValueChange"
            }, {
                "propName": "effectiveLocale",
                "methodName": "effectiveLocaleWatcher"
            }];
    }
}
