// @arcgis/charts-components/dist/cjs/errors.t9n.tr-01b78ee4-c848bf77.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="Çizge yüklenirken bir hata oluştu.",e="Negatif veya sıfır değerlere işlem kaydı dönüştürme uygulanamaz.",r="Negatif değerlere kare kök dönüştürme uygulanamaz.",i="Katman yüklenirken bir hata oluştu. URL = ${ url }. Portal ögesi kimliği = ${ portalItemId }.",l="${ dataPath } benzersiz olmalıdır. ${ seriesName } adlı serinin başka bir seri tarafından zaten kullanılan bir kimliği (${ seriesID }) var.",t="${ dataPath } sayısal olmayan alanda sayım dışı kümeleme gerçekleştiremez.",n="${ dataPath }'de ${ missingProperty } adlı özellik eksik.",d="${ dataPath }, ${ limit } karakterden daha kısa olmamalıdır.",m="${ dataPath }, ${ limit }'den az ögeye sahip olmamalıdır.",s="${ dataPath }, ${ limit }'den fazla ögeye sahip olmamalıdır.",o="${ dataPath }, en az bir adet boşluk olmayan karaktere sahip olmalıdır.",u="${ dataPath }'de ${ additionalProperty } olmamalıdır.",k="${ dataPath }, izin verilen şu değerlerden birine eşit olmalıdır: ${ allowedValues }.",y="${ dataPath }, şunlardan birinin şemasıyla eşleşmelidir: ${ schemaOptions }.",$="Orantısal sembollere sahip dağılım grafikleri desteklenmez. Varsayılan sembol boyutu uygulandı.",h="Girdi verileri okunamadı.",g="${ seriesIndex } dizininde beklenen seri tipi '${ expectedType }' ancak '${ receivedType }' alındı",p="veya";var b={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:r,layerLoadFailure:i,duplicateSeriesID:l,nonNumericAggregation:t,requiredProperty:n,minLength:d,minItems:m,maxItems:s,whiteSpacePattern:o,additionalProperty:u,enumValues:k,anyOfValues:y,bubbleChartValidateMsg:$,queryError:h,invalidSeriesType:g,or:p};

exports.additionalProperty = u;
exports.anyOfValues = y;
exports.bubbleChartValidateMsg = $;
exports['default'] = b;
exports.defaultError = a;
exports.duplicateSeriesID = l;
exports.enumValues = k;
exports.invalidSeriesType = g;
exports.layerLoadFailure = i;
exports.maxItems = s;
exports.minItems = m;
exports.minLength = d;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = r;
exports.nonNumericAggregation = t;
exports.or = p;
exports.queryError = h;
exports.requiredProperty = n;
exports.whiteSpacePattern = o;
