// @arcgis/charts-components/dist/cjs/errors.t9n.vi-1a6ce09c-0acfa127-562cbc94-d64948ef.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="Đã xảy ra lỗi khi tải biểu đồ.",t="Không thể áp dụng phép biến đổi log lên giá trị âm hoặc bằng không.",n="Không thể áp dụng phép biến đổi căn bậc hai lên giá trị âm.",e="Đã xảy ra lỗi khi tải lớp. URL = ${ url }. ID mục cổng thông tin = ${ portalItemId }.",i="${ dataPath } phải là duy nhất. Chuỗi có tên ${ seriesName } có một ID (${ seriesID }) đã được chuỗi khác sử dụng.",h="${ dataPath } không thể thực hiện tổng hợp không đếm trên trường không phải là số.",r="${ dataPath } đang thiếu một tên phù hợp ${ missingProperty }.",g="${ dataPath } không được ngắn hơn ${ limit } ký tự.",s="${ dataPath } không được chứa dưới ${ limit } mục.",o="${ dataPath } không được chứa nhiều hơn ${ limit } mục.",l="${ dataPath } phải chứa ít nhất một ký tự không phải khoảng trắng.",c="${ dataPath } không được chứa ${ additionalProperty }.",d="${ dataPath } phải bằng một trong những giá trị được phép này: ${ allowedValues }.",u="${ dataPath } phải khớp với lược đồ của một trong những thứ sau: ${ schemaOptions }.",m="Biểu đồ phân tán có ký hiệu tỷ lệ không được hỗ trợ. Đã áp dụng kích cỡ ký hiệu theo mặc định.",p="Không thể đọc dữ liệu đầu vào.",$="Loại chuỗi mong đợi tại chỉ mục ${ seriesIndex } là '${ expectedType }' nhưng thay vào đó lại nhận được '${ receivedType }'",y="hoặc";var k={defaultError:a,negativeValueInDataForLogTransformation:t,negativeValueInDataForSqrtTransformation:n,layerLoadFailure:e,duplicateSeriesID:i,nonNumericAggregation:h,requiredProperty:r,minLength:g,minItems:s,maxItems:o,whiteSpacePattern:l,additionalProperty:c,enumValues:d,anyOfValues:u,bubbleChartValidateMsg:m,queryError:p,invalidSeriesType:$,or:y};

exports.additionalProperty = c;
exports.anyOfValues = u;
exports.bubbleChartValidateMsg = m;
exports['default'] = k;
exports.defaultError = a;
exports.duplicateSeriesID = i;
exports.enumValues = d;
exports.invalidSeriesType = $;
exports.layerLoadFailure = e;
exports.maxItems = o;
exports.minItems = s;
exports.minLength = g;
exports.negativeValueInDataForLogTransformation = t;
exports.negativeValueInDataForSqrtTransformation = n;
exports.nonNumericAggregation = h;
exports.or = y;
exports.queryError = p;
exports.requiredProperty = r;
exports.whiteSpacePattern = l;
