// @arcgis/charts-components/dist/cjs/errors.t9n.da-4412c16d-a63a9872.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const e="Der opstod en fejl under indlæsning af diagrammet.",a="Kan ikke anvende log-transformation på negative værdier eller nulværdier.",r="Kan ikke anvende kvadratrod-transformation på negative værdier.",t="Der opstod en fejl under indlæsning af laget. URL = ${ url }. Portalelement-id = ${ portalItemId }.",n="${ dataPath } skal være entydig. Serien med navnet ${ seriesName } har et id (${ seriesID }), der allerede bruges i en anden serie.",i="${ dataPath } kan ikke udføre ikke-antal-aggregering på ikke-numeriske felter.",d="${ dataPath } mangler en egenskab med navnet ${ missingProperty }.",l="${ dataPath } skal ikke være kortere end ${ limit } tegn.",s="${ dataPath } må ikke have færre end ${ limit } elementer.",m="${ dataPath } må ikke have flere end ${ limit } elementer.",o="${ dataPath } skal have mindst et tegn, der ikke er et mellemrumstegn.",k="${ dataPath } må ikke have ${ additionalProperty }.",g="${ dataPath } skal være lig med en af disse tilladte værdier: ${ allowedValues }.",v="${ dataPath } skal matche skemaet fra en af disse: ${ schemaOptions }.",u="Punktdiagrammer med proportionale symboler understøttes ikke. Der er blevet anvendt standardstørrelse på symboler.",$="Kunne ikke læse inputdata.",p="Den forventede serietype ved indekset ${ seriesIndex } er '${ expectedType }', men '${ receivedType }' blev modtaget i stedet for",f="eller";var h={defaultError:e,negativeValueInDataForLogTransformation:a,negativeValueInDataForSqrtTransformation:r,layerLoadFailure:t,duplicateSeriesID:n,nonNumericAggregation:i,requiredProperty:d,minLength:l,minItems:s,maxItems:m,whiteSpacePattern:o,additionalProperty:k,enumValues:g,anyOfValues:v,bubbleChartValidateMsg:u,queryError:$,invalidSeriesType:p,or:f};

exports.additionalProperty = k;
exports.anyOfValues = v;
exports.bubbleChartValidateMsg = u;
exports['default'] = h;
exports.defaultError = e;
exports.duplicateSeriesID = n;
exports.enumValues = g;
exports.invalidSeriesType = p;
exports.layerLoadFailure = t;
exports.maxItems = m;
exports.minItems = s;
exports.minLength = l;
exports.negativeValueInDataForLogTransformation = a;
exports.negativeValueInDataForSqrtTransformation = r;
exports.nonNumericAggregation = i;
exports.or = f;
exports.queryError = $;
exports.requiredProperty = d;
exports.whiteSpacePattern = o;
