// @arcgis/charts-components/dist/cjs/errors.t9n.hu-40e92ed7-6dbd6389-b1e54ae0-af4e9d62.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const e="Hiba történt a diagram betöltése közben.",a="A log transzformáció nem alkalmazható negatív vagy nulla értékre.",t="A négyzetgyök transzformáció nem alkalmazható negatív értékre.",r="Hiba történt a réteg betöltés közben. URL = ${ url }. Portál elemazonosítója = ${ portalItemId }.",n="Egyedinek kell lennie: ${ dataPath } A(z) ${ seriesName } nevű sorozat azonosítóját (${ seriesID }) egy másik sorozat már használja.",l="A(z) ${ dataPath } nem hajthat végre nem megszámlálható összevonást nem numerikus mezőn.",s="A(z) ${ dataPath } esetében hiányzik a(z) ${ missingProperty } nevű tulajdonság.",i="A(z) ${ dataPath } hossza legalább ${ limit } karakter kell legyen.",m="A(z) ${ dataPath } elemeinek száma legalább ${ limit } kell legyen.",o="A(z) ${ dataPath } elemeinek száma legfeljebb ${ limit } lehet.",k="A következőnek legalább egy nem-whitespace karaktert kell tartalmaznia: ${ dataPath }",z="${ dataPath } nem rendelkezhet ezzel: ${ additionalProperty }.",g="${ dataPath } értékének egyeznie kell a következő megengedett értékek egyikével: ${ allowedValues }.",d="A(z) ${ dataPath } értékének egyeznie kell a következők egyikének sémájával: ${ schemaOptions }.",u="A méretarányos szimbólummal rendelkező szórásdiagramok nem támogatottak. Az alapértelmezett szimbólum méret lesz alkalmazva.",y="Nem sikerült beolvasni a bemeneti adatokat.",h="A várt sorozattípus a(z) ${ seriesIndex } indexnél '${ expectedType }', de helyette '${ receivedType }' érkezett.",b="vagy";var v={defaultError:e,negativeValueInDataForLogTransformation:a,negativeValueInDataForSqrtTransformation:t,layerLoadFailure:r,duplicateSeriesID:n,nonNumericAggregation:l,requiredProperty:s,minLength:i,minItems:m,maxItems:o,whiteSpacePattern:k,additionalProperty:z,enumValues:g,anyOfValues:d,bubbleChartValidateMsg:u,queryError:y,invalidSeriesType:h,or:b};

exports.additionalProperty = z;
exports.anyOfValues = d;
exports.bubbleChartValidateMsg = u;
exports['default'] = v;
exports.defaultError = e;
exports.duplicateSeriesID = n;
exports.enumValues = g;
exports.invalidSeriesType = h;
exports.layerLoadFailure = r;
exports.maxItems = o;
exports.minItems = m;
exports.minLength = i;
exports.negativeValueInDataForLogTransformation = a;
exports.negativeValueInDataForSqrtTransformation = t;
exports.nonNumericAggregation = l;
exports.or = b;
exports.queryError = y;
exports.requiredProperty = s;
exports.whiteSpacePattern = k;
