// @arcgis/charts-components/dist/cjs/errors.t9n.ro-ccdd46b5-4f19d396-fba09341.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const e="A survenit o eroare în timpul încărcării diagramei.",a="Nu se poate aplica o transformare de jurnal în valori negative sau zero.",r="Nu se poate aplica o transformare a rădăcinii pătrate în valori negative.",i="A survenit o eroare în timpul încărcării stratului tematic. URL = ${ url }. ID-ul elementului de portal = ${ portalItemId }.",t="${ dataPath } trebuie să fie unic. Seria denumită ${ seriesName } are un ID (${ seriesID }) utilizat deja de către o altă serie.",n="${ dataPath } nu poate efectua agregarea non-numărare pe un câmp non-numeric.",s="${ dataPath } are lipsă o proprietate denumită ${ missingProperty }.",u="${ dataPath } nu trebuie să fie mai scurt de ${ limit } caractere.",o="${ dataPath } nu trebuie să aibă mai puțin de ${ limit } articole.",l="${ dataPath } nu trebuie să aibă mai mult de ${ limit } articole.",d="${ dataPath } trebuie să aibă cel puțin un caracter non-spațiu gol.",m="${ dataPath } nu trebuie să aibă ${ additionalProperty }.",p="${ dataPath } trebuie să fie egal cu una dintre aceste valori permise: ${ allowedValues }.",c="${ dataPath } trebuie să corespundă schemei uneia dintre acestea: ${ schemaOptions }.",g="Proprietățile diagramei de dispersie cu simbolurile proporționale nu sunt acceptate. A fost aplicată dimensiunea implicită a simbolului.",$="A eșuat citirea datelor de intrare.",b="Tipul așteptat al serie la indicele ${ seriesIndex } este „${ expectedType }”, dar a fost primit „${ receivedType }”",P="sau";var h={defaultError:e,negativeValueInDataForLogTransformation:a,negativeValueInDataForSqrtTransformation:r,layerLoadFailure:i,duplicateSeriesID:t,nonNumericAggregation:n,requiredProperty:s,minLength:u,minItems:o,maxItems:l,whiteSpacePattern:d,additionalProperty:m,enumValues:p,anyOfValues:c,bubbleChartValidateMsg:g,queryError:$,invalidSeriesType:b,or:"sau"};

exports.additionalProperty = m;
exports.anyOfValues = c;
exports.bubbleChartValidateMsg = g;
exports['default'] = h;
exports.defaultError = e;
exports.duplicateSeriesID = t;
exports.enumValues = p;
exports.invalidSeriesType = b;
exports.layerLoadFailure = i;
exports.maxItems = l;
exports.minItems = o;
exports.minLength = u;
exports.negativeValueInDataForLogTransformation = a;
exports.negativeValueInDataForSqrtTransformation = r;
exports.nonNumericAggregation = n;
exports.or = P;
exports.queryError = $;
exports.requiredProperty = s;
exports.whiteSpacePattern = d;
