// @arcgis/charts-components/dist/cjs/errors.t9n.ja-1d42dfcd-53791f19-f49fa9de.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="チャートの読み込み中にエラーが発生しました。",e="対数変換は負またはゼロの値に適用できません。",t="平方根変換は負の値に適用できません。",r="レイヤーの読み込み中にエラーが発生しました。 URL = ${ url } ポータル アイテム ID = ${ portalItemId }",i="${ dataPath } は一意である必要があります。 ${ seriesName } という系列の ID (${ seriesID }) は、別の系列ですでに使用されています。",s="${ dataPath } は、数値以外のフィールドで、個数以外の集約を実行することはできません。",n="${ dataPath } に ${ missingProperty } というプロパティがありません。",o="${ dataPath } は ${ limit } 文字以上である必要があります。",d="${ dataPath } のアイテムは ${ limit } 以上にする必要があります。",l="${ dataPath } のアイテムは ${ limit } 以下にする必要があります。",u="${ dataPath } には空白以外の文字が 1 つ以上含まれている必要があります。",m="${ dataPath } が ${ additionalProperty } を持つことはできません。",$="${ dataPath } は、次のいずれかの許可された値と等しい必要があります: ${ allowedValues }",p="${ dataPath } は、次のいずれかのスキーマと一致する必要があります: ${ schemaOptions }",P="比例シンボルを使用した散布図はサポートされていません。 デフォルトのシンボル サイズが適用されています。",g="入力データを読み取れません。",h="インデックス ${ seriesIndex } で必要な系列タイプは ${ expectedType } ですが、代わりに ${ receivedType } が指定されました",y="または";var I={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:t,layerLoadFailure:r,duplicateSeriesID:i,nonNumericAggregation:s,requiredProperty:n,minLength:o,minItems:d,maxItems:l,whiteSpacePattern:u,additionalProperty:m,enumValues:$,anyOfValues:p,bubbleChartValidateMsg:P,queryError:g,invalidSeriesType:h,or:"または"};

exports.additionalProperty = m;
exports.anyOfValues = p;
exports.bubbleChartValidateMsg = P;
exports['default'] = I;
exports.defaultError = a;
exports.duplicateSeriesID = i;
exports.enumValues = $;
exports.invalidSeriesType = h;
exports.layerLoadFailure = r;
exports.maxItems = l;
exports.minItems = d;
exports.minLength = o;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = t;
exports.nonNumericAggregation = s;
exports.or = y;
exports.queryError = g;
exports.requiredProperty = n;
exports.whiteSpacePattern = u;
