// @arcgis/charts-components/dist/cjs/errors.t9n.lt-63e1dc12-0e9eab9a-3008e9fa.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="Įkeliant diagramą įvyko klaida.",i="Negalima pritaikyti log. transformacijos neigiamoms arba nulinėms reikšmėms.",e="Negalima pritaikyti kvadratinės šaknies transformacijos neigiamoms reikšmėms.",t="Įkeliant sluoksnį įvyko klaida. URL = ${ url }. Portalo elemento ID = ${ portalItemId }.",r="${ dataPath } turi būti unikalus. Serijos, kurios pavadinimas ${ seriesName }, ID (${ seriesID }) jau naudojamas kitai serijai.",s="${ dataPath } negali atikti neskaičiavimo agregavimo ne skaičių lauke.",n="${ dataPath } trūksta savybės pavadinimu ${ missingProperty }.",o="${ dataPath } negali būti trumpesnis negu ${ limit } simb.",u="${ dataPath } negali turėti mažiau negu ${ limit } elementų.",l="${ dataPath } negali turėti daugiau negu ${ limit } elementų.",m="${ dataPath } turi sudaryti bent vienas simbolis, kuris nėra tarpas.",d="${ dataPath } negali sudaryti ${ additionalProperty }.",g="${ dataPath } turi būti lygus vienai iš šių leidžiamų reikšmių: ${ allowedValues }.",y="${ dataPath } turi sutapti su viena iš šių schemų: ${ schemaOptions }.",p="Sklaidos diagramos su proporciniais simboliais nėra palaikomos. Pritaikytas numatytasis simbolių dydis.",k="Nepavyko nuskaityti įvesties duomenų.",$="Numatytas serijos tipas indeksui ${ seriesIndex } yra ${ expectedType }, bet vietoj to gauta ${ receivedType }",v="arba";var P={defaultError:a,negativeValueInDataForLogTransformation:i,negativeValueInDataForSqrtTransformation:e,layerLoadFailure:t,duplicateSeriesID:r,nonNumericAggregation:s,requiredProperty:n,minLength:o,minItems:u,maxItems:l,whiteSpacePattern:m,additionalProperty:d,enumValues:g,anyOfValues:y,bubbleChartValidateMsg:p,queryError:k,invalidSeriesType:$,or:v};

exports.additionalProperty = d;
exports.anyOfValues = y;
exports.bubbleChartValidateMsg = p;
exports['default'] = P;
exports.defaultError = a;
exports.duplicateSeriesID = r;
exports.enumValues = g;
exports.invalidSeriesType = $;
exports.layerLoadFailure = t;
exports.maxItems = l;
exports.minItems = u;
exports.minLength = o;
exports.negativeValueInDataForLogTransformation = i;
exports.negativeValueInDataForSqrtTransformation = e;
exports.nonNumericAggregation = s;
exports.or = v;
exports.queryError = k;
exports.requiredProperty = n;
exports.whiteSpacePattern = m;
