// @arcgis/charts-components/dist/cjs/errors.t9n.pt-BR-61e39f00-c6d99994-617f0d79.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="Ocorreu um erro ao carregar o gráfico.",e="Não é possível aplicar transformação de log para valores negativos ou zero.",r="Não é possível aplicar transformação de raiz quadrada para valores negativos.",o="Ocorreu um erro ao carregar a camada. URL = ${ url }. D de item do portal = ${ portalItemId }.",t="${ dataPath } deve ser único. A série denominada ${ seriesName } tem um id (${ seriesID }) que já é usado por outra série.",s="${ dataPath } não deve realizar agregação sem contagem em campo não numérico.",i="${ dataPath } está sem uma propriedade denominada ${ missingProperty }.",d="${ dataPath } não deve ser menor que ${ limit } caracteres.",n="${ dataPath } não deve ter menos que ${ limit } itens.",m="${ dataPath } não deve ter mais de ${ limit } itens.",l="${ dataPath } deve ter pelo menos um caractere diferente de espaço em branco.",u="${ dataPath } não deve ter ${ additionalProperty }.",p="${ dataPath } deve ser igual a um destes valores permitidos: ${ allowedValues }.",c="${ dataPath } deve corresponder ao esquema de um destes: ${ schemaOptions }.",g="Gráficos de dispersão com símbolos proporcionais não são suportados. O tamanho padrão do símbolo foi aplicado.",v="Falha ao ler os dados de entrada.",$="O tipo de série esperado no índice${ seriesIndex } é '${ expectedType }' mas '${ receivedType }' foi recebido em seu lugar",h="ou";var P={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:r,layerLoadFailure:o,duplicateSeriesID:t,nonNumericAggregation:s,requiredProperty:i,minLength:d,minItems:n,maxItems:m,whiteSpacePattern:l,additionalProperty:u,enumValues:p,anyOfValues:c,bubbleChartValidateMsg:g,queryError:v,invalidSeriesType:$,or:"ou"};

exports.additionalProperty = u;
exports.anyOfValues = c;
exports.bubbleChartValidateMsg = g;
exports['default'] = P;
exports.defaultError = a;
exports.duplicateSeriesID = t;
exports.enumValues = p;
exports.invalidSeriesType = $;
exports.layerLoadFailure = o;
exports.maxItems = m;
exports.minItems = n;
exports.minLength = d;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = r;
exports.nonNumericAggregation = s;
exports.or = h;
exports.queryError = v;
exports.requiredProperty = i;
exports.whiteSpacePattern = l;
