// @arcgis/charts-components/dist/cjs/errors.t9n.it-cde6174a-2558ff6b-2835dd51.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const e="Si è verificato un errore durante il caricamento del grafico.",a="Impossibile applicare la trasformazione di log su valori negativi o zero.",i="Impossibile applicare la trasformazione di radice quadrata su valori negativi.",r="Si è verificato un errore durante il caricamento del layer. URL = ${ url }. ID elemento del portale = ${ portalItemId }.",t="${ dataPath } deve essere univoco. La serie denominata ${ seriesName } ha un ID (${ seriesID }) che è già utilizzato da un'altra serie.",o="${ dataPath } non può effettuare l'aggregazione senza conteggio su un campo non numerico.",n="A ${ dataPath } manca una proprietà denominata ${ missingProperty }.",s="${ dataPath } non deve essere più corto di ${ limit } caratteri.",l="${ dataPath } non deve contenere meno di ${ limit } elementi.",d="${ dataPath } non deve contenere più di ${ limit } elementi.",u="${ dataPath } deve contenere almeno un carattere.",m="${ dataPath } non deve contenere ${ additionalProperty }.",p="${ dataPath } deve essere uguale a uno di questi valori consentiti: ${ allowedValues }.",c="${ dataPath } deve corrispondere allo schema di uno di questi‎: ${ schemaOptions }.",g="I grafici a dispersione con simboli proporzionali non sono supportati. Sono state applicate delle dimensioni simbolo predefinite.",v="Impossibile leggere i dati di input.",$="Il tipo di serie atteso all'indice ${ seriesIndex } è '${ expectedType }' ma invece è stato ricevuto '${ receivedType }'",I="oppure";var h={defaultError:e,negativeValueInDataForLogTransformation:a,negativeValueInDataForSqrtTransformation:i,layerLoadFailure:r,duplicateSeriesID:t,nonNumericAggregation:o,requiredProperty:n,minLength:s,minItems:l,maxItems:d,whiteSpacePattern:u,additionalProperty:m,enumValues:p,anyOfValues:c,bubbleChartValidateMsg:g,queryError:v,invalidSeriesType:$,or:I};

exports.additionalProperty = m;
exports.anyOfValues = c;
exports.bubbleChartValidateMsg = g;
exports['default'] = h;
exports.defaultError = e;
exports.duplicateSeriesID = t;
exports.enumValues = p;
exports.invalidSeriesType = $;
exports.layerLoadFailure = r;
exports.maxItems = d;
exports.minItems = l;
exports.minLength = s;
exports.negativeValueInDataForLogTransformation = a;
exports.negativeValueInDataForSqrtTransformation = i;
exports.nonNumericAggregation = o;
exports.or = I;
exports.queryError = v;
exports.requiredProperty = n;
exports.whiteSpacePattern = u;
