// @arcgis/charts-components/dist/cjs/errors.t9n.ca-db180ea1-fff7c626-fd4a3439.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="S'ha produït un error en carregar el gràfic.",e="No es pot aplicar una transformació del registre a valors negatius o de zero.",r="No es pot aplicar una transformació d'arrel quadrada a valors negatius.",t="S'ha produït un error en carregar la capa. URL = ${ url }. ID d'element del portal = ${ portalItemId }.",s="L'element ${ dataPath } ha de ser únic. L'ID (${ seriesID }) de la sèrie anomenada ${ seriesName } ja l'utilitza una altra sèrie.",i="${ dataPath } no pot dur a terme una agregació que no sigui de recompte en un camp que no sigui numèric.",n="A ${ dataPath } li falta una propietat anomenada ${ missingProperty }.",o="${ dataPath } no pot tenir menys de ${ limit } caràcters.",d="${ dataPath } no pot tenir menys de ${ limit } elements.",l="${ dataPath } no pot tenir més de ${ limit } elements.",u="${ dataPath } ha de tenir almenys un caràcter que no sigui un espai en blanc.",m="${ dataPath } no pot tenir ${ additionalProperty }.",p="${ dataPath } ha de ser igual a un d'aquests valors permesos: ${ allowedValues }.",c="${ dataPath } ha de coincidir amb l'esquema d'un d'aquests: ${ schemaOptions }.",g="No s'admeten gràfics de dispersió amb símbols proporcionals. S'ha aplicat la mida de símbol per defecte.",h="Error de lectura de les dades d'entrada.",$='El tipus de sèrie esperat a l\'índex ${ seriesIndex } és "${ expectedType }", però s\'ha rebut el tipus "${ receivedType }"',y="o";var P={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:r,layerLoadFailure:t,duplicateSeriesID:s,nonNumericAggregation:i,requiredProperty:n,minLength:o,minItems:d,maxItems:l,whiteSpacePattern:u,additionalProperty:m,enumValues:p,anyOfValues:c,bubbleChartValidateMsg:g,queryError:h,invalidSeriesType:$,or:"o"};

exports.additionalProperty = m;
exports.anyOfValues = c;
exports.bubbleChartValidateMsg = g;
exports['default'] = P;
exports.defaultError = a;
exports.duplicateSeriesID = s;
exports.enumValues = p;
exports.invalidSeriesType = $;
exports.layerLoadFailure = t;
exports.maxItems = l;
exports.minItems = d;
exports.minLength = o;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = r;
exports.nonNumericAggregation = i;
exports.or = y;
exports.queryError = h;
exports.requiredProperty = n;
exports.whiteSpacePattern = u;
