// @arcgis/charts-components/dist/cjs/errors.t9n.tr-01b78ee4-07aadb86-ba0c1200.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="Çizge yüklenirken bir hata oluştu.",e="Negatif veya sıfır değerlere işlem kaydı dönüştürme uygulanamaz.",r="Negatif değerlere kare kök dönüştürme uygulanamaz.",i="Katman yüklenirken bir hata oluştu. URL = ${ url }. Portal ögesi kimliği = ${ portalItemId }.",t="${ dataPath } benzersiz olmalıdır. ${ seriesName } adlı serinin başka bir seri tarafından zaten kullanılan bir kimliği (${ seriesID }) var.",l="${ dataPath } sayısal olmayan alanda sayım dışı kümeleme gerçekleştiremez.",n="${ dataPath }'de ${ missingProperty } adlı özellik eksik.",d="${ dataPath }, ${ limit } karakterden daha kısa olmamalıdır.",s="${ dataPath }, ${ limit }'den az ögeye sahip olmamalıdır.",m="${ dataPath }, ${ limit }'den fazla ögeye sahip olmamalıdır.",o="${ dataPath }, en az bir adet boşluk olmayan karaktere sahip olmalıdır.",u="${ dataPath }'de ${ additionalProperty } olmamalıdır.",y="${ dataPath }, izin verilen şu değerlerden birine eşit olmalıdır: ${ allowedValues }.",g="${ dataPath }, şunlardan birinin şemasıyla eşleşmelidir: ${ schemaOptions }.",k="Orantısal sembollere sahip dağılım grafikleri desteklenmez. Varsayılan sembol boyutu uygulandı.",h="Girdi verileri okunamadı.",$="${ seriesIndex } dizininde beklenen seri tipi '${ expectedType }' ancak '${ receivedType }' alındı",p="veya";var b={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:r,layerLoadFailure:i,duplicateSeriesID:t,nonNumericAggregation:l,requiredProperty:n,minLength:d,minItems:s,maxItems:m,whiteSpacePattern:o,additionalProperty:u,enumValues:y,anyOfValues:g,bubbleChartValidateMsg:k,queryError:h,invalidSeriesType:$,or:p};

exports.additionalProperty = u;
exports.anyOfValues = g;
exports.bubbleChartValidateMsg = k;
exports['default'] = b;
exports.defaultError = a;
exports.duplicateSeriesID = t;
exports.enumValues = y;
exports.invalidSeriesType = $;
exports.layerLoadFailure = i;
exports.maxItems = m;
exports.minItems = s;
exports.minLength = d;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = r;
exports.nonNumericAggregation = l;
exports.or = p;
exports.queryError = h;
exports.requiredProperty = n;
exports.whiteSpacePattern = o;
