// @arcgis/charts-components/dist/cjs/errors.t9n.lv-df11315d-7c6d1628-d7ab25be-0d918c0b.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="Ielādējot diagrammu, radās kļūda.",e="Žurnālfaila transformāciju nevar piemērot negatīvām vai nulles vērtībām.",t="Kvadrātsaknes transformāciju nevar lietot negatīvām vērtībām.",r="Ielādējot slāni, radās kļūda. Vietrādis (URL) = ${ url }. Portāla vienības ID = ${ portalItemId }.",i="${ dataPath } jābūt unikālam. Sērijai ar nosaukumu ${ seriesName } ir ID (${ seriesID }), ko jau izmanto cita sērija.",s="${ dataPath } nevar veikt apkopošanu laukā, kurā nav skaitlisku vērtību.",n="${ dataPath } trūkst parametra ar nosaukumu ${ missingProperty }.",o="${ dataPath } nedrīkst būt īsāks par ${ limit } rakstzīmēm.",m="${ dataPath } nedrīkst būt mazāk par ${ limit } vienībām.",d="${ dataPath } nedrīkst būt vairāk par ${ limit } vienībām.",l="${ dataPath } jābūt vismaz vienai rakstzīmei, kas nav atstarpe.",u="${ dataPath } nedrīkst būt ${ additionalProperty }.",v="${ dataPath } jābūt vienādam ar kādu no šīm atļautajām vērtībām: ${ allowedValues }.",k="${ dataPath } jāatbilst kādai no šādām shēmām: ${ schemaOptions }.",p="Izkliedes diagrammas ar proporcionāliem simboliem netiek atbalstītas. Ticis piemērots noklusējuma simbola izmērs.",b="Neizdevās nolasīt ievades datus.",$="Indeksa ${ seriesIndex } prognozētais sērijas veids ir “${ expectedType }”, bet tā vietā tika saņemts “${ receivedType }”",g="vai";var I={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:t,layerLoadFailure:r,duplicateSeriesID:i,nonNumericAggregation:s,requiredProperty:n,minLength:o,minItems:m,maxItems:d,whiteSpacePattern:l,additionalProperty:u,enumValues:v,anyOfValues:k,bubbleChartValidateMsg:p,queryError:b,invalidSeriesType:$,or:"vai"};

exports.additionalProperty = u;
exports.anyOfValues = k;
exports.bubbleChartValidateMsg = p;
exports['default'] = I;
exports.defaultError = a;
exports.duplicateSeriesID = i;
exports.enumValues = v;
exports.invalidSeriesType = $;
exports.layerLoadFailure = r;
exports.maxItems = d;
exports.minItems = m;
exports.minLength = o;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = t;
exports.nonNumericAggregation = s;
exports.or = g;
exports.queryError = b;
exports.requiredProperty = n;
exports.whiteSpacePattern = l;
