// @arcgis/charts-components/dist/cjs/errors.t9n.ru-b4412898-d449f481-19b6c421.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="При загрузке диаграммы произошла ошибка.",e="Невозможно применить Логарифмическое преобразование к отрицательным или нулевым значениям.",t="Невозможно применить преобразование по методу квадратного корня к отрицательным значениям.",r="При загрузке слоя произошла ошибка. URL = ${ url }. ID элемента портала = ${ portalItemId }.",i="${ dataPath } должен быть уникальным. У серии с именем ${ seriesName } есть ID (${ seriesID }), который уже используется другой серией.",s="${ dataPath } не должен выполнять не количественное агрегирование для не числового поля.",n="В ${ dataPath } отсутствует свойство с именем ${ missingProperty }.",o="${ dataPath } не должен быть короче, чем ${ limit } символов.",d="В ${ dataPath } должно быть не менее ${ limit } элементов.",l="В ${ dataPath } не должно быть более ${ limit } элементов.",u="${ dataPath } должен содержать, как минимум, один символ, не являющейся пробелом.",m="${ dataPath } не должны иметь ${ additionalProperty }",$="${ dataPath } должен быть равен одному из этих допустимых значений: ${ allowedValues }.",p="${ dataPath } должен совпадать со схемой одного из следующих элементов: ${ schemaOptions }.",P="Диаграммы рассеяния с пропорциональными символами не поддерживаются. Будет использован размер символа по умолчанию.",g="Ошибка чтения входных данных.",h="Ожидаемый тип рядов в индексе ${ seriesIndex } - '${ expectedType }' но вместо этого было получено '${ receivedType }'",y="или";var I={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:t,layerLoadFailure:r,duplicateSeriesID:i,nonNumericAggregation:s,requiredProperty:n,minLength:o,minItems:d,maxItems:l,whiteSpacePattern:u,additionalProperty:m,enumValues:$,anyOfValues:p,bubbleChartValidateMsg:P,queryError:g,invalidSeriesType:h,or:"или"};

exports.additionalProperty = m;
exports.anyOfValues = p;
exports.bubbleChartValidateMsg = P;
exports['default'] = I;
exports.defaultError = a;
exports.duplicateSeriesID = i;
exports.enumValues = $;
exports.invalidSeriesType = h;
exports.layerLoadFailure = r;
exports.maxItems = l;
exports.minItems = d;
exports.minLength = o;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = t;
exports.nonNumericAggregation = s;
exports.or = y;
exports.queryError = g;
exports.requiredProperty = n;
exports.whiteSpacePattern = u;
