// @arcgis/charts-components/dist/cjs/errors.t9n.nb-83d135ad-a65a1170-f28ead81.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const e="Det oppsto en feil under innlasting av diagrammet.",a="Loggtransformasjon kan ikke brukes på negative verdier eller nullverdier.",r="Kvadratrottransformasjon kan ikke brukes på negative verdier.",t="Det oppsto en feil under innlasting av laget. URL = ${ url }. Portalelement-ID = ${ portalItemId }.",n="${ dataPath } må være unik. Serien med navnet ${ seriesName } har en ID (${ seriesID }) som allerede brukes av en annen serie.",i="${ dataPath } kan ikke utføre aggregering av annet enn antall på ikke-numeriske felt.",s="${ dataPath } mangler en egenskap ved navn ${ missingProperty }.",l="${ dataPath } må ikke være kortere enn ${ limit } tegn.",o="${ dataPath } kan ikke inneholde færre enn ${ limit } elementer.",d="${ limit } kan ikke inneholde flere enn ${ dataPath } elementer.",m="${ dataPath } må inneholde minst ett tegn som ikke er mellomrom.",k="${ dataPath } kan ikke ha ${ additionalProperty }.",u="${ dataPath } må være lik én av følgende tillatte verdier: ${ allowedValues }.",g="${ dataPath } må samsvare med skjemaet for én av følgende: ${ schemaOptions }.",v="Punktdiagrammer med proporsjonale symboler støttes ikke. Standard symbolstørrelse er brukt.",p="Kan ikke lese inndata.",$="Forventet serietype ved indeks ${ seriesIndex } er ${ expectedType }, men ${ receivedType } ble mottatt i stedet",h="eller";var f={defaultError:e,negativeValueInDataForLogTransformation:a,negativeValueInDataForSqrtTransformation:r,layerLoadFailure:t,duplicateSeriesID:n,nonNumericAggregation:i,requiredProperty:s,minLength:l,minItems:o,maxItems:d,whiteSpacePattern:m,additionalProperty:k,enumValues:u,anyOfValues:g,bubbleChartValidateMsg:v,queryError:p,invalidSeriesType:$,or:h};

exports.additionalProperty = k;
exports.anyOfValues = g;
exports.bubbleChartValidateMsg = v;
exports['default'] = f;
exports.defaultError = e;
exports.duplicateSeriesID = n;
exports.enumValues = u;
exports.invalidSeriesType = $;
exports.layerLoadFailure = t;
exports.maxItems = d;
exports.minItems = o;
exports.minLength = l;
exports.negativeValueInDataForLogTransformation = a;
exports.negativeValueInDataForSqrtTransformation = r;
exports.nonNumericAggregation = i;
exports.or = h;
exports.queryError = p;
exports.requiredProperty = s;
exports.whiteSpacePattern = m;
