// @arcgis/charts-components/dist/cjs/errors.t9n.nl-5ff392d7-e3258f43.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const e="Er is een fout opgetreden bij het laden van de grafiek.",a="Kan logtransformatie niet toepassen op negatieve of nulwaarden.",t="Kan transformatie vierkantswortel niet toepassen op negatieve waarden.",n="Er is een fout opgetreden bij het laden van de laag. URL = ${ url }. Portal item ID = ${ portalItemId }.",i="${ dataPath } moet uniek zijn. De serie genaamd ${ seriesName } heeft een ID (${ seriesID }) die al wordt gebruikt door een andere serie.",r="${ dataPath } kan geen niet-telbare aggregatie uitvoeren op een niet-numeriek veld.",o="${ dataPath } mist een eigenschap met de naam ${ missingProperty }.",d="${ dataPath } mag niet korter zijn dan ${ limit } tekens.",s="${ dataPath } mag niet minder dan ${ limit } items hebben.",m="${ dataPath } mag niet meer dan ${ limit } items hebben.",l="${ dataPath } moet minstens één non-whitespace teken hebben.",g="${ dataPath } mag geen ${ additionalProperty } hebben.",p="${ dataPath } moet gelijk zijn aan een van deze toegelaten waarden: ${ allowedValues }.",h="${ dataPath } moet overeenkomen met het schema van een van deze: ${ schemaOptions }.",$="Spreidingsdiagrammen met proportionele symbolen worden niet ondersteund. Standaardsymboolgrootte is toegepast.",u="Kan invoergegevens niet lezen.",v="Het verwachte serietype bij index ${ seriesIndex } is '${ expectedType }' maar '${ receivedType }' is in plaats daarvan ontvangen",b="of";var P={defaultError:e,negativeValueInDataForLogTransformation:a,negativeValueInDataForSqrtTransformation:t,layerLoadFailure:n,duplicateSeriesID:i,nonNumericAggregation:r,requiredProperty:o,minLength:d,minItems:s,maxItems:m,whiteSpacePattern:l,additionalProperty:g,enumValues:p,anyOfValues:h,bubbleChartValidateMsg:$,queryError:u,invalidSeriesType:v,or:"of"};

exports.additionalProperty = g;
exports.anyOfValues = h;
exports.bubbleChartValidateMsg = $;
exports['default'] = P;
exports.defaultError = e;
exports.duplicateSeriesID = i;
exports.enumValues = p;
exports.invalidSeriesType = v;
exports.layerLoadFailure = n;
exports.maxItems = m;
exports.minItems = s;
exports.minLength = d;
exports.negativeValueInDataForLogTransformation = a;
exports.negativeValueInDataForSqrtTransformation = t;
exports.nonNumericAggregation = r;
exports.or = b;
exports.queryError = u;
exports.requiredProperty = o;
exports.whiteSpacePattern = l;
