// @arcgis/charts-components/dist/cjs/errors.t9n.cs-105948e7-deae329f.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="Při načítání grafu došlo k chybě.",e="Nelze uplatnit logaritmickou transformaci na záporné nebo nulové hodnoty.",t="Nelze uplatnit transformaci odmocniny na záporné hodnoty.",o="Při načítání vrstvy došlo k chybě. URL = ${ url }. ID položky portálu = ${ portalItemId }.",n="${ dataPath } musí být jedinečné. Řada s názvem ${ seriesName } má ID (${ seriesID }), které již používá jiná řada.",r="${ dataPath } nemůže provádět jiné seskupení než podle počtu na jiném než numerickém poli.",s="V ${ dataPath } chybí vlastnost s názvem ${ missingProperty }.",i="${ dataPath } nesmí být kratší než ${ limit } znaků.",d="${ dataPath } nesmí obsahovat méně než ${ limit } položek.",l="${ dataPath } nesmí obsahovat více než ${ limit } položek.",m="${ dataPath } musí obsahovat alespoň jeden znak jiný než mezera.",p="${ dataPath } nesmí obsahovat ${ additionalProperty }.",u="${ dataPath } se musí rovnat jedné z těchto povolených hodnot: ${ allowedValues }.",h="${ dataPath } se musí shodovat s jedním z těchto schémat: ${ schemaOptions }.",v="Rozptýlené bodové grafy s proporcionálními symboly nejsou podporovány. Byla použita výchozí velikost symbolu.",y="Vstupní data se nepodařilo přečíst.",$="Očekávaný typ řady s indexem ${ seriesIndex } je '${ expectedType }', ale obdrželi jsme '${ receivedType }'",c="nebo";var b={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:t,layerLoadFailure:o,duplicateSeriesID:n,nonNumericAggregation:r,requiredProperty:s,minLength:i,minItems:d,maxItems:l,whiteSpacePattern:m,additionalProperty:p,enumValues:u,anyOfValues:h,bubbleChartValidateMsg:v,queryError:y,invalidSeriesType:$,or:c};

exports.additionalProperty = p;
exports.anyOfValues = h;
exports.bubbleChartValidateMsg = v;
exports['default'] = b;
exports.defaultError = a;
exports.duplicateSeriesID = n;
exports.enumValues = u;
exports.invalidSeriesType = $;
exports.layerLoadFailure = o;
exports.maxItems = l;
exports.minItems = d;
exports.minLength = i;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = t;
exports.nonNumericAggregation = r;
exports.or = c;
exports.queryError = y;
exports.requiredProperty = s;
exports.whiteSpacePattern = m;
