// @arcgis/charts-components/dist/cjs/errors.t9n.ru-b4412898-b644feea.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="При загрузке диаграммы произошла ошибка.",e="Невозможно применить Логарифмическое преобразование к отрицательным или нулевым значениям.",t="Невозможно применить преобразование по методу квадратного корня к отрицательным значениям.",r="При загрузке слоя произошла ошибка. URL = ${ url }. ID элемента портала = ${ portalItemId }.",i="${ dataPath } должен быть уникальным. У серии с именем ${ seriesName } есть ID (${ seriesID }), который уже используется другой серией.",d="${ dataPath } не должен выполнять не количественное агрегирование для не числового поля.",n="В ${ dataPath } отсутствует свойство с именем ${ missingProperty }.",o="${ dataPath } не должен быть короче, чем ${ limit } символов.",$="В ${ dataPath } должно быть не менее ${ limit } элементов.",l="В ${ dataPath } не должно быть более ${ limit } элементов.",s="${ dataPath } должен содержать, как минимум, один символ, не являющейся пробелом.",m="${ dataPath } не должны иметь ${ additionalProperty }",u="${ dataPath } должен быть равен одному из этих допустимых значений: ${ allowedValues }.",P="${ dataPath } должен совпадать со схемой одного из следующих элементов: ${ schemaOptions }.",h="Диаграммы рассеяния с пропорциональными символами не поддерживаются. Будет использован размер символа по умолчанию.",p="Ошибка чтения входных данных.",I="Ожидаемый тип рядов в индексе ${ seriesIndex } - '${ expectedType }' но вместо этого было получено '${ receivedType }'",y="или";var g={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:t,layerLoadFailure:r,duplicateSeriesID:i,nonNumericAggregation:d,requiredProperty:n,minLength:o,minItems:$,maxItems:l,whiteSpacePattern:s,additionalProperty:m,enumValues:u,anyOfValues:P,bubbleChartValidateMsg:h,queryError:p,invalidSeriesType:I,or:"или"};

exports.additionalProperty = m;
exports.anyOfValues = P;
exports.bubbleChartValidateMsg = h;
exports['default'] = g;
exports.defaultError = a;
exports.duplicateSeriesID = i;
exports.enumValues = u;
exports.invalidSeriesType = I;
exports.layerLoadFailure = r;
exports.maxItems = l;
exports.minItems = $;
exports.minLength = o;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = t;
exports.nonNumericAggregation = d;
exports.or = y;
exports.queryError = p;
exports.requiredProperty = n;
exports.whiteSpacePattern = s;
