// @arcgis/charts-components/dist/cjs/errors.t9n.id-a5bc1429-f2463130.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="Terjadi kesalahan saat memuat grafik.",i="Tidak dapat menerapkan transformasi log ke nilai negatif atau nol.",e="Tidak dapat menerapkan transformasi akar pangkat dua ke nilai negatif.",t="Terjadi kesalahan saat memuat layer. URL = ${ url }. ID item portal = ${ portalItemId }.",n="${ dataPath } harus unik. Rangkaian yang bernama ${ seriesName } memiliki ID (${ seriesID }) yang sudah digunakan oleh rangkaian lain.",r="${ dataPath } tidak dapat melakukan agregasi non-hitungan di kolom non-numerik.",d="${ dataPath } tidak memiliki properti bernama ${ missingProperty }.",l="${ dataPath } tidak boleh lebih pendek dari ${ limit } karakter.",s="${ dataPath } tidak boleh memiliki kurang dari ${ limit } item.",m="${ dataPath } tidak boleh memiliki lebih dari ${ limit } item.",o="${ dataPath } harus memiliki setidaknya satu karakter non-spasi.",k="${ dataPath } tidak boleh memiliki ${ additionalProperty }.",u="${ dataPath } harus sama dengan salah satu nilai yang diizinkan ini: ${ allowedValues }.",h="${ dataPath } harus cocok dengan skema salah satu dari ini: ${ schemaOptions }.",g="Scatter plot dengan simbol proporsional tidak didukung. Ukuran simbol default sudah diterapkan.",p="Gagal membaca input data.",$="Jenis seri yang diharapkan pada indeks ${ seriesIndex } adalah '${ expectedType }' tetapi '${ receivedType }' diterima sebagai gantinya",y="atau";var b={defaultError:a,negativeValueInDataForLogTransformation:i,negativeValueInDataForSqrtTransformation:e,layerLoadFailure:t,duplicateSeriesID:n,nonNumericAggregation:r,requiredProperty:d,minLength:l,minItems:s,maxItems:m,whiteSpacePattern:o,additionalProperty:k,enumValues:u,anyOfValues:h,bubbleChartValidateMsg:g,queryError:p,invalidSeriesType:$,or:y};

exports.additionalProperty = k;
exports.anyOfValues = h;
exports.bubbleChartValidateMsg = g;
exports['default'] = b;
exports.defaultError = a;
exports.duplicateSeriesID = n;
exports.enumValues = u;
exports.invalidSeriesType = $;
exports.layerLoadFailure = t;
exports.maxItems = m;
exports.minItems = s;
exports.minLength = l;
exports.negativeValueInDataForLogTransformation = i;
exports.negativeValueInDataForSqrtTransformation = e;
exports.nonNumericAggregation = r;
exports.or = y;
exports.queryError = p;
exports.requiredProperty = d;
exports.whiteSpacePattern = o;
