// @arcgis/charts-components/dist/cjs/errors.t9n.da-4412c16d-1c88cd92-17a700f7-bc1e3b74.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const e="Der opstod en fejl under indlæsning af diagrammet.",a="Kan ikke anvende log-transformation på negative værdier eller nulværdier.",r="Kan ikke anvende kvadratrod-transformation på negative værdier.",t="Der opstod en fejl under indlæsning af laget. URL = ${ url }. Portalelement-id = ${ portalItemId }.",n="${ dataPath } skal være entydig. Serien med navnet ${ seriesName } har et id (${ seriesID }), der allerede bruges i en anden serie.",i="${ dataPath } kan ikke udføre ikke-antal-aggregering på ikke-numeriske felter.",s="${ dataPath } mangler en egenskab med navnet ${ missingProperty }.",d="${ dataPath } skal ikke være kortere end ${ limit } tegn.",l="${ dataPath } må ikke have færre end ${ limit } elementer.",o="${ dataPath } må ikke have flere end ${ limit } elementer.",m="${ dataPath } skal have mindst et tegn, der ikke er et mellemrumstegn.",g="${ dataPath } må ikke have ${ additionalProperty }.",u="${ dataPath } skal være lig med en af disse tilladte værdier: ${ allowedValues }.",k="${ dataPath } skal matche skemaet fra en af disse: ${ schemaOptions }.",v="Punktdiagrammer med proportionale symboler understøttes ikke. Der er blevet anvendt standardstørrelse på symboler.",p="Kunne ikke læse inputdata.",f="Den forventede serietype ved indekset ${ seriesIndex } er '${ expectedType }', men '${ receivedType }' blev modtaget i stedet for",$="eller";var h={defaultError:e,negativeValueInDataForLogTransformation:a,negativeValueInDataForSqrtTransformation:r,layerLoadFailure:t,duplicateSeriesID:n,nonNumericAggregation:i,requiredProperty:s,minLength:d,minItems:l,maxItems:o,whiteSpacePattern:m,additionalProperty:g,enumValues:u,anyOfValues:k,bubbleChartValidateMsg:v,queryError:p,invalidSeriesType:f,or:$};

exports.additionalProperty = g;
exports.anyOfValues = k;
exports.bubbleChartValidateMsg = v;
exports['default'] = h;
exports.defaultError = e;
exports.duplicateSeriesID = n;
exports.enumValues = u;
exports.invalidSeriesType = f;
exports.layerLoadFailure = t;
exports.maxItems = o;
exports.minItems = l;
exports.minLength = d;
exports.negativeValueInDataForLogTransformation = a;
exports.negativeValueInDataForSqrtTransformation = r;
exports.nonNumericAggregation = i;
exports.or = $;
exports.queryError = p;
exports.requiredProperty = s;
exports.whiteSpacePattern = m;
