// @arcgis/charts-components/dist/cjs/errors.t9n.uk-b25edecd-8945d522-dc6dd68b.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="Під час завантаження діаграми виникла помилка.",e="Неможливо застосувати перетворення логарифму до від’ємних або нульових значень.",t="Неможливо застосувати перетворення квадратного кореня до від’ємних значень.",r="Під час завантаження шару виникла помилка. URL = ${ url }. ІД елемента порталу = ${ portalItemId }.",i="${ dataPath } повинен бути унікальним. Серія з назвою ${ seriesName } має ідентифікатор (${ seriesID }), який вже використовується іншою серією.",s="${ dataPath } не може виконувати агрегування без підрахунку в нечисловому полі.",n="${ dataPath } не має властивості з назвою ${ missingProperty }.",o="${ dataPath } не може мати менше ${ limit } символів.",d="${ dataPath } не може мати менше ${ limit } елементів.",l="${ dataPath } не може мати більше ${ limit } елементів.",u="${ dataPath } повинен мати принаймні один символ, який не є пробілом.",m="${ dataPath } не може мати ${ additionalProperty }.",$="${ dataPath } має дорівнювати одному з цих допустимих значень: ${ allowedValues }.",p="${ dataPath } має збігатися зі схемою одного з цих: ${ schemaOptions }.",P="Діаграми розсіювання з пропорційними символами не підтримуються. Застосовано розмір символу за замовчуванням.",g="Помилка читання вхідних даних.",h="Очікуваним типом серії з індексом ${ seriesIndex } є '${ expectedType }', але замість нього було отримано '${ receivedType }'",y="або";var I={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:t,layerLoadFailure:r,duplicateSeriesID:i,nonNumericAggregation:s,requiredProperty:n,minLength:o,minItems:d,maxItems:l,whiteSpacePattern:u,additionalProperty:m,enumValues:$,anyOfValues:p,bubbleChartValidateMsg:P,queryError:g,invalidSeriesType:h,or:"або"};

exports.additionalProperty = m;
exports.anyOfValues = p;
exports.bubbleChartValidateMsg = P;
exports['default'] = I;
exports.defaultError = a;
exports.duplicateSeriesID = i;
exports.enumValues = $;
exports.invalidSeriesType = h;
exports.layerLoadFailure = r;
exports.maxItems = l;
exports.minItems = d;
exports.minLength = o;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = t;
exports.nonNumericAggregation = s;
exports.or = y;
exports.queryError = g;
exports.requiredProperty = n;
exports.whiteSpacePattern = u;
