// @arcgis/charts-components/dist/cjs/errors.t9n.ar-0213ec3f-6c4a1aa4-b48ac51c-564f1236.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="حدث خطأ أثناء تحميل المخطط.",e="يتعذر تطبيق تحويل السِّجل إلى قيم سلبية أو إلى صفر.",t="يتعذر تطبيق تحويل الجذر التربيعي إلى قيم سلبية.",r="حدث خطأ أثناء تحميل الطبقة. URL = ${ url }. معرف عنصر البوابة = ${ portalItemId }.",i="يجب أن يكون ${ dataPath } مميزًا. السلسلة التي اسمها ${ seriesName } تتضمن معرفًا (${ seriesID }) مستخدمًا بالفعل من قبل سلسلة أخرى.",s="يتعذر على ${ dataPath } إجراء تجميع من دون حساب في حقل غير رقمي.",n="يتفقد ${ dataPath } خاصية باسم ${ missingProperty }.",o="يجب ألا يكون ${ dataPath } أقصر من ${ limit } من الحروف.",l="يجب ألا يحتوي ${ dataPath } على أقل من ${ limit } من العناصر.",d="يجب ألا يحتوي ${ dataPath } على أكثر من ${ limit } من العناصر.",u="يجب أن يتضمن ${ dataPath } حرفًا واحدًا على الأقل ليس مسافة بيضاء.",m="يجب أن يحتوي ${ dataPath } على ${ additionalProperty }.",$="يجب أن يكون ${ dataPath } مساويًا لإحدى القيم هذه المسموح بها: ${ allowedValues }",p="يجب أن يتطابق ${ dataPath } مع مخطط أحد ما يلي: ${ schemaOptions }.",P="Scatterplots ذات الرموز المتناسبة غير مدعومة. تم تطبيق حجم الرمز الافتراضي.",g="فشلت قراءة البيانات المدخلة.",h="نوع السلسلة المتوقع في الفهرس ${ seriesIndex } هو '${ expectedType }' ولكن تم استلام '${ receivedType }' بدلاً من ذلك",y="أو";var I={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:t,layerLoadFailure:r,duplicateSeriesID:i,nonNumericAggregation:s,requiredProperty:n,minLength:o,minItems:l,maxItems:d,whiteSpacePattern:u,additionalProperty:m,enumValues:$,anyOfValues:p,bubbleChartValidateMsg:P,queryError:g,invalidSeriesType:h,or:"أو"};

exports.additionalProperty = m;
exports.anyOfValues = p;
exports.bubbleChartValidateMsg = P;
exports['default'] = I;
exports.defaultError = a;
exports.duplicateSeriesID = i;
exports.enumValues = $;
exports.invalidSeriesType = h;
exports.layerLoadFailure = r;
exports.maxItems = d;
exports.minItems = l;
exports.minLength = o;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = t;
exports.nonNumericAggregation = s;
exports.or = y;
exports.queryError = g;
exports.requiredProperty = n;
exports.whiteSpacePattern = u;
