// @arcgis/charts-components/dist/cjs/errors.t9n.de-d07a5b5e-35add0f4-8d318b83-bd5151cf.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const e="Beim Laden des Diagramms ist ein Fehler aufgetreten.",a="Eine logarithmische Transformation kann nicht auf negative oder Nullwerte angewendet werden.",t="Eine Quadratwurzeltransformation kann nicht auf negative Werte angewendet werden.",n="Beim Laden des Layers ist ein Fehler aufgetreten. URL = ${ url } Portal-Element-ID = ${ portalItemId }.",r="${ dataPath } muss eindeutig sein. Die Reihe mit dem Namen ${ seriesName } hat eine ID (${ seriesID }), die schon von einer anderen Reihe verwendet wird.",i="${ dataPath } kann keine Aggregation, die nicht anzahlbasiert ist, für nicht numerische Felder durchführen.",s="${ dataPath } fehlt eine Eigenschaft mit dem Namen ${ missingProperty }.",d="${ dataPath } darf nicht kürzer als ${ limit } Zeichen sein.",l="${ dataPath } darf nicht weniger als ${ limit } Elemente enthalten.",m="${ dataPath } darf nicht mehr als ${ limit } Elemente enthalten.",o="${ dataPath } muss mindestens ein Zeichen enthalten, bei dem es sich nicht um ein Leerstellenzeichen handelt.",h="${ dataPath } darf nicht ${ additionalProperty } aufweisen.",u="${ dataPath } muss einem der folgenden zulässigen Werte entsprechen: ${ allowedValues }.",g="${ dataPath } muss mit einem der folgenden Schemas übereinstimmen: ${ schemaOptions }.",c="Scatterplots mit proportionalen Symbolen werden nicht unterstützt. Die standardmäßige Symbolgröße wurde angewendet.",f="Fehler beim Lesen der Eingabedaten.",p='Der erwartete Serientyp bei Index ${ seriesIndex } lautet "${ expectedType }", stattdessen wurde jedoch "${ receivedType }" empfangen.',$="oder";var y={defaultError:e,negativeValueInDataForLogTransformation:a,negativeValueInDataForSqrtTransformation:t,layerLoadFailure:n,duplicateSeriesID:r,nonNumericAggregation:i,requiredProperty:s,minLength:d,minItems:l,maxItems:m,whiteSpacePattern:o,additionalProperty:h,enumValues:u,anyOfValues:g,bubbleChartValidateMsg:c,queryError:f,invalidSeriesType:p,or:$};

exports.additionalProperty = h;
exports.anyOfValues = g;
exports.bubbleChartValidateMsg = c;
exports['default'] = y;
exports.defaultError = e;
exports.duplicateSeriesID = r;
exports.enumValues = u;
exports.invalidSeriesType = p;
exports.layerLoadFailure = n;
exports.maxItems = m;
exports.minItems = l;
exports.minLength = d;
exports.negativeValueInDataForLogTransformation = a;
exports.negativeValueInDataForSqrtTransformation = t;
exports.nonNumericAggregation = i;
exports.or = $;
exports.queryError = f;
exports.requiredProperty = s;
exports.whiteSpacePattern = o;
