// @arcgis/charts-components/dist/cjs/errors.t9n.sl-896f0a80-7a89f86a-3ee1cf57-8b9f0ab0.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="Med nalaganjem grafikona je prišlo do napake.",e="Pretvorbe dnevnika ni mogoče uporabiti za negativne ali ničelne vrednosti.",i="Pretvorbe kvadratnega korena ni mogoče uporabiti za negativne vrednosti.",t="Med nalaganjem sloja je prišlo do napake. URL = ${ url }. ID elementa portala = ${ portalItemId }.",r="${ dataPath } mora biti edinstveno. Serija z imenom ${ seriesName } ima ID (${ seriesID }), ki ga že uporablja druga serija.",n="${ dataPath } ne more izvajati združevanja brez števila na neštevilskem polju.",o="${ dataPath } manjka lastnost imenovana ${ missingProperty }.",s="${ dataPath } ne sme biti krajše od ${ limit } znakov.",l="${ dataPath } ne sme imeti manj kot ${ limit } elementov.",m="${ dataPath } ne sme imeti več kot ${ limit } elementov.",d="${ dataPath } mora vsebovati vsaj en znak, ki ni presledek.",p="${ dataPath } ne sme imeti ${ additionalProperty }.",v="${ dataPath } mora biti enak eni od teh dovoljenih vrednosti: ${ allowedValues }.",u="${ dataPath } se mora ujemati s shemo ene od teh: ${ schemaOptions }.",g="Raztreseni grafikoni s proporcionalnimi simboli niso podprti. Uporabljena je bila privzeta velikost simbolov.",$="Branje vhodnih podatkov ni uspelo.",h="Pričakovani tip niza z indeksom ${ seriesIndex } je '${ expectedType }', ampak '${ receivedType }' je bil prejet namesto",j="ali";var k={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:i,layerLoadFailure:t,duplicateSeriesID:r,nonNumericAggregation:n,requiredProperty:o,minLength:s,minItems:l,maxItems:m,whiteSpacePattern:d,additionalProperty:p,enumValues:v,anyOfValues:u,bubbleChartValidateMsg:g,queryError:$,invalidSeriesType:h,or:"ali"};

exports.additionalProperty = p;
exports.anyOfValues = u;
exports.bubbleChartValidateMsg = g;
exports['default'] = k;
exports.defaultError = a;
exports.duplicateSeriesID = r;
exports.enumValues = v;
exports.invalidSeriesType = h;
exports.layerLoadFailure = t;
exports.maxItems = m;
exports.minItems = l;
exports.minLength = s;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = i;
exports.nonNumericAggregation = n;
exports.or = j;
exports.queryError = $;
exports.requiredProperty = o;
exports.whiteSpacePattern = d;
