// @arcgis/charts-components/dist/cjs/errors.t9n.lt-63e1dc12-13cf5973.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="Įkeliant diagramą įvyko klaida.",i="Negalima pritaikyti log. transformacijos neigiamoms arba nulinėms reikšmėms.",e="Negalima pritaikyti kvadratinės šaknies transformacijos neigiamoms reikšmėms.",t="Įkeliant sluoksnį įvyko klaida. URL = ${ url }. Portalo elemento ID = ${ portalItemId }.",s="${ dataPath } turi būti unikalus. Serijos, kurios pavadinimas ${ seriesName }, ID (${ seriesID }) jau naudojamas kitai serijai.",r="${ dataPath } negali atikti neskaičiavimo agregavimo ne skaičių lauke.",n="${ dataPath } trūksta savybės pavadinimu ${ missingProperty }.",o="${ dataPath } negali būti trumpesnis negu ${ limit } simb.",m="${ dataPath } negali turėti mažiau negu ${ limit } elementų.",u="${ dataPath } negali turėti daugiau negu ${ limit } elementų.",l="${ dataPath } turi sudaryti bent vienas simbolis, kuris nėra tarpas.",d="${ dataPath } negali sudaryti ${ additionalProperty }.",g="${ dataPath } turi būti lygus vienai iš šių leidžiamų reikšmių: ${ allowedValues }.",k="${ dataPath } turi sutapti su viena iš šių schemų: ${ schemaOptions }.",p="Sklaidos diagramos su proporciniais simboliais nėra palaikomos. Pritaikytas numatytasis simbolių dydis.",y="Nepavyko nuskaityti įvesties duomenų.",$="Numatytas serijos tipas indeksui ${ seriesIndex } yra ${ expectedType }, bet vietoj to gauta ${ receivedType }",v="arba";var P={defaultError:a,negativeValueInDataForLogTransformation:i,negativeValueInDataForSqrtTransformation:e,layerLoadFailure:t,duplicateSeriesID:s,nonNumericAggregation:r,requiredProperty:n,minLength:o,minItems:m,maxItems:u,whiteSpacePattern:l,additionalProperty:d,enumValues:g,anyOfValues:k,bubbleChartValidateMsg:p,queryError:y,invalidSeriesType:$,or:v};

exports.additionalProperty = d;
exports.anyOfValues = k;
exports.bubbleChartValidateMsg = p;
exports['default'] = P;
exports.defaultError = a;
exports.duplicateSeriesID = s;
exports.enumValues = g;
exports.invalidSeriesType = $;
exports.layerLoadFailure = t;
exports.maxItems = u;
exports.minItems = m;
exports.minLength = o;
exports.negativeValueInDataForLogTransformation = i;
exports.negativeValueInDataForSqrtTransformation = e;
exports.nonNumericAggregation = r;
exports.or = v;
exports.queryError = y;
exports.requiredProperty = n;
exports.whiteSpacePattern = l;
