// @arcgis/charts-components/dist/cjs/errors.t9n.es-5078b06f-654258a6-68ac387b-6cc34ae6.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const e="Se ha producido un error al cargar el gráfico.",a="No se puede aplicar una transformación del registro a valores negativos o de cero.",r="No se puede aplicar una transformación de raíz cuadrada a valores negativos.",o="Se ha producido un error al cargar la capa. URL = ${ url }. Id. de elemento de portal = ${ portalItemId }.",n="El elemento ${ dataPath } debe ser único. La serie denominada ${ seriesName } tiene un Id. (${ seriesID }) que ya se utiliza en otra serie.",t="${ dataPath } no puede realizar una agregación no numérica en un campo no numérico.",i="A ${ dataPath } le falta la propiedad denominada ${ missingProperty }.",s="${ dataPath } no debe tener menos de ${ limit } caracteres.",d="${ dataPath } no debe tener menos de ${ limit } elementos.",l="${ dataPath } no debe tener más de ${ limit } elementos.",u="${ dataPath } debe tener al menos un carácter que no sea un espacio en blanco.",m="${ dataPath } no debe tener ${ additionalProperty }",c="${ dataPath } debe ser igual a uno de los siguientes valores permitidos: ${ allowedValues }.",p="${ dataPath } debe coincidir con el esquema de uno de estos: ${ schemaOptions }.",g="No se admiten diagramas de dispersión con símbolos proporcionales. Se ha aplicado el tamaño de símbolo predeterminado.",$="Error al leer los datos de entrada.",h='El tipo de serie esperado en el índice ${ seriesIndex } es "${ expectedType }", pero en su lugar se ha recibido "${ receivedType }"',b="o";var P={defaultError:e,negativeValueInDataForLogTransformation:a,negativeValueInDataForSqrtTransformation:r,layerLoadFailure:o,duplicateSeriesID:n,nonNumericAggregation:t,requiredProperty:i,minLength:s,minItems:d,maxItems:l,whiteSpacePattern:u,additionalProperty:m,enumValues:c,anyOfValues:p,bubbleChartValidateMsg:g,queryError:$,invalidSeriesType:h,or:"o"};

exports.additionalProperty = m;
exports.anyOfValues = p;
exports.bubbleChartValidateMsg = g;
exports['default'] = P;
exports.defaultError = e;
exports.duplicateSeriesID = n;
exports.enumValues = c;
exports.invalidSeriesType = h;
exports.layerLoadFailure = o;
exports.maxItems = l;
exports.minItems = d;
exports.minLength = s;
exports.negativeValueInDataForLogTransformation = a;
exports.negativeValueInDataForSqrtTransformation = r;
exports.nonNumericAggregation = t;
exports.or = b;
exports.queryError = $;
exports.requiredProperty = i;
exports.whiteSpacePattern = u;
