// @arcgis/charts-components/dist/cjs/errors.t9n.lv-df11315d-51e288e4.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="Ielādējot diagrammu, radās kļūda.",t="Žurnālfaila transformāciju nevar piemērot negatīvām vai nulles vērtībām.",e="Kvadrātsaknes transformāciju nevar lietot negatīvām vērtībām.",r="Ielādējot slāni, radās kļūda. Vietrādis (URL) = ${ url }. Portāla vienības ID = ${ portalItemId }.",i="${ dataPath } jābūt unikālam. Sērijai ar nosaukumu ${ seriesName } ir ID (${ seriesID }), ko jau izmanto cita sērija.",s="${ dataPath } nevar veikt apkopošanu laukā, kurā nav skaitlisku vērtību.",n="${ dataPath } trūkst parametra ar nosaukumu ${ missingProperty }.",m="${ dataPath } nedrīkst būt īsāks par ${ limit } rakstzīmēm.",d="${ dataPath } nedrīkst būt mazāk par ${ limit } vienībām.",o="${ dataPath } nedrīkst būt vairāk par ${ limit } vienībām.",l="${ dataPath } jābūt vismaz vienai rakstzīmei, kas nav atstarpe.",u="${ dataPath } nedrīkst būt ${ additionalProperty }.",v="${ dataPath } jābūt vienādam ar kādu no šīm atļautajām vērtībām: ${ allowedValues }.",k="${ dataPath } jāatbilst kādai no šādām shēmām: ${ schemaOptions }.",p="Izkliedes diagrammas ar proporcionāliem simboliem netiek atbalstītas. Ticis piemērots noklusējuma simbola izmērs.",$="Neizdevās nolasīt ievades datus.",b="Indeksa ${ seriesIndex } prognozētais sērijas veids ir “${ expectedType }”, bet tā vietā tika saņemts “${ receivedType }”",P="vai";var h={defaultError:a,negativeValueInDataForLogTransformation:t,negativeValueInDataForSqrtTransformation:e,layerLoadFailure:r,duplicateSeriesID:i,nonNumericAggregation:s,requiredProperty:n,minLength:m,minItems:d,maxItems:o,whiteSpacePattern:l,additionalProperty:u,enumValues:v,anyOfValues:k,bubbleChartValidateMsg:p,queryError:$,invalidSeriesType:b,or:"vai"};

exports.additionalProperty = u;
exports.anyOfValues = k;
exports.bubbleChartValidateMsg = p;
exports['default'] = h;
exports.defaultError = a;
exports.duplicateSeriesID = i;
exports.enumValues = v;
exports.invalidSeriesType = b;
exports.layerLoadFailure = r;
exports.maxItems = o;
exports.minItems = d;
exports.minLength = m;
exports.negativeValueInDataForLogTransformation = t;
exports.negativeValueInDataForSqrtTransformation = e;
exports.nonNumericAggregation = s;
exports.or = P;
exports.queryError = $;
exports.requiredProperty = n;
exports.whiteSpacePattern = l;
