// @arcgis/charts-components/dist/cjs/errors.t9n.et-1a8b328a-ea162d3d-07bd1349-8963f9bd.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="Diagrammi laadimisel ilmnes viga.",e="Logi teisendamisel ei saa kasutada negatiivseid väärtusi ega nullväärtusi.",t="Ruutjuure teisendamist ei saa rakendada negatiivsetele väärtustele.",i="Kihi laadimisel tekkis viga. URL = ${ url }. Portaaliüksuse ID = ${ portalItemId }.",s="${ dataPath } peab olema kordumatu. Seerial nimega ${ seriesName } on ID (${ seriesID }), mida kasutatakse juba muudes seeriates.",r="${ dataPath } ei tohi mittenumbrilisel väljal teha loendamisega mitteseotud agregeerimist.",u="Omadusel ${ dataPath } puudub omadus nimega ${ missingProperty }.",l="${ dataPath } ei tohi olla lühem kui ${ limit } märki.",o="Muutujal ${ dataPath } ei tohi olla vähem kui ${ limit } üksust.",n="Muutujal ${ dataPath } ei tohi olla rohkem kui ${ limit } üksust.",d="${ dataPath } peab sisaldama vähemalt ühte märki, mis pole tühimärk.",m="Muutujal ${ dataPath } ei tohi olla: ${ additionalProperty }.",h="${ dataPath } peab võrduma ühega allpool nimetatud väärtustest: ${ allowedValues }.",g="${ dataPath } peab vastama ühele nendest skeemidest: ${ schemaOptions }.",p="Proportsionaalsete sümbolitega hajuvusdiagramme ei toetata. Rakendatud on sümboli vaikesuurus.",v="Sisendandmete lugemine nurjus.",$="Eeldatud seeriatüüp indeksis ${ seriesIndex } on '${ expectedType }', kuid saadi hoopis '${ receivedType }'",k="või";var P={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:t,layerLoadFailure:i,duplicateSeriesID:s,nonNumericAggregation:r,requiredProperty:u,minLength:l,minItems:o,maxItems:n,whiteSpacePattern:d,additionalProperty:m,enumValues:h,anyOfValues:g,bubbleChartValidateMsg:p,queryError:v,invalidSeriesType:$,or:"või"};

exports.additionalProperty = m;
exports.anyOfValues = g;
exports.bubbleChartValidateMsg = p;
exports['default'] = P;
exports.defaultError = a;
exports.duplicateSeriesID = s;
exports.enumValues = h;
exports.invalidSeriesType = $;
exports.layerLoadFailure = i;
exports.maxItems = n;
exports.minItems = o;
exports.minLength = l;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = t;
exports.nonNumericAggregation = r;
exports.or = k;
exports.queryError = v;
exports.requiredProperty = u;
exports.whiteSpacePattern = d;
