// @arcgis/charts-components/dist/cjs/errors.t9n.id-a5bc1429-8b946bef-f2d8a7ea-fd56152e.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="Terjadi kesalahan saat memuat grafik.",e="Tidak dapat menerapkan transformasi log ke nilai negatif atau nol.",i="Tidak dapat menerapkan transformasi akar pangkat dua ke nilai negatif.",t="Terjadi kesalahan saat memuat layer. URL = ${ url }. ID item portal = ${ portalItemId }.",r="${ dataPath } harus unik. Rangkaian yang bernama ${ seriesName } memiliki ID (${ seriesID }) yang sudah digunakan oleh rangkaian lain.",n="${ dataPath } tidak dapat melakukan agregasi non-hitungan di kolom non-numerik.",s="${ dataPath } tidak memiliki properti bernama ${ missingProperty }.",d="${ dataPath } tidak boleh lebih pendek dari ${ limit } karakter.",l="${ dataPath } tidak boleh memiliki kurang dari ${ limit } item.",o="${ dataPath } tidak boleh memiliki lebih dari ${ limit } item.",m="${ dataPath } harus memiliki setidaknya satu karakter non-spasi.",u="${ dataPath } tidak boleh memiliki ${ additionalProperty }.",k="${ dataPath } harus sama dengan salah satu nilai yang diizinkan ini: ${ allowedValues }.",g="${ dataPath } harus cocok dengan skema salah satu dari ini: ${ schemaOptions }.",h="Scatter plot dengan simbol proporsional tidak didukung. Ukuran simbol default sudah diterapkan.",p="Gagal membaca input data.",$="Jenis seri yang diharapkan pada indeks ${ seriesIndex } adalah '${ expectedType }' tetapi '${ receivedType }' diterima sebagai gantinya",y="atau";var b={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:i,layerLoadFailure:t,duplicateSeriesID:r,nonNumericAggregation:n,requiredProperty:s,minLength:d,minItems:l,maxItems:o,whiteSpacePattern:m,additionalProperty:u,enumValues:k,anyOfValues:g,bubbleChartValidateMsg:h,queryError:p,invalidSeriesType:$,or:y};

exports.additionalProperty = u;
exports.anyOfValues = g;
exports.bubbleChartValidateMsg = h;
exports['default'] = b;
exports.defaultError = a;
exports.duplicateSeriesID = r;
exports.enumValues = k;
exports.invalidSeriesType = $;
exports.layerLoadFailure = t;
exports.maxItems = o;
exports.minItems = l;
exports.minLength = d;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = i;
exports.nonNumericAggregation = n;
exports.or = y;
exports.queryError = p;
exports.requiredProperty = s;
exports.whiteSpacePattern = m;
