// @arcgis/charts-components/dist/cjs/errors.t9n.sk-b9987a2e-588e1114-e04ab02b.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="Pri načítavaní grafu sa vyskytla chyba.",e="Nie je možné aplikovať Log transformáciu na záporné alebo nulové hodnoty.",o="Nie je možné aplikovať Square Root transformáciu na záporné hodnoty.",t="Pri načítavaní vrstvy sa vyskytla chyba. URL = ${ url }. ID položky portálu = ${ portalItemId }.",n="${ dataPath } musí byť jedinečný. Séria pomenovaná ${ seriesName } má id (${ seriesID }), ktoré už je používané inou sériou.",r="${ dataPath } nemôže vykonávať agregáciu bez počtov na nečíselnom poli.",i="${ dataPath } chýba vlastnosť s názvom ${ missingProperty }.",s="${ dataPath } musí byť kratší než ${ limit } znakov.",l="${ dataPath } nesmie mať menej než ${ limit } položiek.",m="${ dataPath } nesmie mať viac než ${ limit } položiek.",d="${ dataPath } musí mať aspoň jeden znak bez prázdneho miesta.",u="${ dataPath } nesmie mať ${ additionalProperty }.",p="${ dataPath } musí byť rovné jednej z týchto povolených hodnôt: ${ allowedValues }.",v="${ dataPath } sa musí zhodovať so schémou jednej z týchto: ${ schemaOptions }.",y="Korelačné grafy s proporcionálnymi symbolmi nie sú podporované. Bola aplikovaná predvolená veľkosť symbolu.",h="Zlyhanie načítania vstupných údajov.",$="Očakávaný typ série pri indexe ${ seriesIndex } je '${ expectedType }', ale namiesto toho bolo prijaté '${ receivedType }'",c="alebo";var g={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:o,layerLoadFailure:t,duplicateSeriesID:n,nonNumericAggregation:r,requiredProperty:i,minLength:s,minItems:l,maxItems:m,whiteSpacePattern:d,additionalProperty:u,enumValues:p,anyOfValues:v,bubbleChartValidateMsg:y,queryError:h,invalidSeriesType:$,or:c};

exports.additionalProperty = u;
exports.anyOfValues = v;
exports.bubbleChartValidateMsg = y;
exports['default'] = g;
exports.defaultError = a;
exports.duplicateSeriesID = n;
exports.enumValues = p;
exports.invalidSeriesType = $;
exports.layerLoadFailure = t;
exports.maxItems = m;
exports.minItems = l;
exports.minLength = s;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = o;
exports.nonNumericAggregation = r;
exports.or = c;
exports.queryError = h;
exports.requiredProperty = i;
exports.whiteSpacePattern = d;
