// @arcgis/charts-components/dist/cjs/errors.t9n.et-1a8b328a-d7e70722.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="Diagrammi laadimisel ilmnes viga.",e="Logi teisendamisel ei saa kasutada negatiivseid väärtusi ega nullväärtusi.",t="Ruutjuure teisendamist ei saa rakendada negatiivsetele väärtustele.",i="Kihi laadimisel tekkis viga. URL = ${ url }. Portaaliüksuse ID = ${ portalItemId }.",s="${ dataPath } peab olema kordumatu. Seerial nimega ${ seriesName } on ID (${ seriesID }), mida kasutatakse juba muudes seeriates.",r="${ dataPath } ei tohi mittenumbrilisel väljal teha loendamisega mitteseotud agregeerimist.",u="Omadusel ${ dataPath } puudub omadus nimega ${ missingProperty }.",l="${ dataPath } ei tohi olla lühem kui ${ limit } märki.",d="Muutujal ${ dataPath } ei tohi olla vähem kui ${ limit } üksust.",m="Muutujal ${ dataPath } ei tohi olla rohkem kui ${ limit } üksust.",o="${ dataPath } peab sisaldama vähemalt ühte märki, mis pole tühimärk.",n="Muutujal ${ dataPath } ei tohi olla: ${ additionalProperty }.",h="${ dataPath } peab võrduma ühega allpool nimetatud väärtustest: ${ allowedValues }.",g="${ dataPath } peab vastama ühele nendest skeemidest: ${ schemaOptions }.",$="Proportsionaalsete sümbolitega hajuvusdiagramme ei toetata. Rakendatud on sümboli vaikesuurus.",p="Sisendandmete lugemine nurjus.",k="Eeldatud seeriatüüp indeksis ${ seriesIndex } on '${ expectedType }', kuid saadi hoopis '${ receivedType }'",v="või";var P={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:t,layerLoadFailure:i,duplicateSeriesID:s,nonNumericAggregation:r,requiredProperty:u,minLength:l,minItems:d,maxItems:m,whiteSpacePattern:o,additionalProperty:n,enumValues:h,anyOfValues:g,bubbleChartValidateMsg:$,queryError:p,invalidSeriesType:k,or:"või"};

exports.additionalProperty = n;
exports.anyOfValues = g;
exports.bubbleChartValidateMsg = $;
exports['default'] = P;
exports.defaultError = a;
exports.duplicateSeriesID = s;
exports.enumValues = h;
exports.invalidSeriesType = k;
exports.layerLoadFailure = i;
exports.maxItems = m;
exports.minItems = d;
exports.minLength = l;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = t;
exports.nonNumericAggregation = r;
exports.or = v;
exports.queryError = p;
exports.requiredProperty = u;
exports.whiteSpacePattern = o;
