// @arcgis/charts-components/dist/cjs/errors.t9n.ro-ccdd46b5-062acc13.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const e="A survenit o eroare în timpul încărcării diagramei.",a="Nu se poate aplica o transformare de jurnal în valori negative sau zero.",i="Nu se poate aplica o transformare a rădăcinii pătrate în valori negative.",t="A survenit o eroare în timpul încărcării stratului tematic. URL = ${ url }. ID-ul elementului de portal = ${ portalItemId }.",r="${ dataPath } trebuie să fie unic. Seria denumită ${ seriesName } are un ID (${ seriesID }) utilizat deja de către o altă serie.",n="${ dataPath } nu poate efectua agregarea non-numărare pe un câmp non-numeric.",u="${ dataPath } are lipsă o proprietate denumită ${ missingProperty }.",o="${ dataPath } nu trebuie să fie mai scurt de ${ limit } caractere.",s="${ dataPath } nu trebuie să aibă mai puțin de ${ limit } articole.",l="${ dataPath } nu trebuie să aibă mai mult de ${ limit } articole.",d="${ dataPath } trebuie să aibă cel puțin un caracter non-spațiu gol.",p="${ dataPath } nu trebuie să aibă ${ additionalProperty }.",c="${ dataPath } trebuie să fie egal cu una dintre aceste valori permise: ${ allowedValues }.",m="${ dataPath } trebuie să corespundă schemei uneia dintre acestea: ${ schemaOptions }.",$="Proprietățile diagramei de dispersie cu simbolurile proporționale nu sunt acceptate. A fost aplicată dimensiunea implicită a simbolului.",b="A eșuat citirea datelor de intrare.",g="Tipul așteptat al serie la indicele ${ seriesIndex } este „${ expectedType }”, dar a fost primit „${ receivedType }”",P="sau";var h={defaultError:e,negativeValueInDataForLogTransformation:a,negativeValueInDataForSqrtTransformation:i,layerLoadFailure:t,duplicateSeriesID:r,nonNumericAggregation:n,requiredProperty:u,minLength:o,minItems:s,maxItems:l,whiteSpacePattern:d,additionalProperty:p,enumValues:c,anyOfValues:m,bubbleChartValidateMsg:$,queryError:b,invalidSeriesType:g,or:"sau"};

exports.additionalProperty = p;
exports.anyOfValues = m;
exports.bubbleChartValidateMsg = $;
exports['default'] = h;
exports.defaultError = e;
exports.duplicateSeriesID = r;
exports.enumValues = c;
exports.invalidSeriesType = g;
exports.layerLoadFailure = t;
exports.maxItems = l;
exports.minItems = s;
exports.minLength = o;
exports.negativeValueInDataForLogTransformation = a;
exports.negativeValueInDataForSqrtTransformation = i;
exports.nonNumericAggregation = n;
exports.or = P;
exports.queryError = b;
exports.requiredProperty = u;
exports.whiteSpacePattern = d;
