// @arcgis/charts-components/dist/cjs/errors.t9n.sv-af93438c-aed38463-7396d610.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="Ett fel uppstod när diagrammet lästes in.",e="Det går inte att tillämpa loggomvandling på negativa värden eller nollvärden.",t="Det går inte att tillämpa kvadratrotomvandling på negativa värden.",r="Ett fel uppstod när lagret lästes in. URL = ${ url }. Portalobjekt-ID = ${ portalItemId }.",n="${ dataPath } måste vara unik. Serien som heter ${ seriesName } har ett ID (${ seriesID }) som redan används av en annan serie.",i="${ dataPath } kan inte utföra icke-tidsmässig aggregering på ett icke-numeriskt fält.",s="${ dataPath } saknar egenskapen ${ missingProperty }.",l="${ dataPath } får inte vara kortare än ${ limit } tecken.",o="${ dataPath } får inte ha färre än ${ limit } objekt.",d="${ dataPath } får inte ha fler än ${ limit } objekt.",m="${ dataPath } måste ha minst ett tecken som inte är ett blanksteg.",g="${ dataPath } får inte innehålla ${ additionalProperty }.",u="${ dataPath } måste vara lika med ett av dessa tillåtna värden: ${ allowedValues }.",p="${ dataPath } måste matcha schemat för någon av dessa: ${ schemaOptions }.",h="Punktdiagram med proportionerliga symboler stöds inte. Standardstorleken för symboler har tillämpats.",v="Det gick inte att läsa indata.",$="Förväntad serietyp vid index ${ seriesIndex } är '${ expectedType }' men '${ receivedType }' togs emot istället",f="eller";var P={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:t,layerLoadFailure:r,duplicateSeriesID:n,nonNumericAggregation:i,requiredProperty:s,minLength:l,minItems:o,maxItems:d,whiteSpacePattern:m,additionalProperty:g,enumValues:u,anyOfValues:p,bubbleChartValidateMsg:h,queryError:v,invalidSeriesType:$,or:f};

exports.additionalProperty = g;
exports.anyOfValues = p;
exports.bubbleChartValidateMsg = h;
exports['default'] = P;
exports.defaultError = a;
exports.duplicateSeriesID = n;
exports.enumValues = u;
exports.invalidSeriesType = $;
exports.layerLoadFailure = r;
exports.maxItems = d;
exports.minItems = o;
exports.minLength = l;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = t;
exports.nonNumericAggregation = i;
exports.or = f;
exports.queryError = v;
exports.requiredProperty = s;
exports.whiteSpacePattern = m;
