// @arcgis/charts-components/dist/cjs/errors.t9n.fr-ec692558-314e971e-8be716fe-ff964558.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const e="Une erreur est survenue lors du chargement du diagramme.",a="Impossible d'appliquer une transformation logarithmique à des valeurs négatives ou égales à zéro.",r="Impossible d'appliquer une transformation de racine carrée à des valeurs négatives.",t="Une erreur est survenue lors du chargement de la couche. URL = ${ url }. ID de l'élément du portail = ${ portalItemId }.",s="${ dataPath } doit être unique. La série nommée ${ seriesName } comporte un ID (${ seriesID }) qui est déjà utilisé par une autre série.",n="${ dataPath } ne peut pas réaliser d'agrégation de type autre que Total sur un champ non numérique.",i="Propriété ${ missingProperty } manquante dans ${ dataPath }.",o="${ dataPath } ne doit pas comporter moins de ${ limit } caractères.",u="${ dataPath } ne doit pas comporter moins de ${ limit } éléments.",d="${ dataPath } ne doit pas comporter plus de ${ limit } éléments.",l="${ dataPath } doit contenir au moins un caractère autre qu'un espace.",m="${ dataPath } ne doit pas contenir ${ additionalProperty }.",p="${ dataPath } doit être égal à l'une de ces valeurs autorisées : ${ allowedValues }.",c="${ dataPath } doit correspondre à l'une des structures suivantes : ${ schemaOptions }.",g="Les nuages de points avec des symboles proportionnels ne sont pas pris en charge. La taille de symbole par défaut a été appliquée.",h="Échec de la lecture des données en entrée.",$="Le type de série attendu à l’index ${ seriesIndex } est '${ expectedType }', mais '${ receivedType }' a été reçu à la place",y="ou";var P={defaultError:e,negativeValueInDataForLogTransformation:a,negativeValueInDataForSqrtTransformation:r,layerLoadFailure:t,duplicateSeriesID:s,nonNumericAggregation:n,requiredProperty:i,minLength:o,minItems:u,maxItems:d,whiteSpacePattern:l,additionalProperty:m,enumValues:p,anyOfValues:c,bubbleChartValidateMsg:g,queryError:h,invalidSeriesType:$,or:"ou"};

exports.additionalProperty = m;
exports.anyOfValues = c;
exports.bubbleChartValidateMsg = g;
exports['default'] = P;
exports.defaultError = e;
exports.duplicateSeriesID = s;
exports.enumValues = p;
exports.invalidSeriesType = $;
exports.layerLoadFailure = t;
exports.maxItems = d;
exports.minItems = u;
exports.minLength = o;
exports.negativeValueInDataForLogTransformation = a;
exports.negativeValueInDataForSqrtTransformation = r;
exports.nonNumericAggregation = n;
exports.or = y;
exports.queryError = h;
exports.requiredProperty = i;
exports.whiteSpacePattern = l;
