// @arcgis/charts-components/dist/cjs/errors.t9n.bg-23efbd89-f733d0da.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="Възникна грешка при зареждането на диаграмата.",e="Не може да се приложи логаритмична трансформация към отрицателни или нулеви стойности.",t="Не може да се приложи трансформация с квадратен корен към отрицателни стойности.",r="Възникна грешка при зареждането на слоя. URL = ${ url }. ИД на елемент на портал = ${ portalItemId }.",i="${ dataPath } трябва да бъде унукално. Серията с името ${ seriesName } има ИД (${ seriesID }), който вече се използва от друга серия.",d="${ dataPath } не може да се извършва обобщение, различно от обобщение на брой, за нецифрово поле.",n="${ dataPath } няма име на свойство ${ missingProperty }.",o="${ dataPath } не трябва да има по-малко от ${ limit } символи.",$="${ dataPath } не трябва да има по-малко от ${ limit } елементи.",l="${ dataPath } не трябва да има повече от ${ limit } елементи.",s="${ dataPath } трябва да има поне един символ, който не е интервал.",m="${ dataPath } не трябва да бъде ${ additionalProperty }.",u="${ dataPath } трябва да бъде равна на една от тези позволени стойности: ${ allowedValues }.",P="${ dataPath }трябва да съответства на схемата на една от тези: ${ schemaOptions }.",h="Не се поддържат диаграми на разсейване с пропорционални символи. Използван е размер на символа по подразбиране.",p="Входните данни не могат да се прочетат.",y="Очакваният тип серия в индекс${ seriesIndex } е „${ expectedType }“, но „${ receivedType }“ е получен вместо това",g="или";var I={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:t,layerLoadFailure:r,duplicateSeriesID:i,nonNumericAggregation:d,requiredProperty:n,minLength:o,minItems:$,maxItems:l,whiteSpacePattern:s,additionalProperty:m,enumValues:u,anyOfValues:P,bubbleChartValidateMsg:h,queryError:p,invalidSeriesType:y,or:"или"};

exports.additionalProperty = m;
exports.anyOfValues = P;
exports.bubbleChartValidateMsg = h;
exports['default'] = I;
exports.defaultError = a;
exports.duplicateSeriesID = i;
exports.enumValues = u;
exports.invalidSeriesType = y;
exports.layerLoadFailure = r;
exports.maxItems = l;
exports.minItems = $;
exports.minLength = o;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = t;
exports.nonNumericAggregation = d;
exports.or = g;
exports.queryError = p;
exports.requiredProperty = n;
exports.whiteSpacePattern = s;
