// @arcgis/charts-components/dist/cjs/errors.t9n.th-2b4710ce-3f592d4f.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="เกิดข้อผิดพลาดขึ้นขณะโหลดแผนภูมิ",e="ไม่สามารถใช้การแปลงค่า log กับค่าติดลบหรือศูนย์ได้",t="ไม่สามารถใช้การแปลงรากที่สองกับค่าติดลบได้",r="เกิดข้อผิดพลาดขึ้นขณะโหลดชั้นข้อมูล URL = ${ url } ID รายการพอร์ทัล = ${ portalItemId }",i="${ dataPath } ต้องไม่ซ้ำกัน ซีรีส์ที่ชื่อ ${ seriesName } มี ID (${ seriesID }) ที่ซีรีส์อื่นใช้งานอยู่แล้ว",d="${ dataPath } ไม่สามารถดำเนินการรวมแบบไม่นับกับฟิลด์ที่ไม่ใช่ตัวเลขได้",o="${ dataPath } ไม่มีชื่อคุณสมบัติ ${ missingProperty }",n="${ dataPath } ต้องมีความยาวมากกว่า ${ limit } อักขระ",$="${ dataPath } ต้องมีไม่น้อยกว่า ${ limit } รายการ",l="${ dataPath } ต้องมีไม่เกิน ${ limit } รายการ",s="${ dataPath } ต้องมีอย่างน้อยหนึ่งอักขระที่ไม่ใช่ไวท์สเปซ",m="${ dataPath } ต้องไม่มี ${ additionalProperty }",u="${ dataPath } ต้องเท่ากับหนึ่งในค่าที่อนุญาตเหล่านี้: ${ allowedValues }",P="${ dataPath } ต้องตรงกับรูปแบบของหนึ่งในรายการเหล่านี้: ${ schemaOptions }",h="ไม่รองรับพล็อตการกระจายที่มีสัญลักษณ์ตามสัดส่วน ใช้ขนาดสัญลักษณ์เริ่มต้นแล้ว",p="ไม่สามารถอ่านข้อมูลขาเข้า",I="ประเภทซีรีส์ที่คาดหวังที่ดัชนี ${ seriesIndex } คือ '${ expectedType }' แต่ได้รับ '${ receivedType }' แทน",g="หรือ";var y={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:t,layerLoadFailure:r,duplicateSeriesID:i,nonNumericAggregation:d,requiredProperty:o,minLength:n,minItems:$,maxItems:l,whiteSpacePattern:s,additionalProperty:m,enumValues:u,anyOfValues:P,bubbleChartValidateMsg:h,queryError:p,invalidSeriesType:I,or:g};

exports.additionalProperty = m;
exports.anyOfValues = P;
exports.bubbleChartValidateMsg = h;
exports['default'] = y;
exports.defaultError = a;
exports.duplicateSeriesID = i;
exports.enumValues = u;
exports.invalidSeriesType = I;
exports.layerLoadFailure = r;
exports.maxItems = l;
exports.minItems = $;
exports.minLength = n;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = t;
exports.nonNumericAggregation = d;
exports.or = g;
exports.queryError = p;
exports.requiredProperty = o;
exports.whiteSpacePattern = s;
