// @arcgis/charts-components/dist/cjs/errors.t9n.fi-0826f959-aee584ca-2fb29093-c1cd0713.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="Kaavion lataamisessa tapahtui virhe.",t="Logaritmimuunnosta ei voi käyttää negatiivisiin arvoihin tai nolla-arvoihin.",e="Neliöjuurimuunnosta ei voi käyttää negatiivisiin arvoihin.",i="Karttatason lataamisessa tapahtui virhe. URL = ${ url }. Portaalikohteen tunnus = ${ portalItemId }.",s="Kohteen ${ dataPath } on oltava yksilöllinen. Sarjan ${ seriesName } tunnus (${ seriesID }) on jo toisen sarjan käytössä.",n="${ dataPath } ei saa suorittaa muuta kuin määrän koostamista muussa kuin numerokentässä.",o="Kohteesta ${ dataPath } puuttuu ominaisuus ${ missingProperty }.",r="${ dataPath } ei saa olla alle ${ limit } merkkiä pitkä.",u="Kohteessa ${ dataPath } ei saa olla alle ${ limit } kohdetta.",l="Kohteessa ${ dataPath } ei saa olla yli ${ limit } kohdetta.",m="Kohteessa ${ dataPath } on oltava vähintään yksi muu kuin välimerkki.",h="Kohteessa ${ dataPath } ei saa olla kohdetta ${ additionalProperty }.",d="Kohteen ${ dataPath } tulee olla yhtä suuri kuin jokin näistä sallituista arvoista: ${ allowedValues }.",k="Kohteen ${ dataPath } on vastattava jotakin näistä skeemoista: ${ schemaOptions }.",y="Pistekaavioita, joissa on suhteutettuja symboleja, ei tueta. Symbolin oletuskokoa on käytetty.",v="Lähtöaineiston lukeminen epäonnistui.",p='Odotettu sarjan tyyppi indeksissä ${ seriesIndex } on "${ expectedType }", mutta sen sijaan vastaanotettiin "${ receivedType }"',$="tai";var g={defaultError:a,negativeValueInDataForLogTransformation:t,negativeValueInDataForSqrtTransformation:e,layerLoadFailure:i,duplicateSeriesID:s,nonNumericAggregation:n,requiredProperty:o,minLength:r,minItems:u,maxItems:l,whiteSpacePattern:m,additionalProperty:h,enumValues:d,anyOfValues:k,bubbleChartValidateMsg:y,queryError:v,invalidSeriesType:p,or:"tai"};

exports.additionalProperty = h;
exports.anyOfValues = k;
exports.bubbleChartValidateMsg = y;
exports['default'] = g;
exports.defaultError = a;
exports.duplicateSeriesID = s;
exports.enumValues = d;
exports.invalidSeriesType = p;
exports.layerLoadFailure = i;
exports.maxItems = l;
exports.minItems = u;
exports.minLength = r;
exports.negativeValueInDataForLogTransformation = t;
exports.negativeValueInDataForSqrtTransformation = e;
exports.nonNumericAggregation = n;
exports.or = $;
exports.queryError = v;
exports.requiredProperty = o;
exports.whiteSpacePattern = m;
