// @arcgis/charts-components/dist/cjs/errors.t9n.th-2b4710ce-a9211bec-985ffbc7-a4c99feb.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="เกิดข้อผิดพลาดขึ้นขณะโหลดแผนภูมิ",e="ไม่สามารถใช้การแปลงค่า log กับค่าติดลบหรือศูนย์ได้",t="ไม่สามารถใช้การแปลงรากที่สองกับค่าติดลบได้",r="เกิดข้อผิดพลาดขึ้นขณะโหลดชั้นข้อมูล URL = ${ url } ID รายการพอร์ทัล = ${ portalItemId }",i="${ dataPath } ต้องไม่ซ้ำกัน ซีรีส์ที่ชื่อ ${ seriesName } มี ID (${ seriesID }) ที่ซีรีส์อื่นใช้งานอยู่แล้ว",s="${ dataPath } ไม่สามารถดำเนินการรวมแบบไม่นับกับฟิลด์ที่ไม่ใช่ตัวเลขได้",n="${ dataPath } ไม่มีชื่อคุณสมบัติ ${ missingProperty }",o="${ dataPath } ต้องมีความยาวมากกว่า ${ limit } อักขระ",l="${ dataPath } ต้องมีไม่น้อยกว่า ${ limit } รายการ",d="${ dataPath } ต้องมีไม่เกิน ${ limit } รายการ",u="${ dataPath } ต้องมีอย่างน้อยหนึ่งอักขระที่ไม่ใช่ไวท์สเปซ",m="${ dataPath } ต้องไม่มี ${ additionalProperty }",$="${ dataPath } ต้องเท่ากับหนึ่งในค่าที่อนุญาตเหล่านี้: ${ allowedValues }",g="${ dataPath } ต้องตรงกับรูปแบบของหนึ่งในรายการเหล่านี้: ${ schemaOptions }",p="ไม่รองรับพล็อตการกระจายที่มีสัญลักษณ์ตามสัดส่วน ใช้ขนาดสัญลักษณ์เริ่มต้นแล้ว",P="ไม่สามารถอ่านข้อมูลขาเข้า",h="ประเภทซีรีส์ที่คาดหวังที่ดัชนี ${ seriesIndex } คือ '${ expectedType }' แต่ได้รับ '${ receivedType }' แทน",y="หรือ";var I={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:t,layerLoadFailure:r,duplicateSeriesID:i,nonNumericAggregation:s,requiredProperty:n,minLength:o,minItems:l,maxItems:d,whiteSpacePattern:u,additionalProperty:m,enumValues:$,anyOfValues:g,bubbleChartValidateMsg:p,queryError:P,invalidSeriesType:h,or:y};

exports.additionalProperty = m;
exports.anyOfValues = g;
exports.bubbleChartValidateMsg = p;
exports['default'] = I;
exports.defaultError = a;
exports.duplicateSeriesID = i;
exports.enumValues = $;
exports.invalidSeriesType = h;
exports.layerLoadFailure = r;
exports.maxItems = d;
exports.minItems = l;
exports.minLength = o;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = t;
exports.nonNumericAggregation = s;
exports.or = y;
exports.queryError = P;
exports.requiredProperty = n;
exports.whiteSpacePattern = u;
