// @arcgis/charts-components/dist/cjs/errors.t9n.el-c9207675-caa97d06-29544d03.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="Παρουσιάστηκε κάποιο σφάλμα κατά τη φόρτωση του γραφήματος.",e="Δεν είναι δυνατή η εφαρμογή του μετασχηματισμού αρχείου καταγραφής σε αρνητικές ή μηδενικές τιμές.",t="Δεν είναι δυνατή η εφαρμογή του μετασχηματισμού τετραγωνικής ρίζας σε αρνητικές τιμές.",r="Παρουσιάστηκε κάποιο σφάλμα κατά τη φόρτωση του θεματικού επιπέδου. Διεύθυνση URL = ${ url }. Αναγνωριστικό αντικειμένου του portal = ${ portalItemId }.",i="Το ${ dataPath } πρέπει να είναι μοναδικό. Η σειρά με το όνομα ${ seriesName } έχει αναγνωριστικό (${ seriesID }) που ήδη χρησιμοποιείται από κάποια άλλη σειρά.",s="Το ${ dataPath } δεν μπορεί να εκτελέσει συνάθροιση χωρίς μέτρηση σε μη αριθμητικό πεδίο.",n="Από το ${ dataPath } λείπει μια ιδιότητα με το όνομα ${ missingProperty }.",o="Το ${ dataPath } δεν πρέπει να έχει λιγότερους από ${ limit } χαρακτήρες.",l="Το ${ dataPath } δεν πρέπει να έχει λιγότερα από ${ limit } στοιχεία.",d="Το ${ dataPath } δεν πρέπει να έχει περισσότερα από ${ limit } στοιχεία.",u="Το ${ dataPath } πρέπει να έχει τουλάχιστον έναν χαρακτήρα που να μην είναι κενό διάστημα.",m="Το ${ dataPath } δεν πρέπει να έχει ${ additionalProperty }.",$="Το ${ dataPath } πρέπει να ισούται με μία από τις εξής επιτρεπόμενες τιμές: ${ allowedValues }.",p="Το ${ dataPath } πρέπει να συμφωνεί με τη δομή ενός από τα εξής: ${ schemaOptions }.",P="Δεν υποστηρίζονται γραφήματα διασποράς με αναλογικά σύμβολα. Έχει εφαρμοστεί το προεπιλεγμένο μέγεθος συμβόλου.",g="Αποτυχία ανάγνωσης δεδομένων εισόδου",h="Ο αναμενόμενος τύπος σειράς στο ευρετήριο ${ seriesIndex } είναι '${ expectedType }', αλλά λήφθηκε '${ receivedType }' αντί γι' αυτό",y="ή";var I={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:t,layerLoadFailure:r,duplicateSeriesID:i,nonNumericAggregation:s,requiredProperty:n,minLength:o,minItems:l,maxItems:d,whiteSpacePattern:u,additionalProperty:m,enumValues:$,anyOfValues:p,bubbleChartValidateMsg:P,queryError:g,invalidSeriesType:h,or:"ή"};

exports.additionalProperty = m;
exports.anyOfValues = p;
exports.bubbleChartValidateMsg = P;
exports['default'] = I;
exports.defaultError = a;
exports.duplicateSeriesID = i;
exports.enumValues = $;
exports.invalidSeriesType = h;
exports.layerLoadFailure = r;
exports.maxItems = d;
exports.minItems = l;
exports.minLength = o;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = t;
exports.nonNumericAggregation = s;
exports.or = y;
exports.queryError = g;
exports.requiredProperty = n;
exports.whiteSpacePattern = u;
