// @arcgis/charts-components/dist/cjs/errors.t9n.he-7950cba0-cdcc9e93-094be63a.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See LICENSE.md for details.
 * v4.31.0-next.103
 */
'use strict';

const a="אירעה שגיאה במהלך טעינת התרשים.",e="לא ניתן לבצע טרנספורמציה לוגריתמית על ערכים שליליים או על אפס.",t="לא ניתן לבצע טרנספורמציה של שורש ריבועי על ערכים שליליים.",r="אירעה שגיאה במהלך טעינת השכבה. URL = ${ url }. מזהה פריט פורטל =${ portalItemId }.",i="${ dataPath } חייב/ת להיות ייחודי/ת. לסדרה בשם ${ seriesName } יש מזהה (${ seriesID }) שבו משתמשת כבר סדרה אחרת.",s="ל-${ dataPath } אסור לבצע צבירה שאינה ספירה על שדה לא מספרי.",n="ב-${ dataPath } חסר מאפיין בשם ${ missingProperty }.",o="${ dataPath } לא יכול להיות קצר יות מ-${ limit } תווים.",d="ל-${ dataPath } לא יכולים להיות פחות מ-${ limit } פריטים.",l="ל-${ dataPath } לא יכולים להיות יותר מ-${ limit } פריטים.",u="ל-${ dataPath } חייב להיות לפחות תו אחד שאינו רווח בלתי נראה.",m="ל-${ dataPath } אסור לכלול ${ additionalProperty }.",$="${ dataPath } חייב להיות שווה לאחד מהערכים המותרים הבאים: ${ allowedValues }.",p="${ dataPath } חייב להתאים לסכמה של אחד מאלה: ${ schemaOptions }.",P="תרשימי פיזור עם סמלים פרופורציונליים לא נתמכים. הוחל גודל סמל שמוגדר כברירת מחדל.",g="קריאת נתוני הקלט נכשלה.",h="סוג הסדרה הצפוי במדד ${ seriesIndex } הוא '${ expectedType }', אבל במקום זאת התקבל '${ receivedType }'.",y="או";var I={defaultError:a,negativeValueInDataForLogTransformation:e,negativeValueInDataForSqrtTransformation:t,layerLoadFailure:r,duplicateSeriesID:i,nonNumericAggregation:s,requiredProperty:n,minLength:o,minItems:d,maxItems:l,whiteSpacePattern:u,additionalProperty:m,enumValues:$,anyOfValues:p,bubbleChartValidateMsg:P,queryError:g,invalidSeriesType:h,or:"או"};

exports.additionalProperty = m;
exports.anyOfValues = p;
exports.bubbleChartValidateMsg = P;
exports['default'] = I;
exports.defaultError = a;
exports.duplicateSeriesID = i;
exports.enumValues = $;
exports.invalidSeriesType = h;
exports.layerLoadFailure = r;
exports.maxItems = l;
exports.minItems = d;
exports.minLength = o;
exports.negativeValueInDataForLogTransformation = e;
exports.negativeValueInDataForSqrtTransformation = t;
exports.nonNumericAggregation = s;
exports.or = y;
exports.queryError = g;
exports.requiredProperty = n;
exports.whiteSpacePattern = u;
