// @arcgis/charts-components-react/dist/components.js
/* eslint-disable */
/* tslint:disable */
/* auto-generated react proxies */
import { createReactComponent } from './react-component-lib';
export const ArcgisChartsActionBar = /*@__PURE__*/ createReactComponent('arcgis-charts-action-bar');
export const ArcgisChartsBarChart = /*@__PURE__*/ createReactComponent('arcgis-charts-bar-chart');
export const ArcgisChartsBarChartBeta = /*@__PURE__*/ createReactComponent('arcgis-charts-bar-chart-beta');
export const ArcgisChartsBarLineChart = /*@__PURE__*/ createReactComponent('arcgis-charts-bar-line-chart');
export const ArcgisChartsBoxPlot = /*@__PURE__*/ createReactComponent('arcgis-charts-box-plot');
export const ArcgisChartsGauge = /*@__PURE__*/ createReactComponent('arcgis-charts-gauge');
export const ArcgisChartsHistogram = /*@__PURE__*/ createReactComponent('arcgis-charts-histogram');
export const ArcgisChartsLineChart = /*@__PURE__*/ createReactComponent('arcgis-charts-line-chart');
export const ArcgisChartsLineChartBeta = /*@__PURE__*/ createReactComponent('arcgis-charts-line-chart-beta');
export const ArcgisChartsPieChart = /*@__PURE__*/ createReactComponent('arcgis-charts-pie-chart');
export const ArcgisChartsScatterPlot = /*@__PURE__*/ createReactComponent('arcgis-charts-scatter-plot');
export const ArcgisChartsScatterPlotBeta = /*@__PURE__*/ createReactComponent('arcgis-charts-scatter-plot-beta');
