// @arcgis/map-components/dist/esm/arcgis-basemap-layer-list.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-a701542b.js';
import { u as usePropertyChange } from './index-266be087.js';
import { o as newWidgetsBasemapLayerList } from './index-c73b2050.js';
import { m as makeWidgetController } from './useWidget-f2119dfc.js';
import { u as useControllerManager, g as getSet } from './index-25757424.js';
import './component-utils-5806dcb0.js';

const basemapLayerListCss = "";
const ArcgisBasemapLayerListStyle0 = basemapLayerListCss;

const useBasemapLayerListWidget = makeWidgetController(newWidgetsBasemapLayerList);
const ArcgisBasemapLayerList = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.widgetTriggerAction = createEvent(this, "arcgisTriggerAction", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        this.widget = useBasemapLayerListWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.headingLevel = this.widget.headingLevel;
        this.label = this.widget.label;
        this.referenceElement = undefined;
        this.baseFilterText = this.widget.baseFilterText;
        this.baseItems = getSet(undefined, {
            get: () => this.widget.baseItems,
            set: "ignore",
        });
        this.baseListItemCreatedFunction = this.widget.baseListItemCreatedFunction;
        this.basemapTitle = this.widget.basemapTitle;
        this.catalogLayerList = getSet(undefined, {
            get: () => this.widget.catalogLayerList,
            set: "ignore",
        });
        this.catalogOptions = this.widget.catalogOptions;
        this.collapsed = this.widget.collapsed;
        this.dragEnabled = this.widget.dragEnabled;
        this.editingTitle = this.widget.editingTitle;
        this.filterPlaceholder = this.widget.filterPlaceholder;
        this.icon = this.widget.icon;
        this.knowledgeGraphOptions = this.widget.knowledgeGraphOptions;
        this.minFilterItems = this.widget.minFilterItems;
        this.openTableListItem = this.widget.openTableListItem;
        this.referenceFilterText = this.widget.referenceFilterText;
        this.referenceItems = getSet(undefined, {
            get: () => this.widget.referenceItems,
            set: "ignore",
        });
        this.referenceListItemCreatedFunction = this.widget.referenceListItemCreatedFunction;
        this.selectedItems = this.widget.selectedItems;
        this.selectionMode = this.widget.selectionMode;
        this.tableList = this.widget.tableList;
        this.visibilityAppearance = this.widget.visibilityAppearance;
        this.hideBaseLayers = this.widget.visibleElements.baseLayers;
        this.showCloseButton = this.widget.visibleElements.closeButton;
        this.showCollapseButton = this.widget.visibleElements.collapseButton;
        this.showEditTitleButton = this.widget.visibleElements.editTitleButton;
        this.showErrors = this.widget.visibleElements.errors;
        this.showFilter = this.widget.visibleElements.filter;
        this.hideFlow = this.widget.visibleElements.flow;
        this.hideHeading = this.widget.visibleElements.heading;
        this.hideReferenceLayers = this.widget.visibleElements.referenceLayers;
        this.hideStatusIndicators = this.widget.visibleElements.statusIndicators;
        this.showTemporaryLayerIndicators = this.widget.visibleElements.temporaryLayerIndicators;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return getElement(this); }
};
ArcgisBasemapLayerList.style = ArcgisBasemapLayerListStyle0;

export { ArcgisBasemapLayerList as arcgis_basemap_layer_list };
