// @arcgis/map-components/dist/esm/arcgis-distance-measurement-2d.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-a701542b.js';
import { E as newWidgetsDistanceMeasurement2D } from './index-c73b2050.js';
import { u as usePropertyChange } from './index-266be087.js';
import { m as makeWidgetController } from './useWidget-f2119dfc.js';
import { u as useControllerManager } from './index-25757424.js';
import './component-utils-5806dcb0.js';

const distanceMeasurement2dCss = "";
const ArcgisDistanceMeasurement2dStyle0 = distanceMeasurement2dCss;

const useDistanceMeasurement2dWidget = makeWidgetController(newWidgetsDistanceMeasurement2D);
const ArcgisDistanceMeasurement2d = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        this.widget = useDistanceMeasurement2dWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.unit = this.widget.unit;
        this.unitOptions = this.widget.unitOptions;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return getElement(this); }
};
ArcgisDistanceMeasurement2d.style = ArcgisDistanceMeasurement2dStyle0;

export { ArcgisDistanceMeasurement2d as arcgis_distance_measurement_2d };
