// @arcgis/map-components/dist/esm/arcgis-search.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-a701542b.js';
import { u as usePropertyChange } from './index-266be087.js';
import { n as newWidgetsSearch } from './index-c73b2050.js';
import { m as makeWidgetController } from './useWidget-f2119dfc.js';
import { u as useControllerManager } from './index-25757424.js';
import './component-utils-5806dcb0.js';

const searchCss = "";
const ArcgisSearchStyle0 = searchCss;

const useSearchWidget = makeWidgetController(newWidgetsSearch);
const ArcgisSearch = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.widgetSearchBlur = createEvent(this, "arcgisBlur", 7);
        this.widgetSearchClear = createEvent(this, "arcgisClear", 7);
        this.widgetSearchComplete = createEvent(this, "arcgisComplete", 7);
        this.widgetSearchFocus = createEvent(this, "arcgisFocus", 7);
        this.widgetSearchStart = createEvent(this, "arcgisStart", 7);
        this.widgetSelectResult = createEvent(this, "arcgisSelectResult", 7);
        this.widgetSuggestComplete = createEvent(this, "arcgisSuggestComplete", 7);
        this.widgetSuggestStart = createEvent(this, "arcgisSuggestStart", 7);
        this.manager = useControllerManager(this);
        this.widget = useSearchWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.activeMenu = this.widget.activeMenu;
        this.activeSource = this.widget.activeSource;
        this.activeSourceIndex = this.widget.activeSourceIndex;
        this.allPlaceholder = this.widget.allPlaceholder;
        this.allSources = this.widget.allSources;
        this.autoSelectDisabled = this.widget.autoSelect;
        this.defaultSources = this.widget.defaultSources;
        this.disabled = this.widget.disabled;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.includeDefaultSourcesDisabled = this.widget.includeDefaultSources;
        this.locationDisabled = this.widget.locationEnabled;
        this.maxResults = this.widget.maxResults;
        this.maxSuggestions = this.widget.maxSuggestions;
        this.minSuggestCharacters = this.widget.minSuggestCharacters;
        this.popupDisabled = this.widget.popupEnabled;
        this.popupTemplate = this.widget.popupTemplate;
        this.portal = this.widget.portal;
        this.resultGraphic = this.widget.resultGraphic;
        this.resultGraphicDisabled = this.widget.resultGraphicEnabled;
        this.results = this.widget.results;
        this.searchAllDisabled = this.widget.searchAllEnabled;
        this.searchTerm = this.widget.searchTerm;
        this.selectedResult = this.widget.selectedResult;
        this.sources = this.widget.sources;
        this.suggestions = this.widget.suggestions;
        this.suggestionsDisabled = this.widget.suggestionsEnabled;
        this.position = "top-right";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    async blurSearch() {
        this.widget?.blur();
    }
    async clearSearch() {
        this.widget?.clear();
    }
    async focusSearch() {
        this.widget?.focus();
    }
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async search(term) {
        return await this.widget?.search(term);
    }
    async suggest(term) {
        return await this.widget?.suggest(term);
    }
    get el() { return getElement(this); }
};
ArcgisSearch.style = ArcgisSearchStyle0;

export { ArcgisSearch as arcgis_search };
