// @arcgis/map-components/dist/esm/arcgis-compass.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { r as registerInstance, c as createEvent, h, g as getElement } from './index-a701542b.js';
import { u as usePropertyChange } from './index-266be087.js';
import { q as classes, w as newWidgetsCompassCompassViewModel } from './index-c73b2050.js';
import { u as useT9n, m as makeViewModelController } from './useViewModel-29d4d04a.js';
import { g as globalCss } from './globalCss-4c5375df.js';
import { u as useControllerManager, g as getSet } from './index-25757424.js';
import './component-utils-5806dcb0.js';

const base = "esri-compass";
const css = {
    base,
    iconContainer: `${base}__icon-container`,
};
const useCompassViewModel = makeViewModelController(newWidgetsCompassCompassViewModel);
const ArcgisCompass = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        //--------------------------------------------------------------------------
        //
        //  Controllers
        //
        //--------------------------------------------------------------------------
        this.manager = useControllerManager(this);
        this.messages = useT9n();
        this.viewModel = useCompassViewModel(this);
        //--------------------------------------------------------------------------
        //
        //  Events
        //
        //--------------------------------------------------------------------------
        this.arcgisPropertyChange = usePropertyChange()("state", "orientation");
        //--------------------------------------------------------------------------
        //
        //  Private Methods
        //
        //--------------------------------------------------------------------------
        this._toRotationTransform = (orientation) => ({
            transform: `rotateZ(${orientation.z}deg)`,
        });
        this._reset = () => {
            this.viewModel.reset();
        };
        this.autoDestroyDisabled = false;
        this.icon = getSet(undefined, {
            get: (currentValue) => currentValue || (this.state === "rotation" ? "arrow-up" : "compass-needle"),
        });
        this.label = undefined;
        this.position = "top-left";
        this.orientation = this.viewModel.orientation;
        this.referenceElement = undefined;
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Public Methods
    //
    //--------------------------------------------------------------------------
    render() {
        const { messages, icon } = this;
        return (h("div", { key: 'cc3a9d3a33f522559ca39c371f16be5a8818c849', class: classes(css.base, globalCss.widget) }, h("calcite-button", { key: '6345fb632e7cc08b302641c433bf60982fa30979', class: globalCss.widgetButton, disabled: this.state === "disabled", kind: "neutral", label: messages.reset, onClick: this._reset, round: true, scale: "s", title: messages.reset }, h("div", { key: 'd8518d5c16a8b8f7a4d302c8a6e0e81120a246b3', "aria-hidden": "true",
            // Fix for Calcite's padding inline/block not being consistent
            class: css.iconContainer, title: messages.reset }, h("calcite-icon", { key: '34816c4f9f85884cd7b23fc3f6818c3bfda6e5e7', icon: icon, style: this._toRotationTransform(this.orientation) })))));
    }
    async reset() {
        return this.viewModel.reset();
    }
    async destroy() {
        await this.manager.destroy();
    }
    static get assetsDirs() { return ["assets"]; }
    get el() { return getElement(this); }
};

export { ArcgisCompass as arcgis_compass };
