// @arcgis/map-components/dist/esm/arcgis-expand.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-a701542b.js';
import { H as newWidgetsExpand } from './index-c73b2050.js';
import { u as useControllerManager } from './index-25757424.js';
import { m as makeWidgetController } from './useWidget-f2119dfc.js';
import './component-utils-5806dcb0.js';

const expandCss = "arcgis-expand{pointer-events:auto}arcgis-expand>.esri-expand__content--expanded{background-color:var(--calcite-color-background)}";
const ArcgisExpandStyle0 = expandCss;

const useExpandWidget = makeWidgetController(newWidgetsExpand);
const ArcgisExpand = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.manager = useControllerManager(this);
        this.widget = useExpandWidget(this);
        this.viewModel = this.widget.viewModel;
        this.autoDestroyDisabled = false;
        this.icon = this.widget.icon;
        this.content = this.widget.content;
        this.label = this.widget.label;
        this.autoCollapse = this.widget.autoCollapse;
        this.closeOnEsc = this.widget.closeOnEsc;
        this.collapseIcon = this.widget.collapseIcon;
        this.collapseTooltip = this.widget.collapseTooltip;
        this.expanded = this.widget.expanded;
        this.expandIcon = this.widget.expandIcon;
        this.expandTooltip = this.widget.expandTooltip;
        this.group = this.widget.group;
        this.iconNumber = this.widget.iconNumber;
        this.mode = this.widget.mode;
        this.placement = this.widget.placement;
        this.position = "top-right";
        this.referenceElement = undefined;
        /**
         * This refers to the content inside the expand's popup. This is different
         * from this.el.childElem which refers to the expand's UI (which in part
         * includes the popup with this.content)
         */
        this.content ?? (this.content = document.createElement("div"));
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async collapse() {
        this.widget?.collapse();
    }
    async expand() {
        this.widget?.expand();
    }
    async toggle() {
        if (this.expanded) {
            await this.collapse();
        }
        else {
            await this.expand();
        }
    }
    get el() { return getElement(this); }
};
ArcgisExpand.style = ArcgisExpandStyle0;

export { ArcgisExpand as arcgis_expand };
