// @arcgis/map-components/dist/esm/arcgis-swipe.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-a701542b.js';
import { ad as newWidgetsSwipe } from './index-c73b2050.js';
import { u as usePropertyChange } from './index-266be087.js';
import { m as makeWidgetController } from './useWidget-f2119dfc.js';
import { u as useControllerManager } from './index-25757424.js';
import './component-utils-5806dcb0.js';

const swipeCss = "arcgis-swipe{}";
const ArcgisSwipeStyle0 = swipeCss;

const useSwipeWidget = makeWidgetController(newWidgetsSwipe);
const ArcgisSwipe = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        this.widget = useSwipeWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.direction = this.widget.direction;
        this.dragLabel = this.widget.dragLabel;
        this.leadingLayers = this.widget.leadingLayers;
        this.swipePosition = this.widget.position;
        this.trailingLayers = this.widget.trailingLayers;
        this.position = "manual";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return getElement(this); }
};
ArcgisSwipe.style = ArcgisSwipeStyle0;

export { ArcgisSwipe as arcgis_swipe };
