// @arcgis/map-components/dist/esm/arcgis-floor-filter.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-a701542b.js';
import { O as newWidgetsFloorFilter } from './index-c73b2050.js';
import { m as makeWidgetController } from './useWidget-f2119dfc.js';
import { u as useControllerManager } from './index-25757424.js';
import './component-utils-5806dcb0.js';

const floorFilterCss = "";
const ArcgisFloorFilterStyle0 = floorFilterCss;

const useFloorFilterWidget = makeWidgetController(newWidgetsFloorFilter);
const ArcgisFloorFilter = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.manager = useControllerManager(this);
        this.widget = useFloorFilterWidget(this);
        this.viewModel = this.widget.viewModel;
        this.autoDestroyDisabled = false;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.referenceElement = undefined;
        this.facility = this.widget.facility;
        this.headingLevel = this.widget.headingLevel;
        this.level = this.widget.level;
        this.longNames = this.widget.longNames;
        this.site = this.widget.site;
        this.position = "bottom-left";
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async updateWebDocument(webMap) {
        this.widget?.updateWebDocument(webMap);
    }
    get el() { return getElement(this); }
};
ArcgisFloorFilter.style = ArcgisFloorFilterStyle0;

export { ArcgisFloorFilter as arcgis_floor_filter };
