// @arcgis/map-components/dist/esm/arcgis-legend.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-a701542b.js';
import { U as newWidgetsLegend } from './index-c73b2050.js';
import { u as usePropertyChange } from './index-266be087.js';
import { m as makeWidgetController } from './useWidget-f2119dfc.js';
import { u as useControllerManager } from './index-25757424.js';
import './component-utils-5806dcb0.js';

const legendCss = "";
const ArcgisLegendStyle0 = legendCss;

const useLegendWidget = makeWidgetController(newWidgetsLegend);
const ArcgisLegend = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        this.widget = useLegendWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.headingLevel = this.widget.headingLevel;
        this.referenceElement = undefined;
        this.activeLayerInfos = this.widget.activeLayerInfos;
        this.basemapLegendVisible = this.widget.basemapLegendVisible;
        this.hideLayersNotInCurrentView = this.widget.hideLayersNotInCurrentView;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.respectLayerVisibilityDisabled = this.widget.respectLayerVisibility;
        this.legendStyle = this.widget.style;
        this.layerInfos = this.widget.layerInfos;
        this.position = "bottom-left";
        this.respectLayerDefinitionExpression = this.widget.respectLayerDefinitionExpression;
        this.state = this.viewModel.state;
    }
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return getElement(this); }
};
ArcgisLegend.style = ArcgisLegendStyle0;

export { ArcgisLegend as arcgis_legend };
