// @arcgis/map-components/dist/esm/arcgis-building-explorer.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-a701542b.js';
import { u as newWidgetsBuildingExplorer } from './index-c73b2050.js';
import { m as makeWidgetController } from './useWidget-f2119dfc.js';
import { u as usePropertyChange } from './index-266be087.js';
import { u as useControllerManager } from './index-25757424.js';
import './component-utils-5806dcb0.js';

const buildingExplorerCss = "";
const ArcgisBuildingExplorerStyle0 = buildingExplorerCss;

const useBuildingExplorerWidget = makeWidgetController(newWidgetsBuildingExplorer);
const ArcgisBuildingExplorer = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        this.widget = useBuildingExplorerWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("level", "phase", "state");
        this.autoDestroyDisabled = false;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.layers = this.widget.layers;
        this.headingLevel = this.widget.headingLevel;
        this.level = this.viewModel.level;
        this.phase = this.viewModel.phase;
        this.state = this.viewModel.state;
        this.hideLevels = this.widget.visibleElements.levels;
        this.hidePhases = this.widget.visibleElements.phases;
        this.hideDisciplines = this.widget.visibleElements.disciplines;
        this.position = "bottom-left";
        this.referenceElement = undefined;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return getElement(this); }
};
ArcgisBuildingExplorer.style = ArcgisBuildingExplorerStyle0;

export { ArcgisBuildingExplorer as arcgis_building_explorer };
