// @arcgis/map-components/dist/esm/arcgis-sketch.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-a701542b.js';
import { u as usePropertyChange } from './index-266be087.js';
import { a9 as newLayersGraphicsLayer, aa as newWidgetsSketch } from './index-c73b2050.js';
import { m as makeWidgetController } from './useWidget-f2119dfc.js';
import { u as useControllerManager } from './index-25757424.js';
import './component-utils-5806dcb0.js';

const sketchCss = "arcgis-sketch{}";
const ArcgisSketchStyle0 = sketchCss;

const useSketchWidget = makeWidgetController(newWidgetsSketch);
const ArcgisSketch = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.widgetCreate = createEvent(this, "arcgisCreate", 7);
        this.widgetDelete = createEvent(this, "arcgisDelete", 7);
        this.widgetRedo = createEvent(this, "arcgisRedo", 7);
        this.widgetUndo = createEvent(this, "arcgisUndo", 7);
        this.widgetUpdate = createEvent(this, "arcgisUpdate", 7);
        this.manager = useControllerManager(this);
        this.widget = useSketchWidget(this, { editConstructorProperties: this._editConstructorProperties });
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.activeTool = this.widget.activeTool;
        this.availableCreateTools = this.widget.availableCreateTools;
        this.createGraphic = this.widget.createGraphic;
        this.creationMode = this.widget.creationMode;
        this.defaultCreateOptions = this.widget.defaultCreateOptions;
        this.defaultUpdateOptions = this.widget.defaultUpdateOptions;
        this.defaultGraphicsLayerDisabled = false;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.labelOptions = this.widget.labelOptions;
        this.layer = this.widget.layer;
        this.layout = this.widget.layout;
        this.pointSymbol = this.viewModel.pointSymbol;
        this.polylineSymbol = this.viewModel.polylineSymbol;
        this.polygonSymbol = this.viewModel.polygonSymbol;
        this.snappingOptions = this.widget.snappingOptions;
        this.tooltipOptions = this.widget.tooltipOptions;
        this.updateGraphics = this.widget.updateGraphics;
        this.hideCreateToolsPoint = this.widget.visibleElements.createTools.point;
        this.hideCreateToolsPolyline = this.widget.visibleElements.createTools.polyline;
        this.hideCreateToolsPolygon = this.widget.visibleElements.createTools.polygon;
        this.hideCreateToolsRectangle = this.widget.visibleElements.createTools.rectangle;
        this.hideCreateToolsCircle = this.widget.visibleElements.createTools.circle;
        this.hideDuplicateButton = this.widget.visibleElements.duplicateButton;
        this.hideSelectionToolsRectangleSelection = this.widget.visibleElements.selectionTools["rectangle-selection"];
        this.hideSelectionToolsLassoSelection = this.widget.visibleElements.selectionTools["lasso-selection"];
        this.hideSettingsMenu = this.widget.visibleElements.settingsMenu;
        this.hideLabelsToggle = this.widget.visibleElements.labelsToggle;
        this.hideTooltipsToggle = this.widget.visibleElements.tooltipsToggle;
        this.hideSnappingControls = this.widget.visibleElements.snappingControls;
        this.showSnappingControlsElementsHeader = this.widget.visibleElements.snappingControlsElements.header;
        this.hideSnappingControlsElementsEnabledToggle = this.widget.visibleElements.snappingControlsElements.enabledToggle;
        this.hideSnappingControlsElementsSelfEnabledToggle = this.widget.visibleElements.snappingControlsElements.selfEnabledToggle;
        this.hideSnappingControlsElementsFeatureEnabledToggle = this.widget.visibleElements.snappingControlsElements.featureEnabledToggle;
        this.hideSnappingControlsElementsLayerList = this.widget.visibleElements.snappingControlsElements.layerList;
        this.hideUndoRedoMenu = this.widget.visibleElements.undoRedoMenu;
        this.position = "bottom-left";
        this.state = this.widget.state;
    }
    async _editConstructorProperties(properties) {
        if (!this.defaultGraphicsLayerDisabled) {
            properties.layer ?? (properties.layer = await newLayersGraphicsLayer({
                title: "Sketch Layer",
            }));
        }
        return properties;
    }
    //--------------------------------------------------------------------------
    //
    //  Listeners
    //
    //--------------------------------------------------------------------------
    arcgisReadyHandler() {
        if (!this.defaultGraphicsLayerDisabled && !this.el.view?.map.layers.includes(this.widget.layer)) {
            this.el.view?.map.add(this.widget.layer);
        }
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    async cancel() {
        this.widget?.cancel();
    }
    async complete() {
        this.widget?.complete();
    }
    async create(tool, createOptions) {
        await this.widget?.create(tool, createOptions);
    }
    async delete() {
        this.widget?.delete();
    }
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async duplicate() {
        this.widget?.duplicate();
    }
    async redo() {
        this.widget?.redo();
    }
    async undo() {
        this.widget?.undo();
    }
    async update(graphic, updateOptions) {
        return void this.widget?.update(graphic, updateOptions);
    }
    get el() { return getElement(this); }
};
ArcgisSketch.style = ArcgisSketchStyle0;

export { ArcgisSketch as arcgis_sketch };
