// @arcgis/map-components/dist/esm/arcgis-feature-form.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.0-next.58
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-a701542b.js';
import { J as newWidgetsFeatureForm } from './index-c73b2050.js';
import { u as usePropertyChange } from './index-266be087.js';
import { m as makeWidgetController } from './useWidget-f2119dfc.js';
import { u as useControllerManager } from './index-25757424.js';
import './component-utils-5806dcb0.js';

const featureFormCss = "";
const ArcgisFeatureFormStyle0 = featureFormCss;

const useFeatureFormWidget = makeWidgetController(newWidgetsFeatureForm);
const ArcgisFeatureForm = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.widgetSubmit = createEvent(this, "arcgisSubmit", 7);
        this.widgetValueChange = createEvent(this, "arcgisValueChange", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        this.widget = useFeatureFormWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.headingLevel = this.widget.headingLevel;
        this.referenceElement = undefined;
        this.position = "bottom-left";
        this.feature = this.widget.feature;
        this.formTemplate = this.widget.formTemplate;
        this.groupDisplay = this.widget.groupDisplay;
        this.layer = this.widget.layer;
        this.timeZone = this.widget.timeZone;
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async getValues() {
        return this.widget?.getValues();
    }
    async submit() {
        return this.widget?.submit();
    }
    get el() { return getElement(this); }
};
ArcgisFeatureForm.style = ArcgisFeatureFormStyle0;

export { ArcgisFeatureForm as arcgis_feature_form };
