// @esri/calcite-components-react/dist/components.js
/* eslint-disable */
/* tslint:disable */
/* auto-generated react proxies */
import { createReactComponent } from './react-component-lib';
export const CalciteAccordion = /*@__PURE__*/ createReactComponent('calcite-accordion');
export const CalciteAccordionItem = /*@__PURE__*/ createReactComponent('calcite-accordion-item');
export const CalciteAction = /*@__PURE__*/ createReactComponent('calcite-action');
export const CalciteActionBar = /*@__PURE__*/ createReactComponent('calcite-action-bar');
export const CalciteActionGroup = /*@__PURE__*/ createReactComponent('calcite-action-group');
export const CalciteActionMenu = /*@__PURE__*/ createReactComponent('calcite-action-menu');
export const CalciteActionPad = /*@__PURE__*/ createReactComponent('calcite-action-pad');
export const CalciteAlert = /*@__PURE__*/ createReactComponent('calcite-alert');
export const CalciteAvatar = /*@__PURE__*/ createReactComponent('calcite-avatar');
export const CalciteBlock = /*@__PURE__*/ createReactComponent('calcite-block');
export const CalciteBlockSection = /*@__PURE__*/ createReactComponent('calcite-block-section');
export const CalciteButton = /*@__PURE__*/ createReactComponent('calcite-button');
export const CalciteCard = /*@__PURE__*/ createReactComponent('calcite-card');
export const CalciteCardGroup = /*@__PURE__*/ createReactComponent('calcite-card-group');
export const CalciteCarousel = /*@__PURE__*/ createReactComponent('calcite-carousel');
export const CalciteCarouselItem = /*@__PURE__*/ createReactComponent('calcite-carousel-item');
export const CalciteCheckbox = /*@__PURE__*/ createReactComponent('calcite-checkbox');
export const CalciteChip = /*@__PURE__*/ createReactComponent('calcite-chip');
export const CalciteChipGroup = /*@__PURE__*/ createReactComponent('calcite-chip-group');
export const CalciteColorPicker = /*@__PURE__*/ createReactComponent('calcite-color-picker');
export const CalciteColorPickerHexInput = /*@__PURE__*/ createReactComponent('calcite-color-picker-hex-input');
export const CalciteColorPickerSwatch = /*@__PURE__*/ createReactComponent('calcite-color-picker-swatch');
export const CalciteCombobox = /*@__PURE__*/ createReactComponent('calcite-combobox');
export const CalciteComboboxItem = /*@__PURE__*/ createReactComponent('calcite-combobox-item');
export const CalciteComboboxItemGroup = /*@__PURE__*/ createReactComponent('calcite-combobox-item-group');
export const CalciteDatePicker = /*@__PURE__*/ createReactComponent('calcite-date-picker');
export const CalciteDatePickerDay = /*@__PURE__*/ createReactComponent('calcite-date-picker-day');
export const CalciteDatePickerMonth = /*@__PURE__*/ createReactComponent('calcite-date-picker-month');
export const CalciteDatePickerMonthHeader = /*@__PURE__*/ createReactComponent('calcite-date-picker-month-header');
export const CalciteDialog = /*@__PURE__*/ createReactComponent('calcite-dialog');
export const CalciteDropdown = /*@__PURE__*/ createReactComponent('calcite-dropdown');
export const CalciteDropdownGroup = /*@__PURE__*/ createReactComponent('calcite-dropdown-group');
export const CalciteDropdownItem = /*@__PURE__*/ createReactComponent('calcite-dropdown-item');
export const CalciteFab = /*@__PURE__*/ createReactComponent('calcite-fab');
export const CalciteFilter = /*@__PURE__*/ createReactComponent('calcite-filter');
export const CalciteFlow = /*@__PURE__*/ createReactComponent('calcite-flow');
export const CalciteFlowItem = /*@__PURE__*/ createReactComponent('calcite-flow-item');
export const CalciteGraph = /*@__PURE__*/ createReactComponent('calcite-graph');
export const CalciteHandle = /*@__PURE__*/ createReactComponent('calcite-handle');
export const CalciteIcon = /*@__PURE__*/ createReactComponent('calcite-icon');
export const CalciteInlineEditable = /*@__PURE__*/ createReactComponent('calcite-inline-editable');
export const CalciteInput = /*@__PURE__*/ createReactComponent('calcite-input');
export const CalciteInputDatePicker = /*@__PURE__*/ createReactComponent('calcite-input-date-picker');
export const CalciteInputMessage = /*@__PURE__*/ createReactComponent('calcite-input-message');
export const CalciteInputNumber = /*@__PURE__*/ createReactComponent('calcite-input-number');
export const CalciteInputText = /*@__PURE__*/ createReactComponent('calcite-input-text');
export const CalciteInputTimePicker = /*@__PURE__*/ createReactComponent('calcite-input-time-picker');
export const CalciteInputTimeZone = /*@__PURE__*/ createReactComponent('calcite-input-time-zone');
export const CalciteLabel = /*@__PURE__*/ createReactComponent('calcite-label');
export const CalciteLink = /*@__PURE__*/ createReactComponent('calcite-link');
export const CalciteList = /*@__PURE__*/ createReactComponent('calcite-list');
export const CalciteListItem = /*@__PURE__*/ createReactComponent('calcite-list-item');
export const CalciteListItemGroup = /*@__PURE__*/ createReactComponent('calcite-list-item-group');
export const CalciteLoader = /*@__PURE__*/ createReactComponent('calcite-loader');
export const CalciteMenu = /*@__PURE__*/ createReactComponent('calcite-menu');
export const CalciteMenuItem = /*@__PURE__*/ createReactComponent('calcite-menu-item');
export const CalciteMeter = /*@__PURE__*/ createReactComponent('calcite-meter');
export const CalciteModal = /*@__PURE__*/ createReactComponent('calcite-modal');
export const CalciteNavigation = /*@__PURE__*/ createReactComponent('calcite-navigation');
export const CalciteNavigationLogo = /*@__PURE__*/ createReactComponent('calcite-navigation-logo');
export const CalciteNavigationUser = /*@__PURE__*/ createReactComponent('calcite-navigation-user');
export const CalciteNotice = /*@__PURE__*/ createReactComponent('calcite-notice');
export const CalciteOption = /*@__PURE__*/ createReactComponent('calcite-option');
export const CalciteOptionGroup = /*@__PURE__*/ createReactComponent('calcite-option-group');
export const CalcitePagination = /*@__PURE__*/ createReactComponent('calcite-pagination');
export const CalcitePanel = /*@__PURE__*/ createReactComponent('calcite-panel');
export const CalcitePickList = /*@__PURE__*/ createReactComponent('calcite-pick-list');
export const CalcitePickListGroup = /*@__PURE__*/ createReactComponent('calcite-pick-list-group');
export const CalcitePickListItem = /*@__PURE__*/ createReactComponent('calcite-pick-list-item');
export const CalcitePopover = /*@__PURE__*/ createReactComponent('calcite-popover');
export const CalciteProgress = /*@__PURE__*/ createReactComponent('calcite-progress');
export const CalciteRadioButton = /*@__PURE__*/ createReactComponent('calcite-radio-button');
export const CalciteRadioButtonGroup = /*@__PURE__*/ createReactComponent('calcite-radio-button-group');
export const CalciteRating = /*@__PURE__*/ createReactComponent('calcite-rating');
export const CalciteScrim = /*@__PURE__*/ createReactComponent('calcite-scrim');
export const CalciteSegmentedControl = /*@__PURE__*/ createReactComponent('calcite-segmented-control');
export const CalciteSegmentedControlItem = /*@__PURE__*/ createReactComponent('calcite-segmented-control-item');
export const CalciteSelect = /*@__PURE__*/ createReactComponent('calcite-select');
export const CalciteSheet = /*@__PURE__*/ createReactComponent('calcite-sheet');
export const CalciteShell = /*@__PURE__*/ createReactComponent('calcite-shell');
export const CalciteShellCenterRow = /*@__PURE__*/ createReactComponent('calcite-shell-center-row');
export const CalciteShellPanel = /*@__PURE__*/ createReactComponent('calcite-shell-panel');
export const CalciteSlider = /*@__PURE__*/ createReactComponent('calcite-slider');
export const CalciteSortableList = /*@__PURE__*/ createReactComponent('calcite-sortable-list');
export const CalciteSplitButton = /*@__PURE__*/ createReactComponent('calcite-split-button');
export const CalciteStack = /*@__PURE__*/ createReactComponent('calcite-stack');
export const CalciteStepper = /*@__PURE__*/ createReactComponent('calcite-stepper');
export const CalciteStepperItem = /*@__PURE__*/ createReactComponent('calcite-stepper-item');
export const CalciteSwitch = /*@__PURE__*/ createReactComponent('calcite-switch');
export const CalciteTab = /*@__PURE__*/ createReactComponent('calcite-tab');
export const CalciteTabNav = /*@__PURE__*/ createReactComponent('calcite-tab-nav');
export const CalciteTabTitle = /*@__PURE__*/ createReactComponent('calcite-tab-title');
export const CalciteTable = /*@__PURE__*/ createReactComponent('calcite-table');
export const CalciteTableCell = /*@__PURE__*/ createReactComponent('calcite-table-cell');
export const CalciteTableHeader = /*@__PURE__*/ createReactComponent('calcite-table-header');
export const CalciteTableRow = /*@__PURE__*/ createReactComponent('calcite-table-row');
export const CalciteTabs = /*@__PURE__*/ createReactComponent('calcite-tabs');
export const CalciteTextArea = /*@__PURE__*/ createReactComponent('calcite-text-area');
export const CalciteTile = /*@__PURE__*/ createReactComponent('calcite-tile');
export const CalciteTileGroup = /*@__PURE__*/ createReactComponent('calcite-tile-group');
export const CalciteTileSelect = /*@__PURE__*/ createReactComponent('calcite-tile-select');
export const CalciteTileSelectGroup = /*@__PURE__*/ createReactComponent('calcite-tile-select-group');
export const CalciteTimePicker = /*@__PURE__*/ createReactComponent('calcite-time-picker');
export const CalciteTip = /*@__PURE__*/ createReactComponent('calcite-tip');
export const CalciteTipGroup = /*@__PURE__*/ createReactComponent('calcite-tip-group');
export const CalciteTipManager = /*@__PURE__*/ createReactComponent('calcite-tip-manager');
export const CalciteTooltip = /*@__PURE__*/ createReactComponent('calcite-tooltip');
export const CalciteTree = /*@__PURE__*/ createReactComponent('calcite-tree');
export const CalciteTreeItem = /*@__PURE__*/ createReactComponent('calcite-tree-item');
export const CalciteValueList = /*@__PURE__*/ createReactComponent('calcite-value-list');
export const CalciteValueListItem = /*@__PURE__*/ createReactComponent('calcite-value-list-item');
