// @esri/calcite-components/dist/cjs/da-d76c8300.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/dev/LICENSE.md for details.
 * v2.13.2
 */
'use strict';

const calciteInputTimePicker_calciteTimePicker_entry = require('./calcite-input-time-picker.calcite-time-picker-4dca8b09.js');
require('./index-712430e2.js');
require('./form-3ea27d18.js');
require('./dom-a6db8c35.js');
require('./guid-76f7196a.js');
require('./resources-ebb1faa2.js');
require('./interactive-079bc430.js');
require('./key-d12cbf1a.js');
require('./label-bb2c2446.js');
require('./component-e7debf62.js');
require('./loadable-ee725b32.js');
require('./browser-443e5c0b.js');
require('./locale-d86a5a22.js');
require('./observers-a1f0d377.js');
require('./focusTrapComponent-0d845485.js');
require('./config-abc4f9c0.js');
require('./math-5c6cbb10.js');
require('./t9n-d7efdb8b.js');
require('./Validation-a2a5b9da.js');
require('./input-4e31156b.js');

// Danish [da]
var locale = {
  name: 'da',
  weekdays: 'søndag_mandag_tirsdag_onsdag_torsdag_fredag_lørdag'.split('_'),
  weekdaysShort: 'søn._man._tirs._ons._tors._fre._lør.'.split('_'),
  weekdaysMin: 'sø._ma._ti._on._to._fr._lø.'.split('_'),
  months: 'januar_februar_marts_april_maj_juni_juli_august_september_oktober_november_december'.split('_'),
  monthsShort: 'jan._feb._mar._apr._maj_juni_juli_aug._sept._okt._nov._dec.'.split('_'),
  weekStart: 1,
  yearStart: 4,
  ordinal: function ordinal(n) {
    return n + ".";
  },
  formats: {
    LT: 'HH:mm',
    LTS: 'HH:mm:ss',
    L: 'DD.MM.YYYY',
    LL: 'D. MMMM YYYY',
    LLL: 'D. MMMM YYYY HH:mm',
    LLLL: 'dddd [d.] D. MMMM YYYY [kl.] HH:mm'
  },
  relativeTime: {
    future: 'om %s',
    past: '%s siden',
    s: 'få sekunder',
    m: 'et minut',
    mm: '%d minutter',
    h: 'en time',
    hh: '%d timer',
    d: 'en dag',
    dd: '%d dage',
    M: 'en måned',
    MM: '%d måneder',
    y: 'et år',
    yy: '%d år'
  }
};
calciteInputTimePicker_calciteTimePicker_entry.dayjs.locale(locale, null, true);

exports['default'] = locale;
