// @esri/calcite-components/dist/cjs/fr-86d6a6b7.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/dev/LICENSE.md for details.
 * v2.13.2
 */
'use strict';

const calciteInputTimePicker_calciteTimePicker_entry = require('./calcite-input-time-picker.calcite-time-picker-4dca8b09.js');
require('./index-712430e2.js');
require('./form-3ea27d18.js');
require('./dom-a6db8c35.js');
require('./guid-76f7196a.js');
require('./resources-ebb1faa2.js');
require('./interactive-079bc430.js');
require('./key-d12cbf1a.js');
require('./label-bb2c2446.js');
require('./component-e7debf62.js');
require('./loadable-ee725b32.js');
require('./browser-443e5c0b.js');
require('./locale-d86a5a22.js');
require('./observers-a1f0d377.js');
require('./focusTrapComponent-0d845485.js');
require('./config-abc4f9c0.js');
require('./math-5c6cbb10.js');
require('./t9n-d7efdb8b.js');
require('./Validation-a2a5b9da.js');
require('./input-4e31156b.js');

// French [fr]
var locale = {
  name: 'fr',
  weekdays: 'dimanche_lundi_mardi_mercredi_jeudi_vendredi_samedi'.split('_'),
  weekdaysShort: 'dim._lun._mar._mer._jeu._ven._sam.'.split('_'),
  weekdaysMin: 'di_lu_ma_me_je_ve_sa'.split('_'),
  months: 'janvier_février_mars_avril_mai_juin_juillet_août_septembre_octobre_novembre_décembre'.split('_'),
  monthsShort: 'janv._févr._mars_avr._mai_juin_juil._août_sept._oct._nov._déc.'.split('_'),
  weekStart: 1,
  yearStart: 4,
  formats: {
    LT: 'HH:mm',
    LTS: 'HH:mm:ss',
    L: 'DD/MM/YYYY',
    LL: 'D MMMM YYYY',
    LLL: 'D MMMM YYYY HH:mm',
    LLLL: 'dddd D MMMM YYYY HH:mm'
  },
  relativeTime: {
    future: 'dans %s',
    past: 'il y a %s',
    s: 'quelques secondes',
    m: 'une minute',
    mm: '%d minutes',
    h: 'une heure',
    hh: '%d heures',
    d: 'un jour',
    dd: '%d jours',
    M: 'un mois',
    MM: '%d mois',
    y: 'un an',
    yy: '%d ans'
  },
  ordinal: function ordinal(n) {
    var o = n === 1 ? 'er' : '';
    return "" + n + o;
  }
};
calciteInputTimePicker_calciteTimePicker_entry.dayjs.locale(locale, null, true);

exports['default'] = locale;
