// @esri/calcite-components/dist/esm/en-ca-32846c7d.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/dev/LICENSE.md for details.
 * v2.13.2
 */
import { d as dayjs } from './calcite-input-time-picker.calcite-time-picker-c0364dc5.js';
import './index-abe3b66e.js';
import './form-8ec29459.js';
import './dom-b6dedd88.js';
import './guid-9c230b6a.js';
import './resources-a23d2491.js';
import './interactive-1a42577a.js';
import './key-3ee05994.js';
import './label-043ae25a.js';
import './component-5f0a8be8.js';
import './loadable-13e9ffb2.js';
import './browser-618ac057.js';
import './locale-60f8ca58.js';
import './observers-9c166451.js';
import './focusTrapComponent-7cfd1d69.js';
import './config-974b1499.js';
import './math-682795d0.js';
import './t9n-75defee3.js';
import './Validation-c59cb4eb.js';
import './input-871f5b21.js';

// English (Canada) [en-ca]
var locale = {
  name: 'en-ca',
  weekdays: 'Sunday_Monday_Tuesday_Wednesday_Thursday_Friday_Saturday'.split('_'),
  months: 'January_February_March_April_May_June_July_August_September_October_November_December'.split('_'),
  weekdaysShort: 'Sun_Mon_Tue_Wed_Thu_Fri_Sat'.split('_'),
  monthsShort: 'Jan_Feb_Mar_Apr_May_Jun_Jul_Aug_Sep_Oct_Nov_Dec'.split('_'),
  weekdaysMin: 'Su_Mo_Tu_We_Th_Fr_Sa'.split('_'),
  ordinal: function ordinal(n) {
    return n;
  },
  formats: {
    LT: 'h:mm A',
    LTS: 'h:mm:ss A',
    L: 'YYYY-MM-DD',
    LL: 'MMMM D, YYYY',
    LLL: 'MMMM D, YYYY h:mm A',
    LLLL: 'dddd, MMMM D, YYYY h:mm A'
  },
  relativeTime: {
    future: 'in %s',
    past: '%s ago',
    s: 'a few seconds',
    m: 'a minute',
    mm: '%d minutes',
    h: 'an hour',
    hh: '%d hours',
    d: 'a day',
    dd: '%d days',
    M: 'a month',
    MM: '%d months',
    y: 'a year',
    yy: '%d years'
  }
};
dayjs.locale(locale, null, true);

export { locale as default };
