// @esri/calcite-components/dist/esm/hi-277b3bf8.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/dev/LICENSE.md for details.
 * v2.13.2
 */
import { d as dayjs } from './calcite-input-time-picker.calcite-time-picker-c0364dc5.js';
import './index-abe3b66e.js';
import './form-8ec29459.js';
import './dom-b6dedd88.js';
import './guid-9c230b6a.js';
import './resources-a23d2491.js';
import './interactive-1a42577a.js';
import './key-3ee05994.js';
import './label-043ae25a.js';
import './component-5f0a8be8.js';
import './loadable-13e9ffb2.js';
import './browser-618ac057.js';
import './locale-60f8ca58.js';
import './observers-9c166451.js';
import './focusTrapComponent-7cfd1d69.js';
import './config-974b1499.js';
import './math-682795d0.js';
import './t9n-75defee3.js';
import './Validation-c59cb4eb.js';
import './input-871f5b21.js';

// Hindi [hi]
var locale = {
  name: 'hi',
  weekdays: 'रविवार_सोमवार_मंगलवार_बुधवार_गुरूवार_शुक्रवार_शनिवार'.split('_'),
  months: 'जनवरी_फ़रवरी_मार्च_अप्रैल_मई_जून_जुलाई_अगस्त_सितम्बर_अक्टूबर_नवम्बर_दिसम्बर'.split('_'),
  weekdaysShort: 'रवि_सोम_मंगल_बुध_गुरू_शुक्र_शनि'.split('_'),
  monthsShort: 'जन._फ़र._मार्च_अप्रै._मई_जून_जुल._अग._सित._अक्टू._नव._दिस.'.split('_'),
  weekdaysMin: 'र_सो_मं_बु_गु_शु_श'.split('_'),
  ordinal: function ordinal(n) {
    return n;
  },
  formats: {
    LT: 'A h:mm बजे',
    LTS: 'A h:mm:ss बजे',
    L: 'DD/MM/YYYY',
    LL: 'D MMMM YYYY',
    LLL: 'D MMMM YYYY, A h:mm बजे',
    LLLL: 'dddd, D MMMM YYYY, A h:mm बजे'
  },
  relativeTime: {
    future: '%s में',
    past: '%s पहले',
    s: 'कुछ ही क्षण',
    m: 'एक मिनट',
    mm: '%d मिनट',
    h: 'एक घंटा',
    hh: '%d घंटे',
    d: 'एक दिन',
    dd: '%d दिन',
    M: 'एक महीने',
    MM: '%d महीने',
    y: 'एक वर्ष',
    yy: '%d वर्ष'
  }
};
dayjs.locale(locale, null, true);

export { locale as default };
