// @esri/calcite-components/dist/esm/lv-257e61b3.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/dev/LICENSE.md for details.
 * v2.13.2
 */
import { d as dayjs } from './calcite-input-time-picker.calcite-time-picker-c0364dc5.js';
import './index-abe3b66e.js';
import './form-8ec29459.js';
import './dom-b6dedd88.js';
import './guid-9c230b6a.js';
import './resources-a23d2491.js';
import './interactive-1a42577a.js';
import './key-3ee05994.js';
import './label-043ae25a.js';
import './component-5f0a8be8.js';
import './loadable-13e9ffb2.js';
import './browser-618ac057.js';
import './locale-60f8ca58.js';
import './observers-9c166451.js';
import './focusTrapComponent-7cfd1d69.js';
import './config-974b1499.js';
import './math-682795d0.js';
import './t9n-75defee3.js';
import './Validation-c59cb4eb.js';
import './input-871f5b21.js';

// Latvian [lv]
var locale = {
  name: 'lv',
  weekdays: 'svētdiena_pirmdiena_otrdiena_trešdiena_ceturtdiena_piektdiena_sestdiena'.split('_'),
  months: 'janvāris_februāris_marts_aprīlis_maijs_jūnijs_jūlijs_augusts_septembris_oktobris_novembris_decembris'.split('_'),
  weekStart: 1,
  weekdaysShort: 'Sv_P_O_T_C_Pk_S'.split('_'),
  monthsShort: 'jan_feb_mar_apr_mai_jūn_jūl_aug_sep_okt_nov_dec'.split('_'),
  weekdaysMin: 'Sv_P_O_T_C_Pk_S'.split('_'),
  ordinal: function ordinal(n) {
    return n;
  },
  formats: {
    LT: 'HH:mm',
    LTS: 'HH:mm:ss',
    L: 'DD.MM.YYYY.',
    LL: 'YYYY. [gada] D. MMMM',
    LLL: 'YYYY. [gada] D. MMMM, HH:mm',
    LLLL: 'YYYY. [gada] D. MMMM, dddd, HH:mm'
  },
  relativeTime: {
    future: 'pēc %s',
    past: 'pirms %s',
    s: 'dažām sekundēm',
    m: 'minūtes',
    mm: '%d minūtēm',
    h: 'stundas',
    hh: '%d stundām',
    d: 'dienas',
    dd: '%d dienām',
    M: 'mēneša',
    MM: '%d mēnešiem',
    y: 'gada',
    yy: '%d gadiem'
  }
};
dayjs.locale(locale, null, true);

export { locale as default };
