// @esri/calcite-components/dist/esm/ro-309a5d9e.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/dev/LICENSE.md for details.
 * v2.13.2
 */
import { d as dayjs } from './calcite-input-time-picker.calcite-time-picker-c0364dc5.js';
import './index-abe3b66e.js';
import './form-8ec29459.js';
import './dom-b6dedd88.js';
import './guid-9c230b6a.js';
import './resources-a23d2491.js';
import './interactive-1a42577a.js';
import './key-3ee05994.js';
import './label-043ae25a.js';
import './component-5f0a8be8.js';
import './loadable-13e9ffb2.js';
import './browser-618ac057.js';
import './locale-60f8ca58.js';
import './observers-9c166451.js';
import './focusTrapComponent-7cfd1d69.js';
import './config-974b1499.js';
import './math-682795d0.js';
import './t9n-75defee3.js';
import './Validation-c59cb4eb.js';
import './input-871f5b21.js';

// Romanian [ro]
var locale = {
  name: 'ro',
  weekdays: 'Duminică_Luni_Marți_Miercuri_Joi_Vineri_Sâmbătă'.split('_'),
  weekdaysShort: 'Dum_Lun_Mar_Mie_Joi_Vin_Sâm'.split('_'),
  weekdaysMin: 'Du_Lu_Ma_Mi_Jo_Vi_Sâ'.split('_'),
  months: 'Ianuarie_Februarie_Martie_Aprilie_Mai_Iunie_Iulie_August_Septembrie_Octombrie_Noiembrie_Decembrie'.split('_'),
  monthsShort: 'Ian._Febr._Mart._Apr._Mai_Iun._Iul._Aug._Sept._Oct._Nov._Dec.'.split('_'),
  weekStart: 1,
  formats: {
    LT: 'H:mm',
    LTS: 'H:mm:ss',
    L: 'DD.MM.YYYY',
    LL: 'D MMMM YYYY',
    LLL: 'D MMMM YYYY H:mm',
    LLLL: 'dddd, D MMMM YYYY H:mm'
  },
  relativeTime: {
    future: 'peste %s',
    past: 'acum %s',
    s: 'câteva secunde',
    m: 'un minut',
    mm: '%d minute',
    h: 'o oră',
    hh: '%d ore',
    d: 'o zi',
    dd: '%d zile',
    M: 'o lună',
    MM: '%d luni',
    y: 'un an',
    yy: '%d ani'
  },
  ordinal: function ordinal(n) {
    return n;
  }
};
dayjs.locale(locale, null, true);

export { locale as default };
