// @esri/calcite-components/dist/collection/custom-theme/accordion-item.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/dev/LICENSE.md for details.
 * v2.13.2
 */
import { placeholderImage } from "../../.storybook/placeholder-image";
import { html } from "../../support/formatting";
export const accordionItemTokens = {
    calciteAccordionItemBackgroundColor: "",
    calciteAccordionItemBorderColor: "",
    calciteAccordionItemContentSpace: "",
    calciteAccordionItemEndIconColor: "",
    calciteAccordionItemExpandIconColor: "",
    calciteAccordionItemHeaderBackgroundColor: "",
    calciteAccordionItemHeadingTextColor: "",
    calciteAccordionItemIconColor: "",
    calciteAccordionItemStartIconColor: "",
    calciteAccordionItemTextColor: "",
};
export const accordionItem = (idx) => html `<calcite-accordion-item heading="${idx === 0 ? "Accordion Item" : `Accordion Item ${idx + 1}`}"
    ><img src="${placeholderImage({ width: 300, height: 200 })}" />
  </calcite-accordion-item>`;
