// @esri/calcite-components/dist/collection/custom-theme/chips.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/dev/LICENSE.md for details.
 * v2.13.2
 */
import { html } from "../../support/formatting";
export const chipTokens = {
    calciteChipBackgroundColor: "",
    calciteChipBorderColor: "",
    calciteChipCornerRadius: "",
    calciteChipTextColor: "",
    calciteChipIconColor: "",
    calciteChipCloseIconColor: "",
    calciteChipSelectIconColor: "",
    calciteChipSelectIconColorPressed: "",
};
export const chips = html `<div>
    <calcite-chip>Neutral</calcite-chip>
    <calcite-chip kind="inverse">Inverse</calcite-chip>
    <calcite-chip kind="brand">Brand</calcite-chip>
  </div>
  <div>
    <calcite-chip appearance="outline-fill">Neutral</calcite-chip>
    <calcite-chip appearance="outline-fill" kind="inverse">Inverse</calcite-chip>
    <calcite-chip appearance="outline-fill" kind="brand">Brand</calcite-chip>
  </div>
  <div>
    <calcite-chip appearance="outline">Neutral</calcite-chip>
    <calcite-chip appearance="outline" kind="inverse">Inverse</calcite-chip>
    <calcite-chip appearance="outline" kind="brand">Brand</calcite-chip>
  </div>`;
